---
title: "Population Structure"
author: "Joe McGirr"
date: '`r Sys.Date()`'
output:
  html_document:
    code_folding: hide
    collapsed: no
    df_print: paged
    number_sections: yes
    theme: paper
    toc: yes
    toc_depth: 5
    toc_float: yes
  html_notebook:
    toc: yes
    toc_depth: 5
editor_options: 
  chunk_output_type: console
---



```{css, echo=FALSE}
pre, code {white-space:pre !important; overflow-x:auto}
```

# Summary
This notebook generates scripts to perform population structure analyses with [PCAngsd](http://www.popgen.dk/software/index.php/PCAngsd),
[PLINK PCA](https://www.cog-genomics.org/plink/1.9/strat), [NGSadmix](http://www.popgen.dk/software/index.php/NgsAdmix), and [popVAE](https://github.com/kr-colab/popvae).

The data set is a filtered `.vcf` with 200k SNPs across 892 Pacific herring low coverage genomes:

Written by Joe McGirr, postdoc in Andrew Whitehead's lab.

# Load R libraries
```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}

knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(tidyverse)
library(reticulate)
library(ggpubr)
library(sf)
library(rnaturalearth)
library(rnaturalearthdata)
library(ggspatial)
library(pophelper)
library(rgeos)
library(data.table)

# color-blind friendly 
# Wong, B. Points of view: Color blindness. Nat Methods (2011).
bla <- "#000000"
blu <- "#0072b2"
grb <- "#56b4e9"
lir <- "#cc79a7"
gre <- "#009e73"
red <- "#d55e00"
org <- "#e69f00"
yel <- "#f0e442"
gry<-  '#BBBBBB'

```

# Population structure

## PCAngsd

```{bash,eval=FALSE,class.source = 'fold-show'}
python pcangsd.py -beagle input_file.gz -o output_name
```


### Plot
```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}

pop_info <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/EVOS_MasterSheet_JoeMcGirr_April2020_plate_rows.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
aligned <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/aligned_samples.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
pop_info <- pop_info[pop_info$Sample %in% aligned$sample,]
vcf_samples <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/plink/plates_1_through_5_rm.txt", header = FALSE, stringsAsFactors = FALSE, sep = "\t")
names(vcf_samples)[names(vcf_samples)=="V1"] <- "Sample"
vcf_sample_info <- inner_join(vcf_samples,pop_info, by = "Sample")

C <- as.matrix(read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/pcangsd/ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05.cov"))
e <- eigen(C)
pca <-data.frame(Sample=vcf_sample_info$Sample, 
                 pop=vcf_sample_info$Population.Year,
                 plate=vcf_sample_info$Sequence.Plate,
                 plate_row=vcf_sample_info$plate_row,
                 PC1=e$vectors[,1],PC2=e$vectors[,2],
                 PC3=e$vectors[,3],PC4=e$vectors[,4],
                 PC5=e$vectors[,5],PC6=e$vectors[,6],
                 PC7=e$vectors[,7],PC8=e$vectors[,8],
                 PC9=e$vectors[,9],PC10=e$vectors[,10],
                 stringsAsFactors=FALSE)

# % variance explained, explained 
# https://stats.stackexchange.com/questions/22569/pca-and-proportion-of-variance-explained
# https://aaronschlegel.me/principal-component-analysis-r-example.html

pca <- pca %>% separate(pop, into = c("pop", "year"), sep = "(?<=[A-Za-z])(?=[0-9])")

prop_explained <- c()
for (s in e$values[1:10]) {
  #print(s / sum(e$values))
  prop_explained <- c(prop_explained,round(((s / sum(e$values))*100),2))
}

barplot(prop_explained, ylab = "% variance explained by PC", xlab = "PC",
        names.arg=c(1:length(prop_explained)))

ggplot(data = pca)+
  geom_point(data = pca, aes(x = PC1, y = PC2, fill = pop, color = pop, shape = year), size = 3)+
  scale_fill_manual(values=c(grb,org,red,yel,blu,lir), guide=FALSE)+
  scale_color_manual(name = "population", values=c(grb,org,red,yel,blu,lir))+
  scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
  xlab(paste("PC 1: ", prop_explained[1],"%\n",sep = ""))+
  ylab(paste("PC 2: ", prop_explained[2],"%\n",sep = ""))+
  theme_bw()#+
  ggtitle("all populations and sampling years")


```

## PLINK PCA

```{bash,eval=FALSE,class.source = 'fold-show'}
plink --file input_file --pca --out output_name

```

### Plot

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}

eigenvec <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/plink/ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05.eigenvec", header=F, stringsAsFactors = F)
eigenval <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/plink/ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05.eigenval", header=F, stringsAsFactors = F)


 names(eigenvec)[names(eigenvec)=="V1"] <- "Sample"
 vcf_sample_info <- inner_join(eigenvec,pop_info, by = "Sample")
 
 pca <-data.frame(sample=vcf_sample_info$Sample, 
                  pop=vcf_sample_info$Population.Year,
                  PC1=vcf_sample_info$V3,
                  PC2=vcf_sample_info$V4,
                  PC3=vcf_sample_info$V5,
                  PC4=vcf_sample_info$V6,
                  stringsAsFactors=FALSE)
 
 prop_explained <- c()
 for (s in eigenval$V1) {
   #print(s / sum(eigenval$V1))
   prop_explained <- c(prop_explained,round(((s / sum(eigenval$V1))*100),2))
 }
 
 barplot(prop_explained, ylab = "% variance explained by PC", xlab = "PC",
         names.arg=c(1:length(prop_explained)))
 
 pca <- pca %>% separate(pop, into = c("pop", "year"), sep = "(?<=[A-Za-z])(?=[0-9])")
 pca1 <- pca
 
 ggplot(data = pca1)+
   geom_point(data = pca1, aes(x = PC1, y = PC2, fill = pop, color = pop, shape = year), size = 3)+
   scale_fill_manual(values=c(grb,org,red,yel,blu,lir), guide=FALSE)+
   scale_color_manual(name = "population", values=c(grb,org,red,yel,blu,lir))+
   scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
   xlab(paste("PC1: ", prop_explained[1],"%",sep = ""))+
   ylab(paste("PC2: ", prop_explained[2],"%",sep = ""))+
   theme_bw()


```


## NGSAdmix

```{bash,eval=FALSE,class.source = 'fold-show'}
for i in {1..3} 
do 
NGSadmix -likes input_file.gz -K $i -o output_name_k$i  
done 

```

### Plot

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}

pop_info <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/EVOS_MasterSheet_JoeMcGirr_April2020.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
aligned <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/aligned_samples.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
pop_info <- pop_info[pop_info$Sample %in% aligned$sample,]

vcf_samples <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/plink/plates_1_through_5_rm.txt", header = FALSE, stringsAsFactors = FALSE, sep = "\t")
names(vcf_samples)[names(vcf_samples)=="V1"] <- "Sample"
vcf_sample_info <- inner_join(vcf_samples,pop_info, by = "Sample")

sfiles <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/ngsadmix/qfiles/", full.names=T)
slist <- readQ(files=sfiles,indlabfromfile = FALSE)

onelabset <-vcf_sample_info[,3,drop=FALSE] 
rownames(slist[[3]]) <- vcf_sample_info$Sample
 
subsetpops <- c("BC17","CA17","PWS17","SS17","TB17","WA17")

for (i in c(12,14,15)){
p1 <- plotQ(slist[i],returnplot=T,exportplot=F,basesize=17,showindlab=F,useindlab=T,
            sortind="all",grplab = onelabset,subsetgrp=subsetpops,
            grplabsize=6,linesize=2,pointsize=8, grplabangle=0, grplabjust=0.3, grplabheight=7,
            divcol="white",divtype=2, divsize=1,clustercol=c(grb,org,red,yel,blu,lir,gry))
plot(p1$plot[[1]])
}

# choose K with CLUMPAK
#https://github.com/alexkrohn/AmargosaVoleTutorials/blob/master/ngsAdmix_tutorial.md


```

## Popvae

```{bash,eval=FALSE,class.source = 'fold-show'}
popvae.py --infile input.vcf.gz --out output_name --seed 42 

```

### Plot
```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}

#theme_set(theme_classic())

pop_info <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/EVOS_MasterSheet_JoeMcGirr_April2020.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
aligned <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/aligned_samples.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
pop_info <- pop_info[pop_info$Sample %in% aligned$sample,]
vcf_samples <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/plink/plates_1_through_5_rm.txt", header = FALSE, stringsAsFactors = FALSE, sep = "\t")
names(vcf_samples)[names(vcf_samples)=="V1"] <- "Sample"
vcf_sample_info <- inner_join(vcf_samples,pop_info, by = "Sample")
vcf_sample_info$sampleID <- vcf_sample_info$Sample

pd <- fread("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/popvae/ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05_latent_coords.txt",header=T)
names(pd)[1:2] <- c("LD1","LD2")
dd <- merge(pd,vcf_sample_info,by="sampleID")
dd <- dd %>% separate(Population.Year, into = c("pop", "year"), sep = "(?<=[A-Za-z])(?=[0-9])")
ggplot(data = dd)+
  geom_point(data = dd, aes(x = LD1, y = LD2, fill = pop, color = pop, shape = year), size = 3)+
  scale_fill_manual(values=c(grb,org,red,yel,blu,lir), guide=FALSE)+
  scale_color_manual(name = "population", values=c(grb,org,red,yel,blu,lir))+
  scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
  xlab("LD1")+
  ylab("LD2")+
  theme_bw()


```

