---
title: "Demultiplexing"
author: "Sara Michele Schaal"
date: "5/20/2020"
output: html_document
---
### Stacks 

## Add stack output file names to summary file
This code may need altering depending on your needs for file and sample naming. For the provided dataset, sample IDs are in the form of two digit year,  "_" three digit sample ID, "_", and then two character species ID (i.e., Gm for Gadus morhua). We will download a file that includes the 8 columns with data already populated and 6 columns that this code will populate. The prefilled columns are: Column 1- sample ids, Column 2 - population IDs, Column 3 - i5 primer ID, Column 4 - i7 primer ID, Column 5 - Adapter ID, Column 6 -  i5 sequence, Column 7 - i7 sequence, Column 8 - adapter sequence. The columns we will populate are file_name_R1, file_name_R2, file_name_R1rem, and file_name_R2rem, & PopID. These are just so we have a record of the names of the files that will be created for each sample after stacks is run. 
```{r}
# set your paths
MYINPATH <- "<input your in path here>"
MYOUTPATH <- "<input your out path here>"

## download your sample file
df.samples <- read.csv(paste0(MYINPATH, "SequencePlateOrder_workingGroup.csv"))
head(df.samples)
str(df.samples)

## input your file path to a datafile with a pop ID for each population in your dataset
df.popIDs <- read.csv(paste0(MYINPATH, "Population_IDs.csv"))
df.popIDs

## These two lines will use the SampleID column to identify the year the sample was collected and the three digit sample identifier. These may need to be edited depending on the naming scheme you use for your samples. 
year00 <- substr(df.samples$SampleID, start= 1, stop= 2)
sample000 <- substr(df.samples$SampleID, start = 4, stop = 6)

## Now paste those two values together to get our five digit unique sample identifier
df.samples$Sample5digit <- paste(year00, sample000, sep = "")

## this loops through the population ids and the data file to make the output file names in the format that stacks will return them
for(i in 1:nrow(df.popIDs)){
  for(j in 1:nrow(df.samples)){
    if(df.popIDs$PopLoc[i] == df.samples$Population[j]){
     df.samples$file_name_R1[j] <- paste(df.popIDs$PopID[i], "_", df.samples$Sample5digit[j], ".1.fq.gz", sep = "")
     df.samples$file_name_R2[j] <- paste(df.popIDs$PopID[i], "_", df.samples$Sample5digit[j], ".2.fq.gz", sep = "")
     df.samples$file_name_R1rem[j] <- paste(df.popIDs$PopID[i], "_", df.samples$Sample5digit[j], ".rem.1.fq.gz", sep = "")
     df.samples$file_name_R2rem[j] <- paste(df.popIDs$PopID[i], "_", df.samples$Sample5digit[j], ".rem.2.fq.gz", sep = "")
     df.samples$PopID[j] <- paste(df.popIDs$PopID[i])
    }
  }
}

# Now make the final new sample id    
df.samples$NewSampleID <- paste(df.samples$PopID, "_", df.samples$Sample5digit, sep = "")

# Write the new dataframe to a new csv file
write.csv(df.samples, paste0(MYOUTPATH,"SequencePlateOrder_20220128.csv"))

## Remove samples that we aren't using due to library prep mistake
write.table(df.samples, paste0(MYOUTPATH, "SampleNames.txt", row.names = FALSE, col.names = FALSE))

```


## Create stacks barcode files
```{r}
# if you don't have stringr installed already uncomment the line below and run it once. You won't need to run it again.
#install.packages("stringr")
library(stringr)

# this takes your primer ids from the main datafile used above and makes a vector of all the unique primer combinations
primer_combos <- unique(paste(df.samples$i5.Primer, df.samples$i7.Primer, sep = ""))

# this uploads the file with sequences of the unique adapter barcodes used 
adapterSeq <- read.csv(paste0(MYINPATH, "Barcode_Sequences.csv", header = FALSE))
colnames(adapterSeq) <- c("AdaptNum", "AdaptSeq")

# for each primer combination we now need to make our barcode file for use in Stacks  
for(i in 1:length(primer_combos)){
  prim.combo.adapters <- NULL
  population.id <- NULL
  ind.id <- NULL
  adapter.num <- NULL
  
  # this if else statement is making file names four the barcode files in the format "barcodeFilei5#_i7#.txt"  
  # where the # is the id of the barcode which can be either 1 or 2 digits which is handled by the if/else
  if(str_count(primer_combos[i]) == 2){
    file.name <- paste("barcodeFilei5", substr(primer_combos[i], start = 1, stop = 1), "_i7", 
                       substr(primer_combos[i], start = 2, stop = 2), ".txt", sep = "")
  } else {
    file.name <- paste("barcodeFilei5", substr(primer_combos[i], start = 1, stop = 1), "_i7", 
                       substr(primer_combos[i], start = 2, stop = 3), ".txt", sep = "")
  }
  
  # this for loop steps through the rows in our sample data frame and if the i5 primer and i7 primer
  # match the current primer_combos from the outer for loop then it grabs the adapter sequence, the population ID,
  # the individual sample ID, and the adapter number ID and adds it to a building vector
  for(j in 1:nrow(df.samples)){
    if(paste(df.samples$i5.Primer[j], df.samples$i7.Primer[j], sep = "") == primer_combos[i]){
      prim.combo.adapters <- c(prim.combo.adapters, paste(df.samples$adapter.sequence[j]))
      population.id <- c(population.id, paste(df.samples$PopID[j]))
      ind.id <- c(ind.id, paste(df.samples$Sample5digit[j]))
      adapter.num <- c(adapter.num, df.samples$Adapter[j])
    }
  }
  # Once we have all the samples with the current i5 and i7 primer combination and their associated adapter sequences
  # we make a table with the vectors we just built and output it in the format needed for Stacks. The final table that
  # you output may differ depending on if you did paired end sequencing or not. We had the same barcode on both 
  # the forward and reverse reads and so stacks requires the barcode file to have two columns with the same sequence
  # and then the third column to have the sample id used for stacks. In our case, it was popID_sampleID.
  
  barcodetable <- data.frame(prim.combo.adapters, prim.combo.adapters, 
                             paste(population.id, "_", ind.id, sep = ""))
  colnames(barcodetable) <- NULL
  rownames(barcodetable) <- NULL
  write.table(barcodetable, file = paste0(MYOUTPATH, "BarcodeFiles/", file.name), 
              sep = "\t", col.names = FALSE, 
              row.names = FALSE, quote = FALSE)
 }

```

