---
layout: default
title: Whole Genome Resequencing
parent: Choosing a PopGenomics Approach
grand_parent: Best Principles
has_children: true
#editor_options: 
#  chunk_output_type: console
---



# Choosing a PopGenomics Approach  
{: .no_toc }

<details open markdown="block">
  <summary>
    Table of contents
  </summary>
  {: .text-delta }
1. TOC
{:toc}
</details>

## Whole Genome Resequencing for Population Genomics  
For the purposes of this guide, we use "whole genome (re)sequencing (WGS)) to refer to  methods where a reference genome exists (whether for the focal species or a related species), and samples are sequenced and then mapped to most or all of the reference genome. This method can provide high density SNPs, structural information...for population genomic analyses. This is separate from whole-genome de novo sequencing, which aims to produce a reference genome by sequencing and assembling a complete genome for the first time.   
WGS can be divided into three main approaches, and choosing between these or RADseq(link to ) mostly depends on your questions of interest.   
1) high coverage individual WGS (hcWGS)  
+ description  
2) low coverage individual WGS (lcWGS)  
+ description
3) pooled individuals (Poolseq)  
RADseq is... and m



 also a factor of your funds and how you choose to allocate them. There is a financial trade-off between the number of samples you can genotype individually and the amount of sequencing coverage per sample. If you have 1000s of samples, it may not be financially feasible to sequence most of them to high depth.  Large genome sizes can also make hcWGS cost-prohibited. This topic was discussed during the MarineOmics WGS panel seminar (link).    
An excellent and thorough review of the pros  between these approaches, esp. as they apply to nonmodel organisms, is [Fuentes-Pardo and Ruzzante 2017)(link). 

### Tutorials  

Here we provide a couple different tutorials by our working group members, as well as links to other detailed tutorials on the web. The goal of the MarineOmics tutorials are to provide extensive detail on "why" certain parameters are chosen, and some guidance on how to evaluate different parameter options to fit your data.  

* Fastq-to-VCF pipeline(link): In-depth explanation of automated short-read mapping and variant calling pipeline maintained by [Harvard Informatics](https://github.com/harvardinformatics/shortRead_mapping_variantCalling). Useful for moderate to high coverage data, where *add description of why one would use pipeline*  
* 

## Considerations Before Lab Work


