<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Best Principles in Genomics Research</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="tutorial.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MarineOmics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="ADMIN_04_best_principles.html">Best Principles</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Contributions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ADMIN_01_submissions_instructions.html">Guide for Building a Page</a>
    </li>
    <li>
      <a href="ADMIN_02_contributions.html">Past and Current Contributors</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Population Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="POP_01_choosing_population_genetics.html">Choosing a Population Genomics Approach</a>
    </li>
    <li>
      <a href="POP_04_WGS_intro.html">Whole Genome Resequencing</a>
    </li>
    <li>
      <a href="RADseq.html">Reduced Representation Sequencing</a>
    </li>
    <li>
      <a href="POP_03_poolseq.html">Poolseq</a>
    </li>
    <li>
      <a href="RDAtraitPredictionTutorial.html">Redundancy Analysis (RDA) Trait Prediction</a>
    </li>
    <li>
      <a href="POP_08_PCA.html">PCA</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Functional Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="DGE_comparison_v2.html">Mutifactorial RNAseq</a>
    </li>
    <li>
      <a href="FUN_02_DNA_methylation.html">DNA Methylation Assessment</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Genome-Phenome
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">coming soon!</li>
  </ul>
</li>
<li>
  <a href="ADMIN_03_panels.html">Panel Seminars</a>
</li>
<li>
  <a href="https://github.com/MarineOmics/marineomics.github.io/discussions">Discussion Forum</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'G-53GH9PV49T', 'auto');
  ga('send', 'pageview');

</script>

<div id="header">



<h1 class="title toc-ignore">Best Principles in Genomics Research</h1>

</div>


<p>The backbone underlying the content on the website comes from a
general set of “best principles” that should be applied in genomics
studies, irrespective of the specific sequencing method used. Instead of
providing rigid “best practices”, the website promotes “best
principles”, defined as a guiding set of values and goals that can be
tailored to one’s hypotheses and guided by one’s data, as opposed to a
specific set of instructions. Motivated by rigor and reproducibility,
“best principles” are designed to encourage data exploration and
critical thinking during analysis and evaluation instead of ticking off
boxes on a protocol or step-list. These principles are outlined
below.</p>
<p><br></p>
<div id="rigor" class="section level2">
<h2>Rigor</h2>
<div
id="understand-the-characteristics-of-your-chosen-sequencing-approach.-take-these-characteristics-into-account-when-designing-a-study-and-during-data-analysis."
class="section level3 unlisted unnumbered">
<h3 class="unlisted unnumbered">1. Understand the characteristics of
your chosen sequencing approach. Take these characteristics into account
when designing a study and during data analysis.</h3>
<p>Goals of study should be chosen before choosing the best sequencing
approach, which will inform the total number of samples and sequencing
coverage needed: e.g., PoolSeq requires larger sample sizes and deeper
coverage given the lack of individual genotyping <span
class="citation">(Guirao-Rico and González 2021)</span>.</p>
<p><br></p>
</div>
<div
id="plot-your-data-early-and-often.-get-to-know-it-in-both-its-raw-and-processed-forms."
class="section level3 unlisted unnumbered">
<h3 class="unlisted unnumbered">2. Plot your data early and often. Get
to know it in both its raw and processed forms.</h3>
<p>Deepen the interpretation of results and flag sources of error
throughout a workflow by plotting data such as (i) read-quality metrics
pre- and post-filtering, (ii) sequence coverage across a reference and
across samples, (iii) principal component analysis of replicates pre-
and post-filtering, and (iv) results and predictions of statistical
tests.</p>
<p><br></p>
</div>
<div
id="all-models-and-pipelines-introduce-some-type-and-magnitude-of-error.-compare-models-nuances-to-find-the-best-approach-given-your-data."
class="section level3 unlisted unnumbered">
<h3 class="unlisted unnumbered">3. All models and pipelines introduce
some type and magnitude of error. Compare models’ nuances to find the
best approach given your data.</h3>
<p>This issue is particularly acute in non-model species. Some
quantitative approaches towards evaluating methods include (i) comparing
the performance of different methods or parameter choices using
simulated data <span class="citation">(Lotterhos, Fitzpatrick, and
Blackmon 2022)</span>, (ii) measuring their predictive strengths using
model selection statistics [<span class="citation">(Hooten and Hobbs
2015)</span>, <span class="citation">(Johnson and Omland 2004)</span>],
and (iii) observed-predicted plots from model outputs. A basic
understanding of the sensitivity of inference in different analyses will
be helpful for determining how robust the results are to nuanced
decisions, especially for non-model organisms or unique experimental
designs.</p>
<p><br></p>
</div>
</div>
<div id="reproducibility" class="section level2">
<h2>Reproducibility</h2>
<div
id="wherever-your-sequencing-data-go-their-associated-metadata-go-with-them."
class="section level3 unlisted unnumbered">
<h3 class="unlisted unnumbered">4. Wherever your sequencing data go,
their associated metadata go with them.</h3>
<p>Any and all metadata that can be reported should accompany sequence
data in databases such as NCBI or SRA. Data on Dryad or GitHub should
crosslink to NCBI/SRA.</p>
</div>
<div
id="take-detailed-records-on-all-analysis-decisions-you-make-including-for-preliminary-analyses-and-errors-that-occurred-so-you-remember-what-you-did-and-can-reproduce-your-own-work."
class="section level3 unlisted unnumbered">
<h3 class="unlisted unnumbered">5. Take detailed records on all analysis
decisions you make, including for preliminary analyses and errors that
occurred, so you remember what you did and can reproduce your own
work.</h3>
<p>Use text-annotated code notebooks for bioinformatic analyses (e.g.,
Rmarkdown, Jupyter).</p>
</div>
<div
id="provide-a-reproducible-text-annotated-code-notebook-for-all-final-analyses-containing-computing-environment-information-i.e.-software-versions-so-that-these-methods-could-be-reproduced-by-someone-else."
class="section level3 unlisted unnumbered">
<h3 class="unlisted unnumbered">6. Provide a reproducible text-annotated
code notebook for all final analyses, containing computing environment
information (i.e. software versions), so that these methods could be
reproduced by someone else.</h3>
<p>Provide these notebooks (in Rmarkdown or Jupyter) in a publically
accessible format on services such as GitHub, GitLab, Dryad, Figshare,
or Zenodo.</p>
<p><br></p>
</div>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="unnumbered">References</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Guirao-Rico2021-sw" class="csl-entry">
Guirao-Rico, Sara, and Josefa González. 2021. <span>“Benchmarking the
Performance of Pool-Seq <span>SNP</span> Callers Using Simulated and
Real Sequencing Data.”</span> <em>Mol. Ecol. Resour.</em> 21 (4):
1216–29.
</div>
<div id="ref-Hooten2015-ts" class="csl-entry">
Hooten, M B, and N T Hobbs. 2015. <span>“A Guide to Bayesian Model
Selection for Ecologists.”</span> <em>Ecol. Monogr.</em> 85 (1): 3–28.
</div>
<div id="ref-Johnson2004-sy" class="csl-entry">
Johnson, Jerald B, and Kristian S Omland. 2004. <span>“Model Selection
in Ecology and Evolution.”</span> <em>Trends Ecol. Evol.</em> 19 (2):
101–8.
</div>
<div id="ref-Lotterhos2022-ff" class="csl-entry">
Lotterhos, Katie E, Matthew C Fitzpatrick, and Heath Blackmon. 2022.
<span>“Simulation Tests of Methods in Evolution, Ecology, and
Systematics: Pitfalls, Progress, and Principles.”</span> <em>Annu. Rev.
Ecol. Evol. Syst.</em> 53 (1): 113–36.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
