<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Choosing a Population Genomics Approach</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="tutorial.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MarineOmics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="contributions.html">Contributions</a>
</li>
<li>
  <a href="panels.html">Panel Seminars</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Population Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="choosing-population-genetics.html">Choosing a Population Genomics Approach</a>
    </li>
    <li>
      <a href="WGS_intro.html">Whole Genome Resequencing</a>
    </li>
    <li>
      <a href="RADseq.html">Reduced Representation Sequencing</a>
    </li>
    <li>
      <a href="poolseq.html">Poolseq</a>
    </li>
    <li>
      <a href="RDAtraitPredictionTutorial.html">Redundancy Analysis (RDA) Trait Prediction</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Functional Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="DGE_comparison_v2.html">Mutifactorial RNAseq</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Genome-Phenome
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">coming soon!</li>
  </ul>
</li>
<li>
  <a href="https://github.com/MarineOmics/marineomics.github.io/discussions">Discussion Forum</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'G-53GH9PV49T', 'auto');
  ga('send', 'pageview');

</script>

<div id="header">



<h1 class="title toc-ignore">Choosing a Population Genomics
Approach</h1>

</div>


<div id="population-genomic-sequencing-approaches"
class="section level1">
<h1>Population Genomic Sequencing Approaches</h1>
<p>Deciding on a sequencing approach for a new project can be
challenging. Your choice depends primarily on the questions of interest,
but also constraints set by funding, existing genomic resources, sample
size, genome characteristics and expertise in the research group.</p>
<p>The main categories of population genomic sequencing approaches
include…</p>
<div id="whole-genome-resequencing" class="section level2">
<h2>1) Whole-genome resequencing</h2>
<p>Two approaches for whole-genome resequencing are distinguished by
their coverage depth. <strong>Coverage depth</strong> refers to the
average number of sequencing reads per region of the genome (distinct
from coverage breadth, see <span class="citation">(Sims et al.
2014)</span>). With more sequencing reads at a region (higher coverage),
errors that arise during sequencing have a smaller impact on accurately
identifying variable sites (termed “genotyping”) <span
class="citation">(Fuentes-Pardo and Ruzzante 2017)</span>.</p>
<div id="high-coverage-individual-whole-genome-resequencing-hcwgs"
class="section level3 unlisted unnumbered">
<h3 class="unlisted unnumbered">High coverage individual whole-genome
resequencing (hcWGS)</h3>
<p><strong>Whole-genome resequencing (WGS)</strong> is where whole
genomes of multiple individuals are sequenced and mapped to an assembled
reference genome in order to compare variable sites between individuals
or populations. High-coverage (&gt;20x) sequencing is able to find rare
variants with high confidence. This approach is considered the ‘gold
standard’ for DNA re-sequencing because it includes all variant types
(i.e SNVs, indels, structural variants and CNVs) <span
class="citation">(Sims et al. 2014)</span>.</p>
</div>
<div id="low-coverage-individual-wgs-lcwgs"
class="section level3 unlisted unnumbered">
<h3 class="unlisted unnumbered">Low coverage individual WGS (lcWGS)</h3>
<p><strong>Low coverage whole genome sequencing (lcWGS)</strong> is
where whole genomes of many individuals are sequenced, but depth of
coverage is so low (&lt;5× per site per individual, on average <span
class="citation">(R. Nielsen et al. 2011)</span>) that individual
genotypes cannot be confidently assigned. Instead of assigning specific
(“hard called”) genotypes, lcWGS data requires probabilistic analyses
that account for uncertainty about the true genotype and can incorporate
uncertainty across all sequenced samples <span
class="citation">(Korneliussen, Albrechtsen, and Nielsen 2014)</span>.
This approach is useful in cases where the research questions of
interest are targeted at the population-level (i.e. allele frequencies,
linkage disequilibrium (LD) patterns, etc) <span class="citation">(Lou
et al. 2021)</span>.</p>
</div>
</div>
<div id="reduced-representation-sequencing" class="section level2">
<h2>2) Reduced-representation sequencing</h2>
<p>As the name suggests, this approach aims to sequence a reduced
portion of the genome across multiple individuals at moderate to high
coverage for variant discovery and genotyping with high confidence. The
genome fragments can be selected at random, or through targeted probe
assays. Variations of this type of sequencing approach include…</p>
<div id="radseqgenotype-by-sequencing"
class="section level3 unlisted unnumbered">
<h3 class="unlisted unnumbered">RADseq/Genotype-by-sequencing</h3>
<p>A class of methods involving sequencing a subset of genomic regions
randomly distributed throughout the genome, often using restriction
enzymes (i.e. <strong>restriction site-associated DNA
(RADseq)</strong>).</p>
</div>
<div id="targeted-capture-methods"
class="section level3 unlisted unnumbered">
<h3 class="unlisted unnumbered">Targeted capture methods</h3>
<p>Sequence capture methods use a set of probes designed with a-priori
knowledge to focus sequencing effort on a set of hundreds to tens of
thousands of specific loci. Several capture approaches for non-model
species can utilise other sequencing strategies to design probes
corresponding to genomic regions (i.e. <strong>Whole-exome sequencing
(WES)</strong>, see <span class="citation">(Jones and Good 2016)</span>
for a review, <strong>Expressed Exome Capture Sequencing
(eecSeq)</strong><span class="citation">(Puritz and Lotterhos
2018)</span>).</p>
</div>
</div>
<div id="pooled-individuals-poolseq" class="section level2">
<h2>3) Pooled individuals (Poolseq)</h2>
<p>Sequencing pools of individuals to provide information on genome-wide
population allele frequencies (see review in <span
class="citation">(Futschik and Schlötterer 2010)</span>).</p>
<div class="figure">
<img src="images/mec16077-fig-0001-m.jpg" alt="" />
<p class="caption"><strong>Figure 1</strong> from <span
class="citation">(Lou et al. 2021)</span> comparing the distribution of
sequencing reads mapped to a reference genome.</p>
</div>
<p>It should be noted that you don’t need to choose only one approach
for all samples! If you are starting a research program on a species
without any genomic resources, you may want to start with denovo RADseq
to understand basic population structure and genetic diversity, then
once a reference genome is assembled move to high/low coverage WGS.In
the <a href="https://marineomics.github.io/panels.html">MarineOmics
seminar</a>, <a href="https://matzlab.weebly.com/">Misha Matz</a>
recommended choosing one individual to use for genome assembly, doing
high coverage WGS on a few other individuals that represent most of the
variation in your taxa, and then WGS many other individuals at low
coverage. The high coverage individuals can be useful for imputing
missing genotypes in the low coverage samples.</p>
<p>The table below briefly summarizes the pros and cons of five
sequencing approaches and their appropriateness for answering specific
questions. If you are interested in answering multiple questions
(i.e. neutral population structure and adaptive variation) then you
would often want a method that can do both (i.e. WGS).</p>
<table class="table table-striped" style="font-size: 13px; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Goals
</th>
<th style="text-align:left;">
hcWGS
</th>
<th style="text-align:left;">
lcWGS
</th>
<th style="text-align:left;">
RADseq
</th>
<th style="text-align:left;">
Poolseq
</th>
<th style="text-align:left;">
Target.Capture
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;font-weight: bold;border-right:1px solid;">
pros
</td>
<td style="text-align:left;">
provides high quality, high density genotypes; can be used to improve
development of a reference genome
</td>
<td style="text-align:left;">
provides high density genotypes at a reduced cost
</td>
<td style="text-align:left;">
generally cheaper per sample, density of genotypes can be tuned to fit
question, does not require a reference
</td>
<td style="text-align:left;">
many individuals can be mixed for a low library prep cost, only option
for some larval studies
</td>
<td style="text-align:left;">
allows consistent capture of loci between batches, good for sequencing
targeted regions across many samples
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;border-right:1px solid;">
cons
</td>
<td style="text-align:left;">
most expensive per sample, requires a reference genome assembly, large
computational resources
</td>
<td style="text-align:left;">
requires a reference, can produce false (+) heterozygotes, sensitive to
batch effects, may be inappropriate to use individual SNP calls for some
analyses
</td>
<td style="text-align:left;">
only 1-5% of genome covered which limits studies on adaptive variation,
de novo assembly can result in paralogs without quality control, can be
hard to capture the same loci between batches
</td>
<td style="text-align:left;">
need higher sequencing depth so it can be more expensive than RAD (but
less than other options), no info on individual genotypes (duh),
requires a reference, requires some replicates to account for batch
effects
</td>
<td style="text-align:left;">
expensive to start a new project and requires reference of some kind to
design baits (unless using method like eecSeq), only gives info on
targeted region so can result in ascertainment bias for some analyses
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;border-right:1px solid;">
population structure, genetic diversity
</td>
<td style="text-align:left;">
great, but cheaper methods work almost as well
</td>
<td style="text-align:left;">
good, but limited to popgen analyses based on genotype likelihoods
</td>
<td style="text-align:left;">
great, esp. for many samples
</td>
<td style="text-align:left;">
good for methods that only use allele frequencies, won’t work for
individual based methods (eg Admixture)
</td>
<td style="text-align:left;">
ascertainment bias is likely to skew results, esp. for measures of
genetic diversity
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;border-right:1px solid;">
demography (mig. rates, pop. size through time)
</td>
<td style="text-align:left;">
great if incorporating haplotype-based methods
</td>
<td style="text-align:left;">
good, esp. with some high coverage samples to help with imputation
</td>
<td style="text-align:left;">
good for methods based on site frequency spectrum (eg moments), but not
ideal for methods using extended haplotype or phasing info
</td>
<td style="text-align:left;">
active area of method development, but still not common except in cases
of multiple temporal samples <span class="citation">(Nielsen et al.
2020)</span>
</td>
<td style="text-align:left;">
likely not appropriate due to ascertainment bias
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;border-right:1px solid;">
signatures of selection, GWAS
</td>
<td style="text-align:left;">
great
</td>
<td style="text-align:left;">
good
</td>
<td style="text-align:left;">
hotly debated, but unless your genome is way too big, WGS is the way to
go
</td>
<td style="text-align:left;">
good if covering majority of the genome
</td>
<td style="text-align:left;">
only works for targeted regions
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;border-right:1px solid;">
phylogenetic inference
</td>
<td style="text-align:left;">
computationally challenging, but good at all divergence levels
</td>
<td style="text-align:left;">
good for tree shapes but not branch lengths, good for extracting
organelle sequences and cost-effective primers for species and hybrid ID
studies
</td>
<td style="text-align:left;">
good at shallow to medium divergence levels, but need to play with
filtering parameters
</td>
<td style="text-align:left;">
would work in certain situations
</td>
<td style="text-align:left;">
an improvement over RAD as there will be less missing data among samples
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;border-right:1px solid;">
genetic crosses/mapping panels
</td>
<td style="text-align:left;">
expensive
</td>
<td style="text-align:left;">
great
</td>
<td style="text-align:left;">
great as it allows for many invididuals
</td>
<td style="text-align:left;">
maybe
</td>
<td style="text-align:left;">
poor
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;border-right:1px solid;">
molecular evolution (accurate low-freq alleles required)
</td>
<td style="text-align:left;">
best
</td>
<td style="text-align:left;">
poor
</td>
<td style="text-align:left;">
poor
</td>
<td style="text-align:left;">
poor
</td>
<td style="text-align:left;">
good if only interested in targeted regions
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="recommended-readings" class="section level1">
<h1>Recommended readings</h1>
<p>Below we list a number of great review articles and empirical
comparison studies that can help when making the decision. This topic
was also discussed during the <a
href="https://marineomics.github.io/panels.html">MarineOmics
seminars</a>.</p>
<p>Review articles:</p>
<ul>
<li><span class="citation">(Matz 2018)</span>: great intro to ecological
genomics methods in species with few to no prior genomic resources,
accessible to those just getting started.<br />
</li>
<li><span class="citation">(Fuentes-Pardo and Ruzzante 2017)</span>: a
VERY thorough review of the different approaches to generating
population genomics data, with non-model species in mind. Some of the
content about relative cost and available software is a little dated,
but otherwise a very valuable reference.<br />
</li>
<li><span class="citation">(Lou et al. 2021)</span>: great recent primer
on low coverage WGS and how it compares to RADseq and Poolseq.</li>
</ul>
<p>Empirical comparisons:</p>
<ul>
<li><span class="citation">(Benjelloun et al. 2019)</span>: an empirical
comparison of high coverage WGS, low coverage WGS, random variants (eg.
from RADseq), exome capture, and commercial SNP chip for mammals (genome
size=2.6Gb). For neutral genomic diversity, found 5k-10k random variants
were enough to accurately estimate, but commercial panels/exome capture
had ascertainment bias. To detect selection and accurate estimates of
LD, at least 1M variants were required. For the studied species, 5x
coverage was just as accurate as 12x coverage. Low coverage WGS (&lt;
5x) should use genotype likelihood methods, however, the rate of false
positives for heterozygotes is higher than classic genotyping methods at
higher coverage.</li>
<li><span class="citation">(Dorant et al. 2019)</span>: a case study
comparing Poolseq vs. reduced representation seq (GBS) vs Rapture
(targeted capture of RAD loci) for inferring population structure with
weak genetic differentiation. All three detected the same weak pop
structure pattern, but Poolseq gave Fst values 3-5x higher than
Rapture/GBS.</li>
</ul>
</div>
<div id="marineomics-guidelines" class="section level1">
<h1>MarineOmics Guidelines</h1>
<p>Presently, our site covers four approaches:</p>
<ul>
<li><a href="RADseq.html">RADseq</a><br />
</li>
<li><a href="WGS_intro.html">hcWGS</a></li>
<li><a
href="https://github.com/therkildsen-lab/genomic-data-analysis/blob/master/lcwgs_data_analysis.md">lcWGS,
maintained by MarineOmics member Nicolas Lou</a></li>
<li><a href="poolseq.html">Poolseq</a></li>
</ul>
<p>For each approach, we provide guiding principles and tutorials for
bioinformatic processing from raw data to genotypes. While we currently
don’t discuss library preparation methods in depth, we do include advice
on how to organize your sequencing runs in order to improve quality
control downstream.</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Benjelloun2019-dq" class="csl-entry">
Benjelloun, Badr, Frédéric Boyer, Ian Streeter, Wahid Zamani, Stefan
Engelen, Adriana Alberti, Florian J Alberto, et al. 2019. <span>“An
Evaluation of Sequencing Coverage and Genotyping Strategies to Assess
Neutral and Adaptive Diversity.”</span> <em>Mol. Ecol. Resour.</em> 19
(6): 1497–1515. <a
href="https://doi.org/10.1111/1755-0998.13070">https://doi.org/10.1111/1755-0998.13070</a>.
</div>
<div id="ref-Dorant2019-fb" class="csl-entry">
Dorant, Yann, Laura Benestan, Quentin Rougemont, Eric Normandeau, Brian
Boyle, Rémy Rochette, and Louis Bernatchez. 2019. <span>“Comparing
Pool-Seq, Rapture, and <span>GBS</span> Genotyping for Inferring Weak
Population Structure: The American Lobster (Homarus Americanus) as a
Case Study.”</span> <em>Ecol. Evol.</em> 9 (11): 6606–23. <a
href="https://doi.org/10.1002/ece3.5240">https://doi.org/10.1002/ece3.5240</a>.
</div>
<div id="ref-Fuentes-Pardo2017-nu" class="csl-entry">
Fuentes-Pardo, Angela P, and Daniel E Ruzzante. 2017.
<span>“Whole-Genome Sequencing Approaches for Conservation Biology:
Advantages, Limitations and Practical Recommendations.”</span> <em>Mol.
Ecol.</em> 26 (20): 5369–5406. <a
href="https://doi.org/10.1111/mec.14264">https://doi.org/10.1111/mec.14264</a>.
</div>
<div id="ref-futschik2010next" class="csl-entry">
Futschik, Andreas, and Christian Schlötterer. 2010. <span>“The Next
Generation of Molecular Markers from Massively Parallel Sequencing of
Pooled DNA Samples.”</span> <em>Genetics</em> 186 (1): 207–18.
</div>
<div id="ref-jones2016targeted" class="csl-entry">
Jones, Matthew R, and Jeffrey M Good. 2016. <span>“Targeted Capture in
Evolutionary and Ecological Genomics.”</span> <em>Molecular Ecology</em>
25 (1): 185–202.
</div>
<div id="ref-korneliussen2014angsd" class="csl-entry">
Korneliussen, Thorfinn Sand, Anders Albrechtsen, and Rasmus Nielsen.
2014. <span>“ANGSD: Analysis of Next Generation Sequencing Data.”</span>
<em>BMC Bioinformatics</em> 15 (1): 1–13.
</div>
<div id="ref-Lou2021-me" class="csl-entry">
Lou, Runyang Nicolas, Arne Jacobs, Aryn P Wilder, and Nina Overgaard
Therkildsen. 2021. <span>“A Beginner’s Guide to Low-Coverage Whole
Genome Sequencing for Population Genomics.”</span> <em>Mol. Ecol.</em>,
July. <a
href="https://doi.org/10.1111/mec.16077">https://doi.org/10.1111/mec.16077</a>.
</div>
<div id="ref-Matz2018-vl" class="csl-entry">
Matz, Mikhail V. 2018. <span>“Fantastic Beasts and How to Sequence Them:
Ecological Genomics for Obscure Model Organisms.”</span> <em>Trends
Genet.</em> 34 (2): 121–32. <a
href="https://doi.org/10.1016/j.tig.2017.11.002">https://doi.org/10.1016/j.tig.2017.11.002</a>.
</div>
<div id="ref-nielsen2011genotype" class="csl-entry">
Nielsen, Rasmus, Joshua S Paul, Anders Albrechtsen, and Yun S Song.
2011. <span>“Genotype and SNP Calling from Next-Generation Sequencing
Data.”</span> <em>Nature Reviews Genetics</em> 12 (6): 443–51.
</div>
<div id="ref-nielsen2020bc" class="csl-entry">
Nielsen, Beger, Henriques, and Heyden. 2020. <span>“A Comparison of
Genetic and Genomic Approaches to Represent Evolutionary Potential in
Conservation Planning.”</span> <em>Biological Conservation</em> 251 (1):
108770.
</div>
<div id="ref-puritz2018expressed" class="csl-entry">
Puritz, Jonathan B, and Katie E Lotterhos. 2018. <span>“Expressed Exome
Capture Sequencing: A Method for Cost-Effective Exome Sequencing for All
Organisms.”</span> <em>Molecular Ecology Resources</em> 18 (6): 1209–22.
</div>
<div id="ref-sims2014sequencing" class="csl-entry">
Sims, David, Ian Sudbery, Nicholas E Ilott, Andreas Heger, and Chris P
Ponting. 2014. <span>“Sequencing Depth and Coverage: Key Considerations
in Genomic Analyses.”</span> <em>Nature Reviews Genetics</em> 15 (2):
121–32.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
