---
bibliography: RCN_site-Nov05.bib
title: Choosing a Population Genomics Approach 
---

```{r setup, include=FALSE}
library(knitr)
knitr::opts_chunk$set(echo = TRUE)
library(knitcitations)
library(kableExtra)
opts_chunk$set(fig.width = 10,
               fig.height = 5,
               cache = FALSE)
cite_options(citation_format = "pandoc", max.names = 3, style = "html", 
             hyperlink = "to.doc")
```

# Population genomic sequencing approaches
Deciding on a sequencing approach for a new project can be challenging, and depends primarily on the questions you are interested in, but also constraints set by funding, existing genomic resources, sample size, genome characteristics, and expertise in the research group. The main categories of population genomic sequencing approaches are:   

### 1) High coverage individual whole genome resequencing (hcWGS)  {.unlisted .unnumbered}
**Whole-genome resequencing (WGS)** is where whole genomes of multiple individuals are sequenced and mapped to an assembled reference genome in order to compare variable sites between individuals or populations. **Coverage depth** refers to the average number of sequencing reads per region of the genome. With more sequencing reads at a region (higher coverage depth), errors that arise during sequencing have a smaller impact on accurately identifying variable sites (termed "genotyping") [@Fuentes-Pardo2017-nu].    

### 2) Low coverage individual WGS (lcWGS)  {.unlisted .unnumbered}
Low coverage whole genome sequencing is where whole genomes of many individuals are sequenced, but depth of coverage so low that individual genotypes cannot be confidently assigned (cite). Instead of assigning specific, "hard called genotypes, lcWGS data requires probabilistic analyses that account for uncertainty about the true genotype and can incororate uncertanty accross all sequenced samples. Typically, sequencing depth of > 3 reads is considered "low coverage".  

### 3) Reduced-representation sequencing {.unlisted .unnumbered}
As the name suggests, this approach aims to sequence a reduced portion of the genome across multiple individuals at moderate to high coverage for variant discovery and genotyping with high confidence. The genome fragments can be selected at random, or through targeted probe assays.
#### RADseq/Genotype-by-sequencing  {.unlisted .unnumbered}
This very popular class of methods involve sequencing a subset of genomic regions randomly distributed throughout the genome, often using restriction enzymes (i.e., RADseq: restriction site-associated DNA). 
#### Targeted capture methods (e.g., exome capture)  {.unlisted .unnumbered}
 
### 4) Pooled individuals (Poolseq)  {.unlisted .unnumbered} 

![Figure 1 from [@Lou2021-me] comparing the distribution of sequencing reads mapped to a reference genome.](images/mec16077-fig-0001-m.jpg)

It should be noted that you don't need to choose only one approach for all samples! If you are starting a research program on a species without any genomic resources, you may want to start with de novo RADseq to understand basic population structure and genetic diversity, then once a reference genome is assembled move to high/low coverage WGS. In the [MarineOmics seminar](link), Misha Matz recommended choosing one individual to use for genome assembly, doing high coverage WGS on a few other individuals that represent most of the variation in your taxa, and then WGS many other individuals at low coverage. The high coverage individuals can be useful for imputing missing genotypes in the low coverage samples.  

This table briefly summarizes the pros and cons of five sequencing approaches and their appropriateness for answering specific questions. If you are interested in answering multiple questions (eg, neutral population structure and adaptive variation) then you would often want a method that can do both (eg, WGS).  


```{r,echo=FALSE}
text_tbl <- data.frame(
  Goals = c("Pros","Cons","Population structure, genetic diversity","Demography (mig. rates, pop. size through time)",
            "Signatures of selection, GWAS","Phylogenetic inference","Genetic crosses/mapping panels",
            "molecular evolution (accurate low-freq alleles required)"),
  hcWGS = c("provides high quality, high density genotypes; can be used to improve development of a reference genome","Most expensive per sample, requires a reference genome assembly, large computational resources","Great, but cheaper methods work almost as well","Great if incorporating haplotype-based methods ","Great","Computationally challenging, but good at all divergence levels ","Expensive","Best"),
  lcWGS = c("Provides high density genotypes at a reduced cost","Requires a reference, can produce false (+) heterozygotes, sensitive to batch effects, may be inappropriate to use individual SNP calls for some analyses","Good, but limited to popgen analyses based on genotype likelihoods ","Good, esp. with some high coverage samples to help with imputation","Good","Good for tree shapes but not branch lengths, good for extracting organelle sequences and cost-effective primers for species and hybrid ID studies ","Great","Poor"),
 RADseq = c("Generally cheaper per sample, density of genotypes can be tuned to fit question, does not require a reference","Only 1-5% of genome covered which limits studies on adaptive variation, de novo assembly can result in paralogs without quality control, can be hard to capture the same loci between batches","Great, esp. for many samples","Good for methods based on site frequency spectrum (eg moments), but not ideal for methods using extended haplotype or phasing info","Hotly debated, but unless your genome is way too big, WGS is the way to go","good at shallow to medium divergence levels, but need to play with filtering parameters ", "great as it allows for many invididuals","poor" ),
 Poolseq = c("many individuals can be mixed for a low library prep cost, only option for some larval studies","need higher sequencing depth so it can be more expensive than RAD (but less than other options), no info on individual genotypes (duh), requires a reference, requires some replicates to account for batch effects","good for methods that only use allele frequencies, won't work for individual based methods (eg Admixture)","active area of method development, but still not common except in cases of multiple temporal samples (cite) ","good if covering majority of the genome","would work in certain situations","maybe","poor"),
 Target.Capture = c("allows consistent capture of loci between batches, good for sequencing targeted regions across many samples","expensive to start a new project and requires reference of some kind to design baits (unless using method like eecSeq), only gives info on targeted region so can result in ascertainment bias for some analyses","ascertainment bias is likely to skew results, esp. for measures of genetic diversity","likely not appropriate due to ascertainment bias","only works for targeted regions ","an improvement over RAD as there will be less missing data among samples","poor","good if only interested in targeted regions")
  )

kbl(text_tbl) %>%
 # kable_paper(full_width = F) %>%
  column_spec(1, bold = T, border_right = T) %>%
  kable_styling(bootstrap_options = "striped", font_size = 13)

```

# Recommended readings  
Below we list a number of great review articles and empirical comparison studies that can help when making the decision. This topic was also discussed during the MarineOmics seminars (link).  

Review articles:

* [@Matz2018-vl]: great intro to ecological genomics methods in species with few to no prior genomic resources, accessible to those just getting started.  
* [@Fuentes-Pardo2017-nu]: a VERY thorough review of the different approaches to generating population genomics data, with non-model species in mind. Some of the content about relative cost and available software is a little dated, but otherwise a very valuable reference.   
* [@Lou2021-me]: Great recent primer on low coverage WGS and how it compares to RADseq and Poolseq.

Empirical comparisons:  

* [@Benjelloun2019-dq]: empirical comparison of high coverage WGS, low coverage WGS, random variants (eg. from RADseq), exome capture, and commercial SNP chip for mammals (genome size=2.6Gb). For neutral genomic diversity, found 5k-10k random variants were enough to accurately estimate, but comercial panales/exome capture had acertainment bias. To detect selection and accurate estimates of LD, at least 1M variants were required. For the studied species, 5x coverage was just as accurate as 12x coverage. Low coverage WGS (< 5x) should use genotype likelihood methods, however the rate of false positives for heterozygotes is higher than classic genotyping methods at higher coverage.  
* [@Dorant2019-fb]: Case study comparing Poolseq vs. reduced representation seq (GBS) vs Rapture (targeted capture of RAD loci) for inferring population structure with weak genetic differentiation. All three detected same weak pop structure pattern, but Poolseq gave Fst values 3-5x higher than Rapture/GBS.     

# MarineOmics Guidelines
Presently, our site covers four approaches: 

* [RADseq](RADseq.html)  
* [hcWGS](WGS_intro.html)
* [lcWGS, maintained by MarineOmics member Nicolas Lou](https://github.com/therkildsen-lab/genomic-data-analysis/blob/master/lcwgs_data_analysis.md)
* [Poolseq](poolseq.html)


For each approach, we provide guiding principles and tutorials for bioinformatic processing from raw data to genotypes. While we currently don't discuss library preparation methods in depth, we do include advice on how to organize your sequencing runs in order to improve quality control downstream.  

# References
