---
title: 'DNA Methylation Analysis'
subtitle: '*Steven Roberts*, and *Sam White* '
bibliography: DNA-Methylation.bib
---

# Introduction
The purpose of this tutorial is to provide an example of taking Bisulfite treated DNA sequence data through identification of methylation at the loci and gene level. The principle behind creating bisulfite treated DNA libraries is that when cytosines are not methylated they will be converted to uracil, which ultimately result in thymine. When aligning to a reference genome the percent methylation at a given CpG loci is determined by examine the ratio to thymines and cytosines. For example if there is a given CpG loci with 30% of the alignment with the cytosine on the reference being and thymine, one would consider this cytosine loci (CpG) to be 70% methylated.

# Sequence Quality
An example of what raw data might look like from a marine invertebrate is shown. _Note that DNA methylation patters are diverse across taxa with most marine invertebrates demonstrating a mosiac pattern and global methylation at CpG loci is approximately 15%._  


Following trimming one would expect to see something similar to this. This particular data set was trimmed with the following parameters: 

![bp-chart](images/meth-bp-chart.png)

