<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Functional Enrichment Tests: Intro and methods comparison</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="tutorial.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MarineOmics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="ADMIN_04_best_principles.html">Best Principles</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Contributions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ADMIN_01_submissions_instructions.html">Guide for Building a Page</a>
    </li>
    <li>
      <a href="ADMIN_02_contributions.html">Past and Current Contributors</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Population Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="POP_01_choosing_population_genetics.html">Choosing a Population Genomics Approach</a>
    </li>
    <li>
      <a href="POP_04_WGS_intro.html">Whole Genome Resequencing</a>
    </li>
    <li>
      <a href="POP_02_RADseq.html">Reduced Representation Sequencing</a>
    </li>
    <li>
      <a href="POP_03_poolseq.html">Poolseq</a>
    </li>
    <li>
      <a href="POP_05_RDAtraitPredictionTutorial.html">Redundancy Analysis (RDA) Trait Prediction</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Functional Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="FUN_01_DGE_comparison_v2.html">Mutifactorial RNAseq</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Genome-Phenome
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">coming soon!</li>
  </ul>
</li>
<li>
  <a href="ADMIN_03_panels.html">Panel Seminars</a>
</li>
<li>
  <a href="https://github.com/MarineOmics/marineomics.github.io/discussions">Discussion Forum</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'G-53GH9PV49T', 'auto');
  ga('send', 'pageview');

</script>

<div id="header">



<h1 class="title toc-ignore">Functional Enrichment Tests: Intro and
methods comparison</h1>
<h3 class="subtitle"><em>Thea Gessler</em>, <em>Samuel N. Bogan</em>,
and <em>Steven Roberts</em></h3>

</div>


<p>Functional gene enrichment analysis is a powerful tool that can aid
in the interpretation of biological data by identifying biological
processes and pathways that are significantly overrepresented in a gene
list of interest. However, to effectively use this tool, it is essential
to have a proper annotation of genes.</p>
<p>Proper annotation of genes is crucial because it ensures the genes
are correctly identified and annotated with their appropriate biological
functions. Incorrect annotation of genes can lead to inaccurate
interpretations of experimental results and hinder progress in
understanding the biological systems of non-model species.</p>
<p>Therefore, the quality of gene annotation is critical for accurate
functional gene enrichment analysis. This involves careful curation of
gene annotations based on experimental evidence from various sources,
including transcriptomic and proteomic data, literature mining, and
functional assays. This curation process can be particularly challenging
for non-model species with limited genomic and transcriptomic
resources.</p>
<p><strong>In some instances high quality annotation might not be
possible but it is critical that before you undergo enrichment analysis
you have a full understanding of how your data set was annotated, and
any associated limitations</strong></p>
<div id="navigating-annotation" class="section level1">
<h1>Navigating Annotation</h1>
<p>The following is a stepwise example or annotation of a gene set using
UniProt::Swiss-Prot (reviewed) such that Gene Ontology terms can be
associated with each gene.</p>
<p>In this following chunk where the fasta file is downloaded the <a
href="https://www.uniprot.org/help/release-statistics">release</a> is
noted and the file name is modified accordingly.</p>
<pre class="bash"><code>cd DRAFT_Funct_Enrich/annot

curl -O https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz

mv uniprot_sprot.fasta.gz uniprot_sprot_r2023_02.fasta.gz
gunzip -k uniprot_sprot_r2023_02.fasta.gz</code></pre>
<p>A protein blast database is then made.</p>
<pre class="bash"><code>/home/shared/ncbi-blast-2.11.0+/bin/makeblastdb \
-in DRAFT_Funct_Enrich/annot/uniprot_sprot_r2023_02.fasta \
-dbtype prot \
-out DRAFT_Funct_Enrich/annot/uniprot_sprot_r2023_02</code></pre>
<p>In a majority of cases you will want to annotate a gene set to get
gene ontology information. If you are creating your own genome or
transcriptome it should be rather straightforward to know what file to
annotate. If using a widely studied system where there are publically
available resources, it is advisable to use those as this is the best
way to facilitate integration of data sets. In this case study we will
be considering the Eastern oyster, (<em>Crassostrea virginica</em>) for
which there is data at <a
href="https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_002022765.2/">NCBI</a>
and <a
href="http://metazoa.ensembl.org/Crassostrea_virginica_gca002022765v4/Info/Index">Ensembl
Metazoa</a>. At NCBI there is both a GenBank and RefSeq assembly
available.</p>
<p>In order to know which of the numerous fasta files should annotated
with gene ontology information one should think downstream (or look to
files already generated) to the identifiers in genesets that would be
subject to functional enrichment tests.</p>
<p>The resulting fpkm count matrix for our case study is from an
experiment where male and female oysters where exposed to low pH (and
control) conditions. The count matrix is accessible <a
href="https://github.com/epigeneticstoocean/2018_L18-adult-methylation/blob/main/analyses/gene_fpkm.csv">here
(csv)</a>. Hisat2/Stringtie was used to generate the count matrix with
<code>GCF_002022765.2_C_virginica-3.0_genomic.gff</code> formatting thus
responsible for gene naming. Specifically the naming format is as
follows
<code>gene-LOC111099033,gene-LOC111099034,gene-LOC111099035</code>.</p>
<p>The following fasta was selected for annotation:
<code>GCF_002022765.2_C_virginica-3.0_translated_cds.faa.gz</code></p>
<pre class="bash"><code>cd DRAFT_Funct_Enrich/annot

curl -O https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/002/022/765/GCF_002022765.2_C_virginica-3.0/GCF_002022765.2_C_virginica-3.0_translated_cds.faa.gz

gunzip -k GCF_002022765.2_C_virginica-3.0_translated_cds.faa.gz</code></pre>
<pre class="bash"><code>head -2 DRAFT_Funct_Enrich/annot/GCF_002022765.2_C_virginica-3.0_translated_cds.faa

echo &quot;number of sequences&quot;
grep -c  &quot;&gt;&quot; DRAFT_Funct_Enrich/annot/GCF_002022765.2_C_virginica-3.0_translated_cds.faa
</code></pre>
<pre><code>&gt;lcl|NC_035780.1_prot_XP_022327646.1_1 [gene=LOC111126949] [db_xref=GeneID:111126949] [protein=UNC5C-like protein] [protein_id=XP_022327646.1] [location=join(30535..31557,31736..31887,31977..32565,32959..33204)] [gbkey=CDS]
MTEVCYIWASSSTTVVICGIFFIVWRCFISIKKRASPLHGSSQQVCQTCQIEGHDFGEFQLSCRRQNTNVGYDLQGRRSD</code></pre>
<p>This protein fasta is used as query for blast of uniprot_sprot
database.</p>
<pre class="bash"><code>/home/shared/ncbi-blast-2.11.0+/bin/blastp \
-query DRAFT_Funct_Enrich/annot/GCF_002022765.2_C_virginica-3.0_translated_cds.faa \
-db DRAFT_Funct_Enrich/annot/uniprot_sprot_r2023_02 \
-out DRAFT_Funct_Enrich/annot/Cvir_transcds-uniprot_blastp.tab \
-evalue 1E-20 \
-num_threads 40 \
-max_target_seqs 1 \
-outfmt 6</code></pre>
<p>Here is what the output file looks like, and at this point we want to
get the UniProt Accession number for each gene</p>
<pre class="bash"><code>head -2 DRAFT_Funct_Enrich/annot/Cvir_transcds-uniprot_blastp.tab</code></pre>
<pre><code>## lcl|NC_035780.1_prot_XP_022327646.1_1    sp|Q8IV45|UN5CL_HUMAN   33.403  479 270 8   151 626 78  510 4.80e-77    258
## lcl|NC_035780.1_prot_XP_022305632.1_4    sp|P39021|MEOX2_XENLA   55.118  127 57  0   104 230 115 241 5.84e-38    138</code></pre>
<pre class="r"><code>blast &lt;- read.csv(&quot;DRAFT_Funct_Enrich/annot/Cvir_transcds-uniprot_blastp.tab&quot;, sep = &#39;\t&#39;, header = FALSE)</code></pre>
<p>Convert fasta to tab</p>
<pre class="bash"><code>perl -e &#39;$count=0; $len=0; while(&lt;&gt;) {s/\r?\n//; s/\t/ /g; if (s/^&gt;//) { if ($. != 1) {print &quot;\n&quot;} s/ |$/\t/; $count++; $_ .= &quot;\t&quot;;} else {s/ //g; $len += length($_)} print $_;} print &quot;\n&quot;; warn &quot;\nConverted $count FASTA records in $. lines to tabular format\nTotal sequence length: $len\n\n&quot;;&#39; \
DRAFT_Funct_Enrich/annot/GCF_002022765.2_C_virginica-3.0_translated_cds.faa &gt; DRAFT_Funct_Enrich/annot/GCF_002022765.2_C_virginica-3.0_translated_cds.tab</code></pre>
<pre><code>## 
## Converted 60213 FASTA records in 580876 lines to tabular format
## Total sequence length: 39274970</code></pre>
<pre class="bash"><code>head -1 DRAFT_Funct_Enrich/annot/GCF_002022765.2_C_virginica-3.0_translated_cds.tab</code></pre>
<pre><code>## lcl|NC_035780.1_prot_XP_022327646.1_1    [gene=LOC111126949] [db_xref=GeneID:111126949] [protein=UNC5C-like protein] [protein_id=XP_022327646.1] [location=join(30535..31557,31736..31887,31977..32565,32959..33204)] [gbkey=CDS]    MTEVCYIWASSSTTVVICGIFFIVWRCFISIKKRASPLHGSSQQVCQTCQIEGHDFGEFQLSCRRQNTNVGYDLQGRRSDVTAIIQGQSVLRPSPVPVEVLPVEGNRVSRTSESVRTHSSTHSAGSSSHVATSPVMSSCSVGKETSLNPYSYCRDIYSISERSTMKTVGGKLVVSVAHYVTSRGDWLMLDDMGISLRIPPNAVPVGEEKLICLVLNWDLGDNPPMTGTDSLVSPVVFVGPHGLKLNKPCELHYRHCAYDPRQIVVLRSETDLHGDKHWTEMCGQENQDGVCRLSSDECILQIDTFTLYTCIQRPLRNLKMAKWLQIAAFSNPLEKRIDHHEVRVYFLNKTPCALQWAIQNEAQFQHVLICPEKAFLFHGQVDDILDIRVTLKYLSADWESIDNDVYERVPYLHIWHGQCPHIVMCFKKKPKHSVRELNFHLNIYQDTLENEGEKIVVHATENPRCVYDNSCIINIHTSDSPEKKTLPNVEVKKTEDNVNVSIKQSHSPSAALLEEMRTESTQYIPYDIRKHLKVKLDPPNPLGNDWRRLAEVLDLSQFIEYLKSLHSSPTETLLSVIEHRNISLEELATMLNEIQRYDAEKIVSDYLHQHQNSSPVCIRGGLQHDERPFAHMSATNNLSNVEPDNVSQEENRAQYLPLGNSGHDNNAYM</code></pre>
<pre class="r"><code>cdsftab &lt;- read.csv(&quot;DRAFT_Funct_Enrich/annot/GCF_002022765.2_C_virginica-3.0_translated_cds.tab&quot;, sep = &#39;\t&#39;, header = FALSE, row.names=NULL)</code></pre>
<p>Now we can take the two data frames: A) blast output of taking
protein fasta and comparing to uniprot_swiss-prot and B) a tabular
version of same fasta file that has ID numbers of importance. Note this
importance was determined based on what we want to use down stream.</p>
<pre class="r"><code>g.spid &lt;- left_join(blast, cdsftab, by = &quot;V1&quot;) %&gt;%
  mutate(gene = str_extract(V2.y, &quot;(?&lt;=\\[gene=)\\w+&quot;)) %&gt;%
  select(gene, V11, V2.x) %&gt;%
  mutate(SPID = str_extract(V2.x, &quot;(?&lt;=\\|)[^\\|]*(?=\\|)&quot;)) %&gt;%
  distinct(gene, SPID, .keep_all = TRUE)</code></pre>
<p>Let’s break it down step by step:</p>
<ol style="list-style-type: decimal">
<li><p><strong><code>g.spid &lt;- left_join(blast, cdsftab, by = "V1")</code></strong>
- This line is using the <strong><code>left_join()</code></strong>
function from <strong><code>dplyr</code></strong> to merge the
<strong><code>blast</code></strong> and
<strong><code>cdsftab</code></strong> datasets by the column “V1”. A
left join retains all the rows in the
<strong><code>blast</code></strong> data frame and appends the matching
rows in the <strong><code>cdsftab</code></strong> data frame. If there
is no match, the result is <strong><code>NA</code></strong>. The result
of this operation is assigned to the
<strong><code>g.spid</code></strong> object.</p></li>
<li><p><strong><code>mutate(gene = str_extract(V2.y, "(?&lt;=\\[gene=)\\w+"))</code></strong>
- This line is using the <strong><code>mutate()</code></strong> function
from <strong><code>dplyr</code></strong> to add a new column called
“gene” to the data frame. The new column is created by extracting
substrings from the “V2.y” column based on the given regular expression
pattern <strong><code>"(?&lt;=\\[gene=)\\w+"</code></strong>. This
regular expression matches and extracts any word (sequence of word
characters, i.e., alphanumeric and underscore) that comes after
“[gene=”.</p></li>
<li><p><strong><code>select(gene, V11, V2.x)</code></strong> - This line
is using the <strong><code>select()</code></strong> function from
<strong><code>dplyr</code></strong> to keep only the specified columns
(“gene”, “V11”, and “V2.x”) in the data frame.</p></li>
<li><p><strong><code>mutate(SPID = str_extract(V2.x, "(?&lt;=\\|)[^\\|]*(?=\\|)"))</code></strong>
- Again, the <strong><code>mutate()</code></strong> function is used to
add another new column named “SPID”. This column is created by
extracting substrings from the “V2.x” column. The regular expression
<strong><code>"(?&lt;=\\|)[^\\|]*(?=\\|)"</code></strong> is designed to
extract any character(s) that is/are surrounded by “|” (pipe symbol).
This is a common format for delimited strings.</p></li>
<li><p><strong><code>distinct(gene, SPID, .keep_all = TRUE)</code></strong>
- This line is using the <strong><code>distinct()</code></strong>
function from <strong><code>dplyr</code></strong> to remove duplicate
rows based on the “gene” and “SPID” columns. The
<strong><code>.keep_all = TRUE</code></strong> argument means that all
other columns are also kept in the result, not just the “gene” and
“SPID” columns.</p></li>
</ol>
<p>The resulting <strong><code>g.spid</code></strong> data frame should
have unique rows with respect to the “gene” and “SPID” columns, and it
should contain these two new columns, “gene” and “SPID”, extracted from
the original data based on specific string patterns.</p>
<p>Now lets just write out SPIDs.</p>
<pre class="r"><code>left_join(blast, cdsftab, by = &quot;V1&quot;) %&gt;%
  mutate(gene = str_extract(V2.y, &quot;(?&lt;=\\[gene=)\\w+&quot;)) %&gt;%
  select(gene, V11, V2.x) %&gt;%
  mutate(SPID = str_extract(V2.x, &quot;(?&lt;=\\|)[^\\|]*(?=\\|)&quot;)) %&gt;%
  distinct(gene, SPID, .keep_all = TRUE) %&gt;%
  select(SPID) %&gt;%
  write.table(file = &quot;DRAFT_Funct_Enrich/annot/SPID.txt&quot;, sep = &quot;\t&quot;, row.names = FALSE, quote = FALSE
 ) </code></pre>
<p>With a list of matching Swiss-Prot IDs, (technically UniProt
Accession number) we can go back to <a href="https://www.uniprot.org"
class="uri">https://www.uniprot.org</a> and grab corresponding GO terms.
This can be done via a web or using Python API.</p>
<p><strong>Using Web</strong></p>
<p>Using ID Mapping</p>
<div class="float">
<img
src="http://gannet.fish.washington.edu/seashell/snaps/Monosnap_RetrieveID_mapping__UniProt_2023-07-10_10-03-33.png"
alt="id" />
<div class="figcaption">id</div>
</div>
<div class="float">
<img
src="http://gannet.fish.washington.edu/seashell/snaps/Monosnap_Tool_results__UniProt_2023-07-10_10-05-07.png"
alt="finished" />
<div class="figcaption">finished</div>
</div>
<p>Now will customize columns to get GO IDs.</p>
<div class="float">
<img
src="http://gannet.fish.washington.edu/seashell/snaps/Monosnap_RetrieveID_mapping_results__UniProtKB__UniProt_2023-07-10_10-07-05.png"
alt="custcol" />
<div class="figcaption">custcol</div>
</div>
<pre class="bash"><code>head -2 DRAFT_Funct_Enrich/annot/uniprotGO.tab</code></pre>
<pre><code>## From Entry   Reviewed    Entry Name  Protein names   Gene Names  Organism    Length  Gene Ontology IDs
## Q8IV45   Q8IV45  reviewed    UN5CL_HUMAN UNC5C-like protein (Protein unc-5 homolog C-like) (ZU5 and death domain-containing protein) UNC5CL ZUD  Homo sapiens (Human)    518 GO:0005042; GO:0005737; GO:0008233; GO:0016020; GO:0043123; GO:0046330</code></pre>
<p>Finally we can join table to get “LOCIDs” the notation for our DEGs,
with GO terms.</p>
<pre class="r"><code>go &lt;- read.csv(&quot;DRAFT_Funct_Enrich/annot/uniprotGO.tab&quot;, sep = &#39;\t&#39;, header = TRUE, row.names=NULL)</code></pre>
<pre class="r"><code>left_join(g.spid, go, by = c(&quot;SPID&quot; = &quot;Entry&quot;)) %&gt;%
  select(gene,Gene.Ontology.IDs) %&gt;%
  write.table(file = &quot;DRAFT_Funct_Enrich/annot/geneGO.txt&quot;, sep = &quot;\t&quot;, row.names = FALSE, quote = FALSE
  )</code></pre>
<pre class="bash"><code>head DRAFT_Funct_Enrich/annot/geneGO.txt</code></pre>
<pre><code>## gene Gene.Ontology.IDs
## LOC111126949 GO:0005042; GO:0005737; GO:0008233; GO:0016020; GO:0043123; GO:0046330
## LOC111112434 GO:0000981; GO:0003700; GO:0005634; GO:0016607; GO:0043565; GO:0045944
## LOC111120752 GO:0004750; GO:0005829; GO:0005975; GO:0006098; GO:0009052; GO:0042802; GO:0042803; GO:0046872; GO:0070062
## LOC111105685 GO:0003682; GO:0004518; GO:0005634; GO:0035098; GO:0035102; GO:0040029; GO:0046872
## LOC111113860 GO:0004062; GO:0005737; GO:0006068; GO:0006805; GO:0008146; GO:0009812; GO:0030855; GO:0042403; GO:0050427; GO:0051923
## LOC111109550 GO:0004062; GO:0005737; GO:0005764; GO:0008146; GO:0051923
## LOC111109753 GO:0004062; GO:0005737; GO:0005829; GO:0006068; GO:0006576; GO:0006805; GO:0008146; GO:0009812; GO:0018958; GO:0030855; GO:0042403; GO:0050427; GO:0051923
## LOC111109452 GO:0004062; GO:0005829; GO:0008146; GO:0009812; GO:0050427
## LOC111101273 GO:0005743; GO:0008535; GO:0032977; GO:0032979; GO:0033617; GO:0051204</code></pre>
<p><strong>Using API</strong> (DOES NOTE WORK)</p>
<pre class="python"><code>#DRAFT_Funct_Enrich/annot/uniprot-retrieval.py DRAFT_Funct_Enrich/annot/SPID.txt</code></pre>
</div>
<div id="considerations-for-a-rigourous-analysis"
class="section level1">
<h1>Considerations for a Rigourous Analysis</h1>
<p>While functional gene enrichment analysis poses many challenges to
non-model systems, it is critical the analyses conducted be as rigorous
and replicable as possible. A recent survey of published enrichment
results indicated that issues in the performance and reporting of
enrichment analyses is not uncommon <span class="citation">(Wijesooriya
et al. 2022)</span>. Basic standards advised by the authors include:</p>
<ul>
<li>Selection of appropriate gene background set. This set should not
necessarily be the full set of genes in the genome. Rather it should be
the set of genes expressed in the experiment. This is especially an
issue with overrepresentation analysis and showed a consider impact on
results when done incorrectly.</li>
<li>Multiple hypothesis correction. A common metric used with enrichment
results is a false discovery rate (FDR) correction.</li>
<li>Adequate methodological reporting including software and parameters
used, versions of software and annotation files, all gene lists analyzed
including subset and background sets, types of statistical tests and
corrections performed, and any code used to conduct the analyses.</li>
</ul>
</div>
<div id="getting-to-enrichment-analysis" class="section level1">
<h1>Getting to Enrichment Analysis</h1>
<p>Once the challenge of annotating data has been met, the next step is
determining the focus of your functional analysis. In some cases, you
will use the complete dataset, but in other cases you may be interested
in a subset of interest. There are numerous approaches that can be used
to achieve this, many with excellent tutorials available online. These
include:</p>
<ul>
<li>Differential Expression Analysis</li>
<li>Clustering</li>
<li>WGCNA</li>
</ul>
<p><br></p>
</div>
<div id="introduction-and-considerations" class="section level1">
<h1>Introduction and Considerations</h1>
<ul>
<li>The Gene Ontology (Thea)</li>
<li>Annotation and filtration decisions (Sam) - include key terms</li>
<li>Overrepresentation vs Enrichment (Thea)</li>
<li>Explanation of statistics (Sam) - include key terms</li>
</ul>
<div id="the-gene-ontology" class="section level2">
<h2>The Gene Ontology</h2>
</div>
<div id="annotation-and-filtration-decisions" class="section level2">
<h2>Annotation and filtration decisions</h2>
</div>
<div id="overrepresentation-vs-enrichment" class="section level2">
<h2>Overrepresentation vs Enrichment</h2>
<p>Often, everything gets referred to as enrichment analysis, but this
is not the case. The two primary analyses to be run that examine
functional patterns are statistical over-representation analysis and
statistial enrichment analysis. There are differences in how data are
input, what kind of data are used, and what statistical tests are
employed by each analysis.</p>
<p>Over-representation analysis is used when you do not have expression
values associated with your data. In this case you will provide a list
of a subset of genes of interest as well as a reference list. This
reference list will contain all the genes in your study. For example, in
a transcriptomic analysis the reference list would include all expressed
genes in the experiment while the gene of interest list might include
genes that are over expressed under a specific condition (e.g., genes
over expressed in females compared to all genes expressed in males and
females). These two lists are compared to one another and a Fisher’s
Exact Test is used to calculate whether any functional categories (based
on how genes are annotated to particular GO terms) are over-represented
in your list of genes of interest, relative to the representation of all
functional categories present in your full data set.</p>
<p>Enrichment analysis is used when you do have expression values
associated with your data. In this case, you will provide the full list
of genes in your experiment and their associated expression values
(e.g., this might be a library from an RNA-seq experiment). Any
expression metric can be used. This flexiblity is because enrichment
analysis uses a Mann-Whitney U Test to calculate enrichment, which is
based on rankings - thus values need to be interpretable for rank-based
relationships, and an appropriate normalization metric should be applied
to your data.</p>
<p>See later sections for further discussion of Fisher’s Exact Test and
Mann-Whitney U Test.</p>
</div>
<div id="explanation-of-statistics" class="section level2">
<h2>Explanation of Statistics</h2>
<p><br></p>
</div>
</div>
<div id="tools-and-approaches" class="section level1">
<h1>Tools and Approaches</h1>
<ul>
<li>Ontologizer (command line) (Thea)</li>
<li>Panther (online GUI) - endorsed by The Gene Ontology (Thea)</li>
<li>R…Mann-Whitney U Test and Fisher’s Exact Test (Sam)</li>
</ul>
</div>
<div id="a-primer-on-the-statistics-of-functional-enrichment"
class="section level1">
<h1>A primer on the statistics of functional enrichment</h1>
<p>Two popular methods of testing for GO term enrichment are the Fishers
exact test and Mann Whitney U tests. Each test can produce results that
are sensitive to the kinds of data they are input with such as
differential expression or WGCNA coefficients [cite]. Neither test
necessarily needs to be used alongside known networks of genes and use
GO terms, enabling their use in non-model systems. These two methods are
summarized and derived below:</p>
<p><br></p>
<div id="fishers-exact-test" class="section level2">
<h2>Fisher’s Exact Test</h2>
<p>The Fisher’s exact test evaluates the relationship between two
categorical variables: whether the ontology of a gene is represented in
a test set or a background set. In the case of differential expression
(DE), the test set is al DE genes and the background set is all genes
included in DE analysis after filtering, not the entire transcriptome.
The test asks whether the number of genes for a given GO term
represented in a test set relative to the background is greater than
what is expected due to chance alone (p).</p>
<p><span class="math display">\[
p = \frac{{\binom{{a+b}}{{a}} \cdot
\binom{{c+d}}{{c}}}}{{\binom{{n}}{{a+c}}}}
\]</span></p>
<p>Where</p>
<ul>
<li><span class="math inline">\(a =\)</span> <span
class="math inline">\(n\)</span> genes of GO term <span
class="math inline">\(i\)</span> in both the test and background
set</li>
<li><span class="math inline">\(b =\)</span> <span
class="math inline">\(n\)</span> genes of GO <span
class="math inline">\(i\)</span> in the background but not test</li>
<li><span class="math inline">\(c =\)</span> <span
class="math inline">\(n\)</span> genes of GO <span
class="math inline">\(i\)</span> absent from the test but present in the
background</li>
<li><span class="math inline">\(d =\)</span> <span
class="math inline">\(n\)</span> genes of GO <span
class="math inline">\(i\)</span> absent in both the category and the
background</li>
<li><span class="math inline">\(n =\)</span> <span
class="math inline">\(n\)</span> genes in the background set</li>
</ul>
<p>A Fisher’s exact test can be executed in base R using the
fishers.text() function</p>
<p><br></p>
</div>
<div id="mann-whitney-u-test" class="section level2">
<h2>Mann Whitney U Test</h2>
<p>The critical difference between Mann Whitney U and Fisher’s exact
tests and are that Mann Whitney U’s permit the analysis of continuous
rather than categorical variables. These continuous values can be log
fold changes of differential expression or -log p-values associated with
DE or inclusion in a WGCNA. Because it doesn’t rely on binary values
denoting the existence of an effect, but rather the strength of an
effect, the Mann Whitney U test is a more quantitative approach toward
functional enrichment. The test asks whether the distribution of ranks
for this continuous value in a GO term is significantly skewed from the
distribution of ranks across all other GO terms. Like the background set
employed in a Fisher’s exact test, “all other GO terms” refers to those
terms included in genes used for analysis following filtering.</p>
<p>The skewedness of a GO term’s rank distribution relative to the
background is represented by the statistic <span
class="math inline">\(U\)</span>. The p-value output by the Mann Whitney
U test is calculated with the following equation:</p>
<p><span class="math display">\[
p = \frac{{U}}{{a \cdot b}}
\]</span></p>
<p>Where</p>
<ul>
<li><span class="math inline">\(a =\)</span> <span
class="math inline">\(n\)</span> genes in GO term <span
class="math inline">\(i\)</span></li>
<li><span class="math inline">\(b =\)</span> <span
class="math inline">\(n\)</span> genes in the background set</li>
</ul>
<p>A Mann Whitney U test can be executed for GO enrichment analysis
using code from <span
class="citation">(<strong>Wright2015-dc?</strong>)</span></p>
<p><br></p>
<p>A number of different parameters can be used to filter results or
modify the execution of both tests, which we detail below.</p>
</div>
<div id="ontologizer" class="section level2">
<h2>Ontologizer</h2>
<p><a href="http://ontologizer.de/">Ontologizer</a> offers a
command-line option for functional analysis using the Gene Ontology and
is one of several tools endorsed by the GO. Functionally, it represents
a tool for overrepresentation analysis as it makes use of the Fisher’s
Exact Test. The most challenging part of running ontologizer is
preparing your input files and ensuring they are annotated properly.</p>
<p>To run Ontologizer on the command line the Ontologizer.jar file will
need to be downloaded from <a
href="http://ontologizer.de/commandline/">here</a>. Additionally,
Ontologizer requires java to run, so ensure you have this installed or
call the relevant module in your hpc system. It needs to be supplied
with the following:</p>
<ul>
<li>The association file (-a) relating genes to GO terms (ends in .gaf
for gene association file). A number of options are available through
the annotations link on <a
href="http://geneontology.org/docs/download-go-annotations/">this
page</a>.</li>
<li>The file (-g) containing the GO terminology and structure <a
href="http://geneontology.org/docs/download-ontology/">go-basic.obo</a>.</li>
<li>A population reference file (-p), which you will generate.</li>
<li>Files representing your gene subsets of interest (-s). Multiple
subsets can be provided at one time to Ontologizer and run in parallel
by specifying a path to the directory containing only these files.</li>
</ul>
<p>There are a number of other options that can be specified. Some worth
consider are your calculation method (-c), whether to ignore genes with
no associations in your calculations (-i), and the name of the directory
that you want your output written to (-o).</p>
<p>A sample script run on an hpc cluster may look like the
following:</p>
<pre><code>java -jar Ontologizer.jar -a goa_uniprot_all.gaf -g go-basic.obo -p pop.CPISwissProtIDs.txt -s Subset/CPIDEA/StudySet -c &quot;MGSA&quot; -d 0.05 -i -o CPIOntologizerSubsetOutput</code></pre>
</div>
<div id="panther" class="section level2">
<h2>PANTHER</h2>
<p>PANTHER is a GUI option that is endorsed by The Gene Ontology and
integrated into their website. PANTHER stands for Protein ANalysis
THrough Evolutionary Relationships.</p>
<div id="preparing-data-for-input-into-panther." class="section level3">
<h3>Preparing data for input into Panther.</h3>
<p>Here we will go over how to map your transcripts to PANTHER IDs for
input into the PANTHER analysis pipeline. PANTHER can accept a number of
ID types, but this is method may be useful if you are working in a
nonmodel system with limited annotation data available.</p>
<p>Download the PANTHER HMM scoring tool from <a
href="http://pantherdb.org/downloads/index.jsp">here</a></p>
<pre><code>wget http://data.pantherdb.org/ftp/hmm_scoring/current_release</code></pre>
<p>Unpack the tar file.</p>
<pre><code>tar -xvzf PANTHER17.0_hmmscoring.tgz</code></pre>
<p>Using the provided script located at
pantherScore2.2/pantherScore2.2.pl, submit amino acid sequences to be
searched against the PANTHER HMM library and assign appropriate IDs.
Thus you may need to translate your transcript sequences using a tool
such as <a
href="https://github.com/TransDecoder/TransDecoder/wiki">TransDecoder</a>.
Searching your sequences against the PANTHER HMM library can be done on
the command line. You will need to load perl and hmmer software
packages. Test data are provided to check that everything installed
properly. Once you are ready to go, you can run some version of the
following. See command options for alternative variables to
consider.</p>
<pre><code>pantherScore2.2.pl -l ../target/famlib/rel/PANTHER17.0_altVersion/hmmscoring/PANTHER17.0/ -D B -V -i translated_coding_sequences.faa -o hits.txt -n</code></pre>
<pre><code>-l PANTER (l)ibrary with HMMs when using hmmsearch, binary PANTHER HMM model when using hmmscan
-D display type for results
    options: I (interproscan), B (best hit), A (all hits)
-o (o)utput file (redirect STDOUT)
-i (i)nput fasta file to score
-V (V)ery verbose (debug info to STDERR)</code></pre>
<p>The results will look something like this:</p>
<pre><code>CpiID   PantherID       GeneName        HmmEvalue       HmmBitscore     AlignRange
lcl|NC_024225.1_prot_XP_023959495.1_6146        PTHR46343:SF3   SUSHI REPEAT-CONTAINING PROTEIN SRPX2   1.2e-266        887.6   9-462
lcl|NW_007359912.1_prot_XP_008164062.1_46642    PTHR10048:SF33  PHOSPHATIDYLINOSITOL 4,5-BISPHOSPHATE 3-KINASE CATALYTIC SUBUNIT BETA ISOFORM   0       2111.3  1-1040
lcl|NW_007281451.1_prot_XP_008170789.1_21777    PTHR23119:SF48  LEUCINE-RICH REPEAT-CONTAINING PROTEIN 7        0       2508.4  7-1351
lcl|NW_007281531.1_prot_XP_005308912.1_25857    PTHR12157:SF18  REGULATING SYNAPTIC MEMBRANE EXOCYTOSIS PROTEIN 1       0       2637.1  1-1502
lcl|NW_007359900.1_prot_XP_005287749.1_44438    PTHR13040:SF2   AUTOPHAGY PROTEIN 5     3.2e-105        355.0   3-272
lcl|NW_007359900.1_prot_XP_005287749.1_44438    PTHR13040       AUTOPHAGY PROTEIN 5     3.2e-105        355.0   3-272
lcl|NW_007282136.1_prot_XP_005313942.1_33204    PTHR26450:SF179 OLFACTORY RECEPTOR      2.6e-146        489.7   1-311
...</code></pre>
<p>Now that you have the PANTHER annotations, the next step is to do any
final parsing of your data for input into PANTHER. For
overrepresentation analysis you will need a reference file representing
your population of genes and a subset file representing your set of
target genes of interest. They can both take the format for Generic
Mapping which follows the form shown above. If performing enrichment
analysis, some extra data formatting will be needed. In this case, input
should contain three columns: a gene ID (in a form useful to you), the
PANTHER ID (resulting from the search against the HMM library described
above), and an expression metric for each gene.</p>
</div>
<div id="running-panther-overrepresentation" class="section level3">
<h3>Running PANTHER Overrepresentation</h3>
<p>Go to <a href="http://pantherdb.org"
class="uri">http://pantherdb.org</a>. This will bring up the start page
which will look something like this:</p>
<p>image: <img src="DRAFT_Funct_Enrich/PantherStartPage.png" /></p>
<p>Step 1 is to upload your file of interest. If you are doing an
overrepresentation analysis, this would be a your subset list of genes.
Your list type will need to be specified. If you are working with a
non-model system with limited genomic resources, chances are you will
want to use the PANTHER Generic Mapping option (see elsewhere for
instructions on preparing this data). See both the Supported IDs and
File format links on this page for descriptions of acceptable IDs.</p>
<p>Step 2 depends on your selection in step one. Here is where you would
specify your organism (and consequent reference file) if it is avialable
within the PANTHER database. If your organism is not avialable, you will
have selected PANTHER Generic Mapping and will provide your own
reference set (for overrepresenation analysis only) at a later stage and
this step will be skipped.</p>
<p>Step 3 selects the type of analysis you would like to run. Here we
will discuss the statistical overrperesentation test.</p>
<p>Here is what things will look like once your data are uploaded:</p>
<p>image: <img
src="DRAFT_Funct_Enrich/PantherStartPageWithData.png" /></p>
<p>Click Submit.</p>
<p>You will now be brought to the second data entry page. Here is where
you will provide your reference list. If you are working with a species
present in the PANTHER DATABASE you can specify that here. Alternatively
you can provide your own. In this example we will provide a PANTHER
Generic Mapping File, since we also provided this type of file for our
subset list.</p>
<p>First, select the list type - in this case we select PANTHER Generic
Mapping File. Then choose file; you will be able to upload a file from
your computer. Select upload list and you will be brought the the
analysis page.</p>
<p>Prior to selecting upload list, the page should look something like
this:</p>
<p>image: <img src="DRAFT_Funct_Enrich/PantherReferencePage.png" /></p>
<p>Once you have uploaded your reference file you will be brought to a
new page. You should see your subset of interest file after Analyzed
List and your population reference file after Reference List. You can
select which annotation data set you want to query:</p>
<ul>
<li>PANTHER Pathways</li>
<li>PANTHER GO-Slim Molecular Function</li>
<li>PANTHER GO-Slim Biological Process</li>
<li>PANTHER GO-Slim Cellular Component</li>
<li>PANTHER Protein Class</li>
</ul>
<p>Select your test type (either Fisher’s Exact Test or Binomial) and
test correction type (False Discovery Rate, Bonferroni, or None).</p>
<p>Here is an example of what the page will look like once it is ready
to go:</p>
<p>image: <img src="DRAFT_Funct_Enrich/PantherAnalysisPage.png" /></p>
<p>Launch Analysis.</p>
<p>Results will load below. The default is to only show significant
results. There are several options for downloading results. Simply
select the type you wish to download and save it to your computer.</p>
<p>Here is an example of the results page when there are significant
hits:</p>
<p>image: <img src="DRAFT_Funct_Enrich/PantherResults.png" /></p>
</div>
<div id="running-panther-enrichment" class="section level3">
<h3>Running PANTHER Enrichment</h3>
<p>Go to <a href="http://pantherdb.org"
class="uri">http://pantherdb.org</a>. This will bring up the start page
which will look something like this:</p>
<p>image: <img src="DRAFT_Funct_Enrich/PantherStartPage.png" /></p>
<p>Step 1 is to upload your file of interest. If you are running
enrichment analysis, then you would upload your full set of genes with
expression values. Your list type will need to be specified. If you are
working with a non-model system with limited genomic resources, chances
are you will want to use the PANTHER Generic Mapping option (see
elsewhere for instructions on preparing this data). See both the
Supported IDs and File format links on this page for descriptions of
acceptable IDs.</p>
<p>Step 2 depends on your selection in step one. Here is where you would
specify your organism if it is avialable within the PANTHER database. If
your organism is not avialable, you will have selected PANTHER Generic
Mapping and this step will be skipped.</p>
<p>Step 3 selects the type of analysis you would like to run. Here we
will discuss the statistical enrichment test.</p>
<p>Here is what things will look like once your data are uploaded:</p>
<p>image: <img
src="DRAFT_Funct_Enrich/PantherStartPageWithDataEnrichment.png" /></p>
<p>Click Submit.</p>
<p>This will bring up the results page for the analysis. You can select
a different type of annotation data set to analyze if you wish as well
as change the multiple hypothesis correction type. The data sets
available are the same as for running the overrepresentation test. A
false discovery rate correction is applied automatically and any
significant results are displayed below, but Bonferroni and none are
options as well. There are several options for downloading results.
Simply select the type you wish to download and save it to your
computer.</p>
<p>Here is an example of the results page when there are significant
hits:</p>
<p>image: <img
src="DRAFT_Funct_Enrich/PantherResultsEnrichment.png" /></p>
</div>
</div>
<div id="using-r" class="section level2">
<h2>Using R</h2>
<p><br></p>
</div>
</div>
<div id="what-to-do-with-the-output" class="section level1">
<h1>What to Do with the Output</h1>
<ul>
<li>Visualizing results</li>
<li>Interpreting results - gather consensus advice</li>
<li>R…Mann-Whitney U Test and Fisher’s Exact Test (Sam)</li>
</ul>
<p><br></p>
</div>
<div id="formatting-examples" class="section level1">
<h1>Formatting Examples</h1>
<p><br></p>
<div id="example-of-table-syntax" class="section level2">
<h2>Example of table syntax</h2>
<table>
<thead>
<tr class="header">
<th align="left">Term A</th>
<th align="left">Term B</th>
<th align="left">Term C</th>
<th align="left">Term D</th>
<th align="left">Term E</th>
<th align="left">Term F</th>
<th align="left">Term G</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">xxx</td>
<td align="left">xxx</td>
<td align="left">xxx</td>
<td align="left">xxx</td>
<td align="left">xxx</td>
<td align="left">xxx</td>
<td align="left">xxx</td>
</tr>
<tr class="even">
<td align="left">yyy</td>
<td align="left">yyy</td>
<td align="left">yyy</td>
<td align="left">yyy</td>
<td align="left">yyy</td>
<td align="left">yyy</td>
<td align="left">yyy</td>
</tr>
<tr class="odd">
<td align="left">zzz</td>
<td align="left">zzz</td>
<td align="left">zzz</td>
<td align="left">zzz</td>
<td align="left">zzz</td>
<td align="left">zzz</td>
<td align="left">zzz</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
<div id="example-of-how-to-insert-an-image" class="section level2">
<h2>Example of how to insert an image</h2>
<p><img src="DRAFT_Funct_Enrich/Rivera_etal_fig.png" width="1137" height="40%" style="display: block; margin: auto;" /></p>
</div>
<div id="example-of-how-to-write-and-print-an-equation"
class="section level2">
<h2>Example of how to write and print an equation</h2>
<p>For illustrative purposes, here is a simple and famous equation</p>
<p><span class="math display">\[
E = mc^2
\]</span></p>
<p><br></p>
<p>As an example of a more complex equation, here is an equation for the
t-statistic</p>
<p><span class="math display">\[
t = \frac{\overline{x_{1}}-\overline{x_{2}}}
{\sqrt{(s^2(
\frac{1}{n_{1}} + \frac{1}{n_{2}}
))}}
\]</span></p>
</div>
<div id="example-of-how-to-cite-a-reference-from-your-.bib-in-text"
class="section level2">
<h2>Example of how to cite a reference from your .bib in-text</h2>
<p>Functional enrichment tests applied to mRNA analyses often impose
significant biases <span class="citation">(Bleazard, Lamb, and
Griffiths-Jones 2015)</span>.</p>
<pre><code>```{r setup, include = FALSE} 

# Set root directory... in this draft, ensure that you have the same path
knitr::opts_knit$set(root.dir = &#39;~/Documents/GitHub/marineomics.github.io/&#39;)

```</code></pre>
<p><br></p>
</div>
</div>
<div id="lets-get-started" class="section level1">
<h1>Let’s get started</h1>
<p>Below is an example of what should be the first, visible code
chunk</p>
<pre class="r"><code>## Unhash and run the code below if you believe you may need to install the packages loaded below
# Edit code below to list packages that are used in this draft... at present these are just examples

if (!require(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)

BiocManager::install(c(&quot;DESeq2&quot;,&quot;edgeR&quot;,&quot;arrayQualityMetrics&quot;))

# Load packages
library(tidyverse)</code></pre>
</div>
<div
id="run-gomwu-and-fishers-exact-tests-on-data-of-interest-see-httpsgithub.comz0ongo_mwu"
class="section level1">
<h1>Run GOMWU and Fisher’s exact tests on data of interest (see <a
href="https://github.com/z0on/GO_MWU"
class="uri">https://github.com/z0on/GO_MWU</a>)</h1>
<pre class="r"><code># GO enrichments using Mann Whitney U tests
setwd(&quot;GO_MWU-master/&quot;)

#Now we multiply them together into a new column

## Ok we also want to bring in our annotations
#I would suggest putting a GO_MWU folder into your working directory for these steps. There is some stuff to download such as the obo files 
#See https://github.com/z0on/GO_MWU

# GO_MWU uses continuous measure of significance (such as fold-change or -log(p-value) ) to identify GO categories that are significantly enriches with either up- or down-regulated genes. The advantage - no need to impose arbitrary significance cutoff.
# If the measure is binary (0 or 1) the script will perform a typical &quot;GO enrichment&quot; analysis based Fisher&#39;s exact test: it will show GO categories over-represented among the genes that have 1 as their measure. 
# On the plot, different fonts are used to indicate significance and color indicates enrichment with either up (red) or down (blue) regulated genes. No colors are shown for binary measure analysis.
# The tree on the plot is hierarchical clustering of GO categories based on shared genes. Categories with no branch length between them are subsets of each other.
# The fraction next to GO category name indicates the fracton of &quot;good&quot; genes in it; &quot;good&quot; genes being the ones exceeding the arbitrary absValue cutoff (option in gomwuPlot). For Fisher&#39;s based test, specify absValue=0.5. This value does not affect statistics and is used for plotting only.
# Stretch the plot manually to match tree to text

# Mikhail V. Matz, UT Austin, February 2015; matz@utexas.edu

################################################################
# First, press command-D on mac or ctrl-shift-H in Rstudio and navigate to the directory containing scripts and input files. Then edit, mark and execute the following bits of code, one after another.

# Edit these to match your data file names: 
input = &quot;GOMWU_input.csv&quot; # two columns of comma-separated values: gene id, continuous measure of significance. To perform standard GO enrichment analysis based on Fisher&#39;s exact test, use binary measure (0 or 1, i.e., either sgnificant or not).
goAnnotations = &quot;goAnnot_spu.tab&quot; # two-column, tab-delimited, one line per gene, multiple GO terms separated by semicolon. If you have multiple lines per gene, use nrify_GOtable.pl prior to running this script.
goDatabase = &quot;go.obo&quot; # download from http://www.geneontology.org/GO.downloads.ontology.shtml
goDivision = &quot;MF&quot; # either MF, or BP, or CC
source(&quot;gomwu.functions.R&quot;)

# Calculating stats. It might take ~3 min for MF and BP. Do not rerun it if you just want to replot the data with different cutoffs, go straight to gomwuPlot. If you change any of the numeric values below, delete the files that were generated in previos runs first.
gomwuStats(input, goDatabase, goAnnotations, goDivision,
    perlPath = &quot;perl&quot;, # replace with full path to perl executable if it is not in your system&#39;s PATH already
    largest = 0.1, # a GO category will not be considered if it contains more than this fraction of the total number of genes
    smallest = 25,   # a GO category should contain at least this many genes to be considered
    clusterCutHeight = 0.25, # threshold for merging similar (gene-sharing) terms. See README for details.
    Alternative = &quot;g&quot;) # # threshold for merging similar (gene-sharing) terms. See README for details.
           #    Alternative=&quot;g&quot; # by default the MWU test is two-tailed; specify &quot;g&quot; or &quot;l&quot; of you want to test for &quot;greater&quot; or &quot;less&quot; instead. 
           #    Module=TRUE,Alternative=&quot;g&quot; # un-remark this if you are analyzing a SIGNED WGCNA module (values: 0 for not in module genes, kME for in-module genes). In the call to gomwuPlot below, specify absValue=0.001 (count number of &quot;good genes&quot; that fall into the module)
           #    Module=TRUE # un-remark this if you are analyzing an UNSIGNED WGCNA module
# do not continue if the printout shows that no GO terms pass 10% FDR.

quartz.save(file = &quot;quartz.png&quot;, type = &quot;png&quot;)
results=gomwuPlot(input, goAnnotations, goDivision,
#   absValue=-log(0.05,10),  # genes with the measure value exceeding this will be counted as &quot;good genes&quot;. Specify absValue=0.001 if you are doing Fisher&#39;s exact test for standard GO enrichment or analyzing a WGCNA module (all non-zero genes = &quot;good genes&quot;).
    absValue= .2,
    level1=0.01, # FDR threshold for plotting. Specify level1=1 to plot all GO categories containing genes exceeding the absValue.
    level2=0.005, # FDR cutoff to print in regular (not italic) font.
    level3=0.001, # FDR cutoff to print in large bold font.
    txtsize=1.2,    # decrease to fit more on one page, or increase (after rescaling the plot so the tree fits the text) for better &quot;word cloud&quot; effect
    treeHeight=0.5 # height of the hierarchical clustering tree
#   colors=c(&quot;dodgerblue2&quot;,&quot;firebrick1&quot;,&quot;skyblue&quot;,&quot;lightcoral&quot;) # these are default colors, un-remar and change if needed
)
# manually rescale the plot so the tree matches the text 
# if there are too many categories displayed, try make it more stringent with level1=0.05,level2=0.01,level3=0.001.  

# text representation of results, with actual adjusted p-values
results

# Export list of GO terms enriched among genes in lowest quartiles of TSS acess and expression
write.csv(results[1], 
          &quot;~/Documents/GitHub/marineomics.github.io/DRAFT_Funct_Enrich/MF_output.csv&quot;, 
          row.names = TRUE)

# Next, manually repeat this code chunk using BP and CC terms and change exported name of file</code></pre>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<p>Below are the markdown-formatted references for our document derived
from the</p>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Bleazard2015-vn" class="csl-entry">
Bleazard, Thomas, Janine A Lamb, and Sam Griffiths-Jones. 2015.
<span>“Bias in <span class="nocase">microRNA</span> Functional
Enrichment Analysis.”</span> <em>Bioinformatics</em> 31 (10): 1592–98.
</div>
<div id="ref-RN335" class="csl-entry">
Wijesooriya, K. L., S. Jadaan, K. L. Perera, T. Kaur, and M. L. Ziemann.
2022. <span>“Urgent Need for Consistent Standards in Functional
Enrichment Analysis.”</span> Journal Article. <em>Plos Computational
Biology</em> 18 (3): 14. <a
href="https://doi.org/10.1371/journal.pcbi.1009935">https://doi.org/10.1371/journal.pcbi.1009935</a>.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
