---
title: 'Functional Enrichment Tests: Intro and methods comparison'
subtitle: '*Thea Gessler*, *Samuel N. Bogan*, and *Steven Roberts*'
bibliography: DRAFT_Funct_Enrich/Funct_Enrich.bib
---

# Background

Functional gene enrichment analysis is a powerful tool that can aid in the interpretation of biological data by identifying biological processes and pathways that are significantly overrepresented in a gene list of interest. However, to effectively use this tool, it is essential to have a proper annotation of genes.

Proper annotation of genes is crucial because it ensures that the genes are correctly identified and annotated with their appropriate biological functions. Incorrect annotation of genes can lead to inaccurate interpretations of experimental results and hinder progress in understanding the biological systems of non-model species. 

Therefore, the quality of gene annotation is critical for accurate functional gene enrichment analysis. This involves careful curation of gene annotations based on experimental evidence from various sources, including transcriptomic and proteomic data, literature mining, and functional assays. This curation process can be particularly challenging for non-model species with limited genomic and transcriptomic resources. More information on this can be found at

- Example 1
- Example 2 


**In some instances high quality annotation might not be possible but it is critical that before you undergo enrichment analysis you have a full understanding of how your data set was annotated, and any associated limitations**






# Getting to Enrichment Analysis

Finding your subset of genes of interest 

* Differential expression
* Clustering
* WGCNA 

<br>

# Introduction and Considerations 

* The Gene Ontology (Thea)
* Annotation and filtration decisions (Sam) - include key terms
* Overrepresentation vs Enrichment (Thea)
* Explanation of statistics (Sam) - include key terms

## The Gene Ontology

## Annotation and filtration decisions

## Overrepresentation vs Enrichment

Often, everything gets referred to as enrichment analysis, but this is not the case. The two primary analyses to be run that examine functional patterns are statistical over-representation analysis and statistial enrichment analysis. There are differences in how data are input, what kind of data are used, and what statistical tests are employed by each analysis.

Over-represenation analysis is used when you do not have expression values associated with your data. In this case you will provide a list of a subset of genes of interest as well as a reference list. This reference list will contain all the genes in your study. For example, in a transcriptomic analysis the reference list would include all expressed genes in the experiment while the gene of interest list might include genes that are over expressed under a specific condition (e.g., genes over expressed in females compared to all genes expressed in males and females). These two lists are compared to one another and a Fisher's Exact Test is used to calculate whether any functional categories (based on how genes are annotated to particular GO terms) are over-represented in your list of genes of interest, relative to the representation of all functional categories present in your full data set. 

Enrichment analysis is used when you do have expression values associated with your data. In this case, you will provide the full list of genes in your experiment and their associated expression values (e.g., this might be a library from an RNA-seq experiment). Any expression metric can be used. This flexiblity is because enrichment analysis uses a Mann-Whitney U Test to calculate enrichment, which is based on rankings - thus values need to be interpretable for rank-based relationships, and an appropriate normalization metric should be applied to your data. 

See later sections for further discussion of Fisher's Exact Test and Mann-Whitney U Test.

## Explanation of Statistics

<br>

# Tools and Approaches

* Ontologizer (command line) (Thea)
* Panther (online GUI) - endorsed by The Gene Ontology (Thea)
* R...Mann-Whitney U Test and Fisher’s Exact Test (Sam) 

## Ontologizer

[Ontologizer](http://ontologizer.de/) offers a command-line option for functional analysis using the Gene Ontology and is one of several tools endorsed by the GO. Functionally, it represents a tool for overrepresentation analysis as it makes use of the Fisher's Exact Test. The most challenging part of running ontologizer is preparing your input files and ensuring they are annotated properly. 

To run Ontologizer on the command line the Ontologizer.jar file will need to be downloaded from [here](http://ontologizer.de/commandline/). Additionally, Ontologizer requires java to run, so ensure you have this installed or call the relevant module in your hpc system.  It needs to be supplied with the following:

* The association file (-a) relating genes to GO terms (ends in .gaf for gene association file). A number of options are available through the annotations link on [this page](http://geneontology.org/docs/download-go-annotations/).
* The file (-g) containing the GO terminology and structure [go-basic.obo](http://geneontology.org/docs/download-ontology/).
* A population reference file (-p), which you will generate.
* Files representing your gene subsets of interest (-s). Multiple subsets can be provided at one time to Ontologizer and run in parallel by specifying a path to the directory containing only these files. 

There are a number of other options that can be specified. Some worth consider are your calculation method (-c), whether to ignore genes with no associations in your calculations (-i), and the name of the directory that you want your output written to (-o).

A sample script run on an hpc cluster may look like the following:

```
java -jar Ontologizer.jar -a goa_uniprot_all.gaf -g go-basic.obo -p pop.CPISwissProtIDs.txt -s Subset/CPIDEA/StudySet -c "MGSA" -d 0.05 -i -o CPIOntologizerSubsetOutput
```

## PANTHER

## Using R

<br>

# What to Do with the Output

* Visualizing results
* Interpreting results - gather consensus advice 
* R...Mann-Whitney U Test and Fisher’s Exact Test (Sam) 

<br>

# Formatting Examples

<br>

## Example of table syntax

| Term A | Term B | Term C | Term D | Term E | Term F | Term G |
| :--- | :--- | :--- | :--- | :--- | :--- | :--- |
| xxx | xxx | xxx | xxx | xxx | xxx | xxx |  
| yyy | yyy | yyy | yyy | yyy | yyy | yyy | 
| zzz | zzz | zzz | zzz | zzz | zzz | zzz | 

<br>

## Example of how to insert an image

```{r, echo = FALSE, out.height = "40%", fig.align = "center", }

knitr::include_graphics("DRAFT_Funct_Enrich/Rivera_etal_fig.png")

```

## Example of how to write and print an equation

For illustrative purposes, here is a simple and famous equation

$$
E = mc^2
$$

<br>

As an example of a more complex equation, here is an equation for the t-statistic

$$
t = \frac{\overline{x_{1}}-\overline{x_{2}}}
{\sqrt{(s^2(
\frac{1}{n_{1}} + \frac{1}{n_{2}}
))}}
$$ 

## Example of how to cite a reference from your .bib in-text

Functional enrichment tests applied to mRNA analyses often impose significant biases [@Bleazard2015-vn].


    ```{r setup, include = FALSE} `r ''`

    # Set root directory... in this draft, ensure that you have the same path
    knitr::opts_knit$set(root.dir = '~/Documents/GitHub/marineomics.github.io/')

    ```

<br>

# Let's get started

Below is an example of what should be the first, visible code chunk

```{r, results = FALSE, message = FALSE, warning = FALSE}

## Unhash and run the code below if you believe you may need to install the packages loaded below
# Edit code below to list packages that are used in this draft... at present these are just examples

if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install(c("DESeq2","edgeR","arrayQualityMetrics"))

# Load packages
library(tidyverse)

```

# Run GOMWU and Fisher's exact tests on data of interest (see https://github.com/z0on/GO_MWU)

```{r, include = TRUE, eval=F}

# GO enrichments using Mann Whitney U tests
setwd("GO_MWU-master/")

#Now we multiply them together into a new column

## Ok we also want to bring in our annotations
#I would suggest putting a GO_MWU folder into your working directory for these steps. There is some stuff to download such as the obo files 
#See https://github.com/z0on/GO_MWU

# GO_MWU uses continuous measure of significance (such as fold-change or -log(p-value) ) to identify GO categories that are significantly enriches with either up- or down-regulated genes. The advantage - no need to impose arbitrary significance cutoff.
# If the measure is binary (0 or 1) the script will perform a typical "GO enrichment" analysis based Fisher's exact test: it will show GO categories over-represented among the genes that have 1 as their measure. 
# On the plot, different fonts are used to indicate significance and color indicates enrichment with either up (red) or down (blue) regulated genes. No colors are shown for binary measure analysis.
# The tree on the plot is hierarchical clustering of GO categories based on shared genes. Categories with no branch length between them are subsets of each other.
# The fraction next to GO category name indicates the fracton of "good" genes in it; "good" genes being the ones exceeding the arbitrary absValue cutoff (option in gomwuPlot). For Fisher's based test, specify absValue=0.5. This value does not affect statistics and is used for plotting only.
# Stretch the plot manually to match tree to text

# Mikhail V. Matz, UT Austin, February 2015; matz@utexas.edu

################################################################
# First, press command-D on mac or ctrl-shift-H in Rstudio and navigate to the directory containing scripts and input files. Then edit, mark and execute the following bits of code, one after another.

# Edit these to match your data file names: 
input = "GOMWU_input.csv" # two columns of comma-separated values: gene id, continuous measure of significance. To perform standard GO enrichment analysis based on Fisher's exact test, use binary measure (0 or 1, i.e., either sgnificant or not).
goAnnotations = "goAnnot_spu.tab" # two-column, tab-delimited, one line per gene, multiple GO terms separated by semicolon. If you have multiple lines per gene, use nrify_GOtable.pl prior to running this script.
goDatabase = "go.obo" # download from http://www.geneontology.org/GO.downloads.ontology.shtml
goDivision = "MF" # either MF, or BP, or CC
source("gomwu.functions.R")

# Calculating stats. It might take ~3 min for MF and BP. Do not rerun it if you just want to replot the data with different cutoffs, go straight to gomwuPlot. If you change any of the numeric values below, delete the files that were generated in previos runs first.
gomwuStats(input, goDatabase, goAnnotations, goDivision,
	perlPath = "perl", # replace with full path to perl executable if it is not in your system's PATH already
	largest = 0.1, # a GO category will not be considered if it contains more than this fraction of the total number of genes
	smallest = 25,   # a GO category should contain at least this many genes to be considered
	clusterCutHeight = 0.25, # threshold for merging similar (gene-sharing) terms. See README for details.
	Alternative = "g") # # threshold for merging similar (gene-sharing) terms. See README for details.
           #	Alternative="g" # by default the MWU test is two-tailed; specify "g" or "l" of you want to test for "greater" or "less" instead. 
           #	Module=TRUE,Alternative="g" # un-remark this if you are analyzing a SIGNED WGCNA module (values: 0 for not in module genes, kME for in-module genes). In the call to gomwuPlot below, specify absValue=0.001 (count number of "good genes" that fall into the module)
           #	Module=TRUE # un-remark this if you are analyzing an UNSIGNED WGCNA module
# do not continue if the printout shows that no GO terms pass 10% FDR.

quartz.save(file = "quartz.png", type = "png")
results=gomwuPlot(input, goAnnotations, goDivision,
#	absValue=-log(0.05,10),  # genes with the measure value exceeding this will be counted as "good genes". Specify absValue=0.001 if you are doing Fisher's exact test for standard GO enrichment or analyzing a WGCNA module (all non-zero genes = "good genes").
	absValue= .2,
	level1=0.01, # FDR threshold for plotting. Specify level1=1 to plot all GO categories containing genes exceeding the absValue.
	level2=0.005, # FDR cutoff to print in regular (not italic) font.
	level3=0.001, # FDR cutoff to print in large bold font.
	txtsize=1.2,    # decrease to fit more on one page, or increase (after rescaling the plot so the tree fits the text) for better "word cloud" effect
	treeHeight=0.5 # height of the hierarchical clustering tree
#	colors=c("dodgerblue2","firebrick1","skyblue","lightcoral") # these are default colors, un-remar and change if needed
)
# manually rescale the plot so the tree matches the text 
# if there are too many categories displayed, try make it more stringent with level1=0.05,level2=0.01,level3=0.001.  

# text representation of results, with actual adjusted p-values
results

# Export list of GO terms enriched among genes in lowest quartiles of TSS acess and expression
write.csv(results[1], 
          "~/Documents/GitHub/marineomics.github.io/DRAFT_Funct_Enrich/MF_output.csv", 
          row.names = TRUE)

# Next, manually repeat this code chunk using BP and CC terms and change exported name of file

```

# References

Below are the markdown-formatted references for our document derived from the 
