<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>DNA Methylation Assessment</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="tutorial.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MarineOmics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="ADMIN_04_best_principles.html">Best Principles</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Contributions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ADMIN_01_submissions_instructions.html">Guide for Building a Page</a>
    </li>
    <li>
      <a href="ADMIN_02_contributions.html">Past and Current Contributors</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Population Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="POP_01_choosing_population_genetics.html">Choosing a Population Genomics Approach</a>
    </li>
    <li>
      <a href="POP_04_WGS_intro.html">Whole Genome Resequencing</a>
    </li>
    <li>
      <a href="RADseq.html">Reduced Representation Sequencing</a>
    </li>
    <li>
      <a href="POP_03_poolseq.html">Poolseq</a>
    </li>
    <li>
      <a href="POP_05_RDAtraitPredictionTutorial.html">Redundancy Analysis (RDA) Trait Prediction</a>
    </li>
    <li>
      <a href="POP_08_PCA.html">PCA</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Functional Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="DGE_comparison_v2.html">Mutifactorial RNAseq</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Genome-Phenome
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">coming soon!</li>
  </ul>
</li>
<li>
  <a href="ADMIN_03_panels.html">Panel Seminars</a>
</li>
<li>
  <a href="https://github.com/MarineOmics/marineomics.github.io/discussions">Discussion Forum</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'G-53GH9PV49T', 'auto');
  ga('send', 'pageview');

</script>

<div id="header">



<h1 class="title toc-ignore">DNA Methylation Assessment</h1>
<h3 class="subtitle"><em>Yaamini Venkataraman</em>, <em>Sam White</em>,
and <em>Steven Roberts</em></h3>

</div>


<p>Initial publication year: TBD <br> <a
href="https://marineomics.github.io/#How_to_Cite">How to cite</a></p>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Thanks to advances in sample preparation and sequencing methods,
there is an influx of research examining epigenomics in non-model
systems. Broadly, epigenetics can be defined as changes to gene
expression that do not arise from changes in the DNA sequence. DNA
methylation, or addition of a methyl (CH<sub>3</sub>) group to a
cytosine base adjacent to a guanine (CpG) is one of the more commonly
studies epigenetic mechanisms, partly because of its environmental
sensitivity and potential role in phenotypic plasticity. For a good
review of epigenetic mechanisms including DNA methylation in marine
organisms see <span class="citation">Eirin-Lopez and Putnam
(2019)</span>.</p>
<p>The purpose of this tutorial is to provide an example of aligning
bisulfite-treated and enzymatically converted DNA sequence data and
discuss needs for potential downstream applications. The principle
behind creating bisulfite-treated DNA libraries is that when cytosines
are not methylated, then they will be converted to uracil, which
ultimately result in thymine nucleotides in the sequence data. <span
class="citation">Trigg et al. (2022)</span> offers a comparison of three
methods for quantifying DNA methylation at single base-pair resolution
using whole genome bisulfite sequencing (WGBS), reduced representation
bisulfite sequencing (RRBS), and methyl-CpG binding domain bisulfite
sequencing (MBDBS). When aligning bisulfite-converted data to a
reference genome the percent methylation at a given CpG loci is
determined by examining the ratio of thymines to cytosines. For example
if there is a given cytosine adjacent to a guanine (ie CpG locus) with
30% of the reads containing thymine, one would consider this cytosine
loci (CpG) to be 70% methylated. Since bisulfite treatment can damage
DNA, newer library preparation and sequencing methods (ex. <a
href="https://www.neb.com/en-us/products/e7120-nebnext-enzymatic-methyl-seq-kit#Protocols,%20Manuals%20&amp;%20Usage">NEBNext®
Enzymatic Methyl-seq Kit; EM-seq</a> are starting to gain popularity. As
EM-Seq produces loci-level data, alignment of EM-Seq data is similar to
WGBS.</p>
<p>For a good review of epigenetic mechanisms including DNA methylation
in marine organisms see <span class="citation">Eirin-Lopez and Putnam
(2019)</span>. <span class="citation">Trigg et al. (2022)</span> offers
a comparison of three methods for quantifying DNA methylation at single
base-pair resolution using whole genome bisulfite sequencing (WGBS),
reduced representation bisulfite sequencing (RRBS), and methyl-CpG
binding domain bisulfite sequencing (MBDBS). The tutorial below is based
on WGBS, though the general workflow would be consistent.</p>
<p>The tutorial will cover the following:<br />
- Sequence quality assessment<br />
- Read alignment<br />
- Methylation quantification<br />
- File conversions<br />
- Potential downstream applications. Although this tutorial will not
provide code for downstream applications such as differential
methylation analysis, it will link to other open-access resources and
examples.</p>
<p>The tutorial below is based on WGBS from a species of marine
intertebrate (<em>Montipora capitata</em> coral) examined in <span
class="citation">Trigg et al. (2022)</span>, and all raw data can be
accessed under NCBI Bioproject PRJNA691891 if you would like to follow
the tutorial with this dataset. The general workflow would be consistent
for other library preparation and sequencing methods that produce
loci-level data. The tutorial indicates places where special analytical
consideration should be given to other data types.</p>
</div>
<div id="sequence-quality" class="section level1">
<h1>Sequence Quality</h1>
<p>Following trimming one would expect to see FastQC per base sequencing
content similar to the plot shown below. This particular data set was
trimmed with the following parameters:</p>
<pre><code>/gscratch/srlab/programs/TrimGalore-0.4.5/trim_galore \
--output_dir /gscratch/scrubbed/strigg/analyses/20200311/WGBS_MBD \
--paired \
--fastqc_args \
--outdir /gscratch/scrubbed/strigg/analyses/20200311/WGBS_MBD/FASTQC \
--threads 28 \
--illumina \
--clip_R1 10 \
--clip_R2 10 \
--three_prime_clip_R1 10 \
--three_prime_clip_R2 10 \
--path_to_cutadapt /gscratch/srlab/programs/miniconda3/bin/cutadapt \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth17_R1_001.fastq.gz \</code></pre>
<p><img src="DRAFT_DNA_methylation/meth-bp-chart.png" width="70%" style="display: block; margin: auto;" /></p>
<p>For WGBS and MBDBS data, it is general practice to hard-trim 10 bp
from the beginning and end of each read. <a
href="https://rawgit.com/FelixKrueger/Bismark/master/Docs/Bismark_User_Guide.html#ix-notes-about-different-library-types-and-commercial-kits">RRBS
data often needs different trimming specifications</a>.</p>
</div>
<div id="read-alignment" class="section level1">
<h1>Read Alignment</h1>
<div id="considerations" class="section level2">
<h2>Considerations</h2>
<p>The software used as part of this tutorial is <strong>Bismark
Bisulfite Mapper</strong> <span class="citation">Krueger and Andrews
(2011)</span>. As with any software it is best to be famililar with the
<a
href="https://rawgit.com/FelixKrueger/Bismark/master/Docs/Bismark_User_Guide.html">User
Manual</a>. The content below will provided with the assumption that the
reader has read the manual and is meant to serve as guidance based on
experience working with marine invertebrates.</p>
<p>The first step in the process in preparing the genome. Example
code:</p>
<pre><code># Directories and programs
bismark_dir=&quot;/programs/Bismark-0.21.0&quot;
bowtie2_dir=&quot;/programs/bowtie2-2.3.4.1-linux-x86_64/&quot;
genome_folder=&quot;/where/the/fastafile/lives/&quot;

${bismark_dir}/bismark_genome_preparation \
--verbose \
--parallel 28 \
--path_to_aligner ${bowtie2_dir} \
${genome_folder}</code></pre>
<ol style="list-style-type: decimal">
<li><p><strong>Bismark Genome Preparation Command</strong>:</p>
<ul>
<li><p><strong><code>${bismark_dir}/bismark_genome_preparation</code></strong>:
This is the command to run the genome preparation part of Bismark. The
<strong><code>${bismark_dir}</code></strong> variable is expanded to the
path where Bismark is installed, so the script knows where to find the
<strong><code>bismark_genome_preparation</code></strong>
program.</p></li>
<li><p><strong><code>--verbose</code></strong>: This flag makes the
program output more detailed information about what it is doing, which
is helpful for debugging or understanding the process.</p></li>
<li><p><strong><code>--parallel 28</code></strong>: This option tells
Bismark to use 28 threads in parallel to speed up the process.</p></li>
<li><p><strong><code>--path_to_aligner ${bowtie2_dir}</code></strong>:
This specifies the path to the aligner (Bowtie2) that Bismark will use.
The <strong><code>${bowtie2_dir}</code></strong> variable is expanded to
the path where Bowtie2 is installed.</p></li>
<li><p><strong><code>${genome_folder}</code></strong>: Finally, this
specifies the location of the genome files. The script uses the
<strong><code>${genome_folder}</code></strong> variable, which holds the
path to these files.</p></li>
</ul></li>
</ol>
<p>Bismark will create two individual folders within this directory, one
for a C-&gt;T converted genome and the other one for the G-&gt;A
converted genome. After creating C-&gt;T and G-&gt;A versions of the
genome they will be indexed in parallel usingthe indexer. Bismark will
create the following subdirectories as outputs.</p>
<pre><code>./Bisulfite_Genome
./Bisulfite_Genome/GA_conversion
./Bisulfite_Genome/CT_conversion</code></pre>
<p>For aligning the trimmed reads to the genome the following code
structure is used</p>
<pre><code>find ${reads_dir}*_R1_001_val_1.fq.gz \
| xargs basename -s _R1_001_val_1.fq.gz | xargs -I{} ${bismark_dir}/bismark \
--path_to_bowtie ${bowtie2_dir} \
-genome ${genome_folder} \
-p 4 \
-score_min L,0,-0.6 \
--non_directional \
-1 ${reads_dir}{}_R1_001_val_1.fq.gz \
-2 ${reads_dir}{}_R2_001_val_2.fq.gz \
-o Mcap_tg</code></pre>
<p>This will create BAM files (sequence alignment files)</p>
<p>Specifically this is what the code chunk does:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Finding Files and Preparing Filenames</strong>:</p>
<ul>
<li><p><strong><code>find ${reads_dir}*_R1_001_val_1.fq.gz</code></strong>:
This command uses the <strong><code>find</code></strong> utility to
search for files in the directory specified by the variable
<strong><code>reads_dir</code></strong>. The files it looks for have a
naming pattern ending with
<strong><code>_R1_001_val_1.fq.gz</code></strong>, which typically
denotes the first file of a pair in paired-end sequencing data.</p></li>
<li><p><strong><code>| xargs basename -s _R1_001_val_1.fq.gz</code></strong>:
The <strong><code>xargs</code></strong> command takes the list of files
found and passes them to <strong><code>basename</code></strong>, which
strips the suffix <strong><code>_R1_001_val_1.fq.gz</code></strong> from
each filename. This step extracts the base name of each file, which is
used to pair the R1 and R2 files correctly in the following
steps.</p></li>
</ul></li>
<li><p><strong>Running Bismark for Sequence Alignment</strong>:</p>
<ul>
<li><p><strong><code>| xargs -I{} ${bismark_dir}/bismark</code></strong>:
The <strong><code>xargs</code></strong> command here is used to pass the
base filenames to Bismark for alignment.
<strong><code>{}</code></strong> is a placeholder for each base
filename.</p></li>
<li><p><strong><code>--path_to_bowtie ${bowtie2_dir}</code></strong>:
This option specifies the path to the Bowtie2 aligner, which Bismark
uses for the alignment process.</p></li>
<li><p><strong><code>-genome ${genome_folder}</code></strong>: Specifies
the directory where the genome files, prepared in advance, are
located.</p></li>
<li><p><strong><code>-p 4</code></strong>: Sets the number of processing
threads to 4, allowing Bismark to perform the alignment in parallel,
which can speed up the process.</p></li>
<li><p><strong><code>-score_min L,0,-0.6</code></strong>: This sets the
scoring threshold for alignments, impacting the sensitivity and
specificity of the alignment.</p></li>
<li><p><strong><code>--non_directional</code></strong>: This flag
indicates the sequencing data is non-directional, which is important for
correctly interpreting the methylation status in bisulfite
sequencing.</p></li>
<li><p><strong><code>-1 ${reads_dir}{}_R1_001_val_1.fq.gz -2 ${reads_dir}{}_R2_001_val_2.fq.gz</code></strong>:
Specifies the files for paired-end reads. R1 and R2 files are paired
based on their base filename.</p></li>
<li><p><strong><code>-o Mcap_tg</code></strong>: Sets the output
directory for the alignment results to
<strong><code>Mcap_tg</code></strong>.</p></li>
</ul></li>
</ol>
<p>Files are then deduplicated if they are whole genome bisulfite
treated samples. This command will deduplicate the Bismark alignment BAM
file and remove all reads that align to the the very same position and
in the same orientation except for one. This step is recommended for
whole-genome bisulfite samples, but <strong>should not</strong> be used
for reduced representation libraries such as RRBS, amplicon, or target
enrichment libraries. The <a
href="https://rawgit.com/FelixKrueger/Bismark/master/Docs/Bismark_User_Guide.html#ix-notes-about-different-library-types-and-commercial-kits"><code>bismark</code>
manual</a> provides additional information.</p>
<pre><code>find *.bam | \
xargs basename -s .bam | \
xargs -I{} ${bismark_dir}/deduplicate_bismark \
--bam \
--paired \
{}.bam</code></pre>
</div>
</div>
<div id="methylation-quantification" class="section level1">
<h1>Methylation Quantification</h1>
<div id="what-to-expect" class="section level2">
<h2>What to expect</h2>
<p>DNA methylation at CpG locus can vary to 0-100%. Like wise
methylation at the gene level can vary and is commonly established as
mean methylation level of CpG across gene, assuming some majority degree
of gene coverage. Similarly with considering the methylation level of a
given loci often time some threshold might be set as seen in code below.
From an taxa perspective it can vary WIDELY on what to expect. The
figure below shows the mehthylation ratio of CpGs across the oyster
genome. In some marine invertebrates we see ~15% methylation of CpGs
(defined as &gt; 75% methylated) across the genome with most CpGs
unmethylated. In these taxa methylation usually occurs in gene
bodies.</p>
<p><img src="DRAFT_DNA_methylation/meth_ratio.png" width="70%" style="display: block; margin: auto;" /></p>
<p><em>Frequency distribution of methylation ratios for CpG
dinucleotides in oyster gill tissue.</em> <span class="citation">Gavery
and Roberts (2013)</span></p>
<p>Location of methylation across the genome varies across taxa. Here is
what you might expect to see in a marine invertebrate.</p>
<p><img src="DRAFT_DNA_methylation/igv.png" width="70%" style="display: block; margin: auto;" /></p>
<p><em>Screenshot from IGV. Bedgraph files are displayed. See below for
code to create these file (File Conversion)</em></p>
<p>Methylation levels are then extracted using the
<code>bismark_methylation_extractor</code>. For example:</p>
<pre><code>${bismark_dir}/bismark_methylation_extractor \
--bedGraph \
--counts \
--comprehensive \
--merge_non_CpG \
--multicore 28 \
--buffer_size 75% \
*deduplicated.bam</code></pre>
<p>This particular command is used to extract and process methylation
data from BAM files (generated after sequencing data alignment). Let’s
break down the options:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Options</strong>:</p>
<ul>
<li><p><strong><code>--bedGraph</code></strong>: Generates a BedGraph
file output, which is a format that allows visualization of the
methylation levels across the genome in genome browsers.</p></li>
<li><p><strong><code>--counts</code></strong>: Outputs the count of
methylated and unmethylated calls for each cytosine in the
genome.</p></li>
<li><p><strong><code>--comprehensive</code></strong>: This option makes
the methylation extractor process all cytosines in the genome, not just
those in a CpG context. It combines all three contexts (CpG, CHG, and
CHH) into one output file.</p></li>
<li><p><strong><code>--merge_non_CpG</code></strong>: Merges the
methylation information for CHG and CHH contexts (non-CpG) into a single
file.</p></li>
<li><p><strong><code>--multicore 28</code></strong>: Specifies that the
program should use 28 cores to process the data in parallel, which can
significantly speed up the analysis.</p></li>
<li><p><strong><code>--buffer_size 75%</code></strong>: This option sets
the buffer size for reading in BAM files to 75% of the available system
memory. It helps optimize the performance of the tool by adjusting how
much memory is used for processing.</p></li>
<li><p><strong><code>*deduplicated.bam</code></strong>: This is the
input file pattern for the command. It indicates that the methylation
extractor should process all BAM files in the current directory that
have names ending with “deduplicated.bam”. These files are typically the
output of a deduplication process applied to aligned bisulfite
sequencing data.</p></li>
</ul></li>
</ol>
<p>This will create a file with the suffix
<code>deduplicated.bismark.cov.gz</code> (example shown below in
uncompressed format). The resultant bedGraph file contains information
from <em>unmerged</em> strands, meaning reads aligning to forward and
reverse strands in the same position are not combined. Later in this
workflow, a bedGraph with <em>merged</em> information will be
created.</p>
<pre><code>NC_035784.1 141 141 37.5    3   5
NC_035784.1 142 142 100 2   0
NC_035784.1 155 155 70  7   3
NC_035784.1 156 156 100 2   0
NC_035784.1 291 291 0   0   2
NC_035784.1 292 292 0   0   3
NC_035784.1 313 313 0   0   1
NC_035784.1 314 314 66.6666666666667    2   1
NC_035784.1 470 470 66.6666666666667    4   2
NC_035784.1 611 611 0   0   4</code></pre>
<p>column organization of the file</p>
<pre><code>&lt;chromosome&gt; &lt;start position&gt; &lt;end position&gt; &lt;methylation percentage&gt; &lt;count methylated&gt; &lt;count unmethylated&gt;</code></pre>
<p>A genome-wide cytosine report is generated (including merged bedGraph
file), with this code as an example:</p>
<pre><code>find *deduplicated.bismark.cov.gz \
| xargs basename -s _trimmed_bismark_bt2.deduplicated.bismark.cov.gz \
| xargs -I{} ${bismark_dir}/coverage2cytosine \
--genome_folder ${genome_folder} \
-o {} \
--merge_CpG \
--zero_based \
{}_trimmed_bismark_bt2.deduplicated.bismark.cov.gz</code></pre>
<p>The script converts coverage files into a format that details
cytosine methylation levels across the genome. Here’s a breakdown of
each step:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Finding Files</strong>:</p>
<ul>
<li><strong><code>find *deduplicated.bismark.cov.gz</code></strong>:
This command is used to search for all files in the current directory
that end with <strong><code>deduplicated.bismark.cov.gz</code></strong>.
These files are coverage files produced by Bismark, typically after
aligning bisulfite-treated DNA sequences. They contain information about
methylation at different sites in the genome.</li>
</ul></li>
<li><p><strong>Extracting Base File Names</strong>:</p>
<ul>
<li><strong><code>| xargs basename -s _trimmed_bismark_bt2.deduplicated.bismark.cov.gz</code></strong>:
This pipeline uses <strong><code>xargs</code></strong> to pass each
found file to <strong><code>basename</code></strong>, which strips the
suffix
<strong><code>_trimmed_bismark_bt2.deduplicated.bismark.cov.gz</code></strong>
from each file name. This step extracts the base names of the files,
which will be used to generate output filenames in the following
steps.</li>
</ul></li>
<li><p><strong>Converting Coverage to Cytosine Methylation
Levels</strong>:</p>
<ul>
<li><p><strong><code>| xargs -I{} ${bismark_dir}/coverage2cytosine</code></strong>:
This pipeline uses <strong><code>xargs</code></strong> again to pass
each base filename to the
<strong><code>coverage2cytosine</code></strong> tool in the Bismark
suite. The placeholder <strong><code>{}</code></strong> is replaced by
the base filename.</p></li>
<li><p><strong><code>--genome_folder ${genome_folder}</code></strong>:
This option specifies the location of the genome folder. The
<strong><code>${genome_folder}</code></strong> variable holds the path
to this folder, which contains reference genome data necessary for the
analysis.</p></li>
<li><p><strong><code>-o {}</code></strong>: The
<strong><code>-o</code></strong> option specifies the output file name,
using the base filename <strong><code>{}</code></strong> obtained from
the previous steps.</p></li>
<li><p><strong><code>--merge_CpG</code></strong>: This flag instructs
the program to merge the methylation information of cytosines in CpG
context, providing a single methylation value for each CpG
site.</p></li>
<li><p><strong><code>--zero_based</code></strong>: This option indicates
that the output file should use zero-based coordinates, which is a
common format in bioinformatics where the first base of a sequence is
numbered 0 instead of 1.</p></li>
<li><p><strong><code>{}_trimmed_bismark_bt2.deduplicated.bismark.cov.gz</code></strong>:
Finally, this specifies the input file for each iteration, where
<strong><code>{}</code></strong> is replaced by the base filename
obtained earlier. These are the coverage files previously identified by
the <strong><code>find</code></strong> command.</p></li>
</ul></li>
</ol>
<p>Output file format <code>.CpG_report.merged_CpG_evidence.cov</code>
(merged bedGraph file)</p>
<pre><code>NC_035785.1 217 219 100.000000  17  0
NC_035785.1 523 525 87.500000   7   1
NC_035785.1 556 558 50.000000   5   5
NC_035785.1 727 729 100.000000  16  0
NC_035785.1 1330    1332    0.000000    0   2
NC_035785.1 1403    1405    0.000000    0   2
NC_035785.1 1494    1496    66.666667   2   1
NC_035785.1 1747    1749    100.000000  8   0
NC_035785.1 2024    2026    100.000000  24  0
NC_035785.1 2054    2056    93.333333   14  1</code></pre>
</div>
</div>
<div id="file-conversions" class="section level1">
<h1>File Conversions</h1>
<p>From here it could be useful to do some simple file reorganizaion to
obtain bedGraph or tab files for downstream analysis (eg bedtools,
GLMs).</p>
<p>Creating bedGraphs post-merge at a specific coverage threshold (ex.
10x or 5x):</p>
<pre><code>for f in *merged_CpG_evidence.cov
do
  STEM=$(basename &quot;${f}&quot; .CpG_report.merged_CpG_evidence.cov)
  cat &quot;${f}&quot; | awk -F $&#39;\t&#39; &#39;BEGIN {OFS = FS} {if ($5+$6 &gt;= 10) {print $1, $2, $3, $4}}&#39; \
  &gt; &quot;${STEM}&quot;_10x.bedgraph
done

for f in *merged_CpG_evidence.cov
do
  STEM=$(basename &quot;${f}&quot; .CpG_report.merged_CpG_evidence.cov)
  cat &quot;${f}&quot; | awk -F $&#39;\t&#39; &#39;BEGIN {OFS = FS} {if ($5+$6 &gt;= 5) {print $1, $2, $3, $4}}&#39; \
  &gt; &quot;${STEM}&quot;_5x.bedgraph
done</code></pre>
<p>Creating tab files with raw count data for GLMs at 10x and 5x
coverage:</p>
<pre><code>for f in *merged_CpG_evidence.cov
do
  STEM=$(basename &quot;${f}&quot; .CpG_report.merged_CpG_evidence.cov)
  cat &quot;${f}&quot; | awk -F $&#39;\t&#39; &#39;BEGIN {OFS = FS} {if ($5+$6 &gt;= 10) {print $1, $2, $3, $4, $5, $6}}&#39; \
  &gt; &quot;${STEM}&quot;_10x.tab
done

for f in *merged_CpG_evidence.cov
do
  STEM=$(basename &quot;${f}&quot; .CpG_report.merged_CpG_evidence.cov)
  cat &quot;${f}&quot; | awk -F $&#39;\t&#39; &#39;BEGIN {OFS = FS} {if ($5+$6 &gt;= 5) {print $1, $2, $3, $4, $5, $6}}&#39; \
  &gt; &quot;${STEM}&quot;_5x.tab
done</code></pre>
<p>It is sometimes useful to sort BAM files for downstream analysis (eg
methylkit, IGV)</p>
<pre><code># Sort files for methylkit and IGV
find *.bam | \
xargs basename -s .bam | \
xargs -I{} ${samtools} \
sort --threads 28 {}.bam \
-o {}.sorted.bam

# Index sorted files for IGV
find *.sorted.bam | \
xargs basename -s .sorted.bam | \
xargs -I{} ${samtools} \
index -@ 28 {}.sorted.bam</code></pre>
</div>
<div id="downstream-applications" class="section level1">
<h1>Downstream applications</h1>
<ul>
<li>Gene-level methylation analysis: people do this, but worth
considering the utility of converting loci-level resolution data to
gene-level resolution data</li>
<li>methylKit</li>
<li>IGV</li>
<li>Analysis with other datasets (ex. pop gen, RNA-Seq)</li>
</ul>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-doi:10.1146/annurev-marine-010318-095114"
class="csl-entry">
Eirin-Lopez, Jose M., and Hollie M. Putnam. 2019. <span>“Marine
Environmental Epigenetics.”</span> <em>Annual Review of Marine
Science</em> 11 (1): 335–68. <a
href="https://doi.org/10.1146/annurev-marine-010318-095114">https://doi.org/10.1146/annurev-marine-010318-095114</a>.
</div>
<div id="ref-10.7717/peerj.215" class="csl-entry">
Gavery, Mackenzie R., and Steven B. Roberts. 2013. <span>“Predominant
Intragenic Methylation Is Associated with Gene Expression
Characteristics in a Bivalve Mollusc.”</span> <em>PeerJ</em> 1
(November): e215. <a
href="https://doi.org/10.7717/peerj.215">https://doi.org/10.7717/peerj.215</a>.
</div>
<div id="ref-10.1093/bioinformatics/btr167" class="csl-entry">
Krueger, Felix, and Simon R. Andrews. 2011. <span>“<span
class="nocase">Bismark: a flexible aligner and methylation caller for
Bisulfite-Seq applications</span>.”</span> <em>Bioinformatics</em> 27
(11): 1571–72. <a
href="https://doi.org/10.1093/bioinformatics/btr167">https://doi.org/10.1093/bioinformatics/btr167</a>.
</div>
<div id="ref-https://doi.org/10.1111/1755-0998.13542" class="csl-entry">
Trigg, Shelly A., Yaamini R. Venkataraman, Mackenzie R. Gavery, Steven
B. Roberts, Debashish Bhattacharya, Alan Downey-Wall, Jose M.
Eirin-Lopez, et al. 2022. <span>“Invertebrate Methylomes Provide Insight
into Mechanisms of Environmental Tolerance and Reveal Methodological
Biases.”</span> <em>Molecular Ecology Resources</em> 22 (4): 1247–61. <a
href="https://doi.org/10.1111/1755-0998.13542">https://doi.org/10.1111/1755-0998.13542</a>.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
