<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>DNA Methylation Assessment</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="tutorial.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MarineOmics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="ADMIN_04_best_principles.html">Best Principles</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Contributions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ADMIN_01_submissions_instructions.html">Guide for Building a Page</a>
    </li>
    <li>
      <a href="ADMIN_02_contributions.html">Past and Current Contributors</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Population Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="POP_01_choosing_population_genetics.html">Choosing a Population Genomics Approach</a>
    </li>
    <li>
      <a href="POP_04_WGS_intro.html">Whole Genome Resequencing</a>
    </li>
    <li>
      <a href="RADseq.html">Reduced Representation Sequencing</a>
    </li>
    <li>
      <a href="POP_03_poolseq.html">Poolseq</a>
    </li>
    <li>
      <a href="POP_05_RDAtraitPredictionTutorial.html">Redundancy Analysis (RDA) Trait Prediction</a>
    </li>
    <li>
      <a href="POP_08_PCA.html">PCA</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Functional Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="DGE_comparison_v2.html">Mutifactorial RNAseq</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Genome-Phenome
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">coming soon!</li>
  </ul>
</li>
<li>
  <a href="ADMIN_03_panels.html">Panel Seminars</a>
</li>
<li>
  <a href="https://github.com/MarineOmics/marineomics.github.io/discussions">Discussion Forum</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'G-53GH9PV49T', 'auto');
  ga('send', 'pageview');

</script>

<div id="header">



<h1 class="title toc-ignore">DNA Methylation Assessment</h1>
<h3 class="subtitle"><em>Yaamini Venkataraman</em>, <em>Kathryn
Greiner-Ferris</em>, <em>Sam White</em>, and <em>Steven
Roberts</em></h3>

</div>


<p>Initial publication year: TBD <br> <a
href="https://marineomics.github.io/#How_to_Cite">How to cite</a></p>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<div id="background" class="section level2">
<h2>Background</h2>
<p>Thanks to advances in sample preparation and sequencing methods,
there is an influx of research examining epigenomics in non-model
systems. Broadly, epigenetics can be defined as changes to gene
expression that do not arise from changes in the DNA sequence. DNA
methylation, or addition of a methyl (CH<sub>3</sub>) group to a
cytosine base adjacent to a guanine (CpG) is one of the more commonly
studies epigenetic mechanisms, partly because of its environmental
sensitivity and potential role in phenotypic plasticity. For a good
review of epigenetic mechanisms including DNA methylation in marine
organisms see <a href="#refs"><span class="citation">Eirin-Lopez and
Putnam (2019)</span></a>.</p>
<p>The purpose of this tutorial is to provide an example of aligning
bisulfite-treated and enzymatically converted DNA sequence data to an
existing genome, and discuss needs for potential downstream
applications. The principle behind creating bisulfite-treated DNA
libraries is that when cytosines are not methylated, then they will be
converted to uracil, which ultimately result in thymine nucleotides in
the sequence data. <a href="#refs"><span class="citation">Wanamaker et
al. (2022)</span></a> offers a comparison of three methods for
quantifying DNA methylation at single base-pair resolution using whole
genome bisulfite sequencing (WGBS), reduced representation bisulfite
sequencing (RRBS), and methyl-CpG binding domain bisulfite sequencing
(MBDBS). When aligning bisulfite-converted data to a reference genome
the percent methylation at a given CpG loci is determined by examining
the ratio of thymines to cytosines. For example if there is a given
cytosine adjacent to a guanine (ie CpG locus) with 30% of the reads
containing thymine, one would consider this cytosine loci (CpG) to be
70% methylated. Since bisulfite treatment can damage DNA, newer library
preparation and sequencing methods (ex. <a
href="https://www.neb.com/en-us/products/e7120-nebnext-enzymatic-methyl-seq-kit#Protocols,%20Manuals%20&amp;%20Usage">NEBNext®
Enzymatic Methyl-seq Kit; EM-seq</a> are starting to gain popularity. As
EM-Seq produces loci-level data, alignment of EM-Seq data is similar to
WGBS. Also, technologies like Nanopore and PacBio sequencing can
directly detect DNA methylation without bisulfite conversion. See <a
href="#refs"><span class="citation">Dimond, Nguyen, and Roberts
(2021)</span></a> as example of using Nanopore sequencing. As DNA passes
through a nanopore or is sequenced by single-molecule real-time (SMRT)
technology, changes in electrical current or fluorescence patterns can
indicate the presence of methylated bases.</p>
</div>
<div id="experimental-considerations" class="section level2">
<h2>Experimental Considerations</h2>
<div id="sample-size" class="section level3">
<h3>Sample Size</h3>
<p>Sample size is a critical factor when conducting DNA methylation
analysis, particularly because of the significant variation in
methylation patterns both between individuals and across different
tissues. This variation, which is not yet fully understood, can
significantly impact the outcomes and interpretations of methylation
studies.</p>
<p>When selecting samples for DNA methylation analysis, the key
principle is that larger sample sizes generally provide more reliable
and generalizable results. A larger number of samples allows for a more
accurate representation of the population, accounting for individual and
tissue-level differences. This is essential for identifying true
biological patterns rather than artifacts of small, non-representative
samples.</p>
<p>However, the practicality of obtaining a large number of samples must
be balanced with resource constraints. In this context, a sample size of
around eight can be considered a decent starting point. This number is
often sufficient to begin observing trends and patterns in methylation
across individuals and tissues, providing a foundational understanding
which can be built upon with further research.</p>
<p>It’s crucial to remember that while more samples can enhance the
reliability of the results, the quality of the samples and the
robustness of the methylation analysis techniques are equally important.
Therefore, researchers must strive for a balance between quantity and
quality in their sample selection to ensure meaningful and accurate
insights into DNA methylation.</p>
</div>
<div id="sequencing-depth" class="section level3">
<h3>Sequencing Depth</h3>
</div>
<div id="bisulfite-conversion-efficiency" class="section level3">
<h3>Bisulfite Conversion Efficiency</h3>
<p>Bisulfite Conversion Efficiency is important to assess. One common
step during library preparation us spiking library with Lambda phage
DNA. Lambda phage DNA lacks cytosine methylation, making it an ideal
control for bisulfite treatment. Here’s how it works and why it’s
useful:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Background on Lambda Phage DNA</strong>: Lambda phage DNA
is a widely used control in bisulfite sequencing because it is naturally
unmethylated. By including this DNA in your sample, you create a known
unmethylated reference.</p></li>
<li><p><strong>Preparation and Spiking</strong>: You prepare the lambda
phage DNA and spike it into your DNA samples before bisulfite treatment.
The proportion of lambda DNA added should be enough to be detected
post-sequencing, but not so much that it overwhelms the sample. A
typical proportion might be around 0.1% to 1% of the total DNA.</p></li>
<li><p><strong>Bisulfite Treatment and Sequencing</strong>: After
spiking, you proceed with bisulfite treatment and sequencing of the
entire sample, which now includes your genomic DNA of interest and the
spiked lambda DNA.</p></li>
<li><p><strong>Assessing Conversion Efficiency</strong>: Once sequencing
is complete, you analyze the lambda phage DNA sequences. Since this DNA
is unmethylated, all cytosines should be converted to thymines after
bisulfite treatment and subsequent PCR. By calculating the percentage of
cytosines that were successfully converted to thymines in the lambda
DNA, you can determine the bisulfite conversion efficiency. An efficient
conversion typically results in over 99% of cytosines being
converted.</p></li>
</ol>
</div>
</div>
<div id="tutorial-details" class="section level2">
<h2>Tutorial Details</h2>
<p>For a good review of epigenetic mechanisms including DNA methylation
in marine organisms see <a href="#refs"><span
class="citation">Eirin-Lopez and Putnam (2019)</span></a>. <a
href="#refs"><span class="citation">Wanamaker et al. (2022)</span></a>
offers a comparison of three methods for quantifying DNA methylation at
single base-pair resolution using whole genome bisulfite sequencing
(WGBS), reduced representation bisulfite sequencing (RRBS), and
methyl-CpG binding domain bisulfite sequencing (MBDBS). The tutorial
below is based on WGBS, though the general workflow would be
consistent.</p>
<p>The tutorial will cover the following:<br />
- Sequence quality assessment<br />
- Read alignment<br />
- Methylation quantification<br />
- File conversions<br />
- Potential downstream applications. Although this tutorial will not
provide code for downstream applications such as differential
methylation analysis, it will link to other open-access resources and
examples.</p>
<p>The tutorial below is based on WGBS from a species of marine
intertebrate (<em>Montipora capitata</em> coral) examined in <a
href="#refs"><span class="citation">Wanamaker et al. (2022)</span></a>,
and all raw data can be accessed under NCBI Bioproject PRJNA691891 if
you would like to follow the tutorial with this dataset. The general
workflow would be consistent for other library preparation and
sequencing methods that produce loci-level data. The tutorial indicates
places where special analytical consideration should be given to other
data types. Given the size of the data files and the computational
resources necessary this is not developed to be actionable and
reproducible. For context this process outlined below took weeks on a
university HPC system.</p>
</div>
</div>
<div id="sequence-quality" class="section level1">
<h1>Sequence Quality</h1>
<p>Following trimming one would expect to see FastQC per base sequencing
content similar to the plot shown below. This particular data set was
trimmed with the following parameters:</p>
<pre><code>/gscratch/srlab/programs/TrimGalore-0.4.5/trim_galore \
--output_dir /gscratch/scrubbed/strigg/analyses/20200311/WGBS_MBD \
--paired \
--fastqc_args \
--outdir /gscratch/scrubbed/strigg/analyses/20200311/WGBS_MBD/FASTQC \
--threads 28 \
--illumina \
--clip_R1 10 \
--clip_R2 10 \
--three_prime_clip_R1 10 \
--three_prime_clip_R2 10 \
--path_to_cutadapt /gscratch/srlab/programs/miniconda3/bin/cutadapt \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth17_R1_001.fastq.gz \</code></pre>
<p><img src="DRAFT_DNA_methylation/meth-bp-chart.png" width="70%" style="display: block; margin: auto;" /></p>
<p>For WGBS and MBDBS data, it is general practice to hard-trim 10 bp
from the beginning and end of each read. RRBS data often needs different
trimming specifications <a
href="https://felixkrueger.github.io/Bismark/">see Manual for
details</a></p>
</div>
<div id="read-alignment" class="section level1">
<h1>Read Alignment</h1>
<div id="software-considerations" class="section level2">
<h2>Software Considerations</h2>
<p>The software used as part of this tutorial is <strong>Bismark
Bisulfite Mapper</strong> <a href="#refs"><span class="citation">Krueger
and Andrews (2011)</span></a>, which is one of the more widely-used
softwares to align methylation data to a reference genome. Another
commonly-used options include bwa-meth <a href="#refs"><span
class="citation">Pedersen et al. (2014)</span></a> and BSMAP <a
href="#refs"><span class="citation">Xi and Li (2009)</span></a> (see <a
href="#refs"><span class="citation">Nunn et al. (2021)</span></a> and
<span class="citation">(Gong et al. 2022)</span> for comparisons of
these software for plant and mammal species, respectively). As with any
software it is best to be famililar with the <a
href="https://felixkrueger.github.io/Bismark/">User Manual</a>.</p>
<p>Within Bismark, the tutorial uses Bowtie 2 <a href="#refs"><span
class="citation">Langmead and Salzberg (2012)</span></a> as the
alignment software. Although the Bismark user manual covers the
necessary Bowtie 2 parameters, it may be useful to refer to the <a
href="https://bowtie-bio.sourceforge.net/bowtie2/manual.shtml">Bowtie 2
manual</a>.</p>
<p>The content below will provided with the assumption that the reader
has read the manual and is meant to serve as guidance based on
experience working with marine invertebrates.</p>
</div>
<div id="genome-preparation" class="section level2">
<h2>Genome Preparation</h2>
<p>The first step in the process in preparing the reference genome by
performing an in silico bisulfite conversion to allow Bowtie alignments.
Example code:</p>
<pre><code># Directories and programs
bismark_dir=&quot;/programs/Bismark-0.21.0&quot;
bowtie2_dir=&quot;/programs/bowtie2-2.3.4.1-linux-x86_64/&quot;
genome_folder=&quot;/where/the/fastafile/lives/&quot;

${bismark_dir}/bismark_genome_preparation \
--verbose \
--parallel 28 \
--path_to_aligner ${bowtie2_dir} \
${genome_folder}</code></pre>
<ol style="list-style-type: decimal">
<li><p><strong>Bismark Genome Preparation Command</strong>:</p>
<ul>
<li><p><strong><code>${bismark_dir}/bismark_genome_preparation</code></strong>:
This is the command to run the genome preparation part of Bismark. The
<strong><code>${bismark_dir}</code></strong> variable is expanded to the
path where Bismark is installed, so the script knows where to find the
<strong><code>bismark_genome_preparation</code></strong>
program.</p></li>
<li><p><strong><code>--verbose</code></strong>: This flag makes the
program output more detailed information about what it is doing, which
is helpful for debugging or understanding the process.</p></li>
<li><p><strong><code>--parallel 28</code></strong>: This option tells
Bismark to use 28 threads in parallel to speed up the process. You can
adjust this based on your computing power.</p></li>
<li><p><strong><code>--path_to_aligner ${bowtie2_dir}</code></strong>:
This specifies the path to the aligner (Bowtie2) that Bismark will use.
The <strong><code>${bowtie2_dir}</code></strong> variable is expanded to
the path where Bowtie2 is installed.</p></li>
<li><p><strong><code>${genome_folder}</code></strong>: Finally, this
specifies the location of the genome files. The script uses the
<strong><code>${genome_folder}</code></strong> variable, which holds the
path to these files.</p></li>
</ul></li>
</ol>
<p>Bismark will create two individual folders within this directory, one
for a C-&gt;T converted genome and the other one for the G-&gt;A
converted genome. After creating C-&gt;T and G-&gt;A versions of the
genome they will be indexed in parallel usingthe indexer. Bismark will
create the following subdirectories as outputs.</p>
<pre><code>./Bisulfite_Genome
./Bisulfite_Genome/GA_conversion
./Bisulfite_Genome/CT_conversion</code></pre>
<p>This preparation step only needs to happen once for a given
genome.</p>
</div>
<div id="aligning-reads" class="section level2">
<h2>Aligning reads</h2>
<p>Once the genome is prepared, reads can be aligned. For aligning the
trimmed reads to the genome the following code structure is used:</p>
<pre><code>find ${reads_dir}*_R1_001_val_1.fq.gz \
| xargs basename -s _R1_001_val_1.fq.gz | xargs -I{} ${bismark_dir}/bismark \
--path_to_bowtie ${bowtie2_dir} \
-genome ${genome_folder} \
-p 4 \
-score_min L,0,-0.6 \
--non_directional \
-1 ${reads_dir}{}_R1_001_val_1.fq.gz \
-2 ${reads_dir}{}_R2_001_val_2.fq.gz \
-o Mcap_tg</code></pre>
<p>This will create BAM files (sequence alignment files) and txt files
(mapping reports). The reports contain information such as mapping
efficiency.</p>
<p>Specifically this is what the code chunk does:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Finding Files and Preparing Filenames</strong>:</p>
<ul>
<li><p><strong><code>find ${reads_dir}*_R1_001_val_1.fq.gz</code></strong>:
This command uses the <strong><code>find</code></strong> utility to
search for files in the directory specified by the variable
<strong><code>reads_dir</code></strong>. The files it looks for have a
naming pattern ending with
<strong><code>_R1_001_val_1.fq.gz</code></strong>, which typically
denotes the first file of a pair in paired-end sequencing data.</p></li>
<li><p><strong><code>| xargs basename -s _R1_001_val_1.fq.gz</code></strong>:
The <strong><code>xargs</code></strong> command takes the list of files
found and passes them to <strong><code>basename</code></strong>, which
strips the suffix <strong><code>_R1_001_val_1.fq.gz</code></strong> from
each filename. This step extracts the base name of each file, which is
used to pair the R1 and R2 files correctly in the following
steps.</p></li>
</ul></li>
<li><p><strong>Running Bismark for Sequence Alignment</strong>:</p>
<ul>
<li><p><strong><code>| xargs -I{} ${bismark_dir}/bismark</code></strong>:
The <strong><code>xargs</code></strong> command here is used to pass the
base filenames to Bismark for alignment.
<strong><code>{}</code></strong> is a placeholder for each base
filename.</p></li>
<li><p><strong><code>--path_to_bowtie ${bowtie2_dir}</code></strong>:
This option specifies the path to the Bowtie 2 aligner, which Bismark
uses for the alignment process.</p></li>
<li><p><strong><code>-genome ${genome_folder}</code></strong>: Specifies
the directory where the genome files, prepared in advance, are
located.</p></li>
<li><p><strong><code>-p 4</code></strong>: Sets the number of processing
threads to 4, allowing Bismark to perform the alignment in parallel,
which can speed up the process. This parameter can be changed based on
your computing power.</p></li>
<li><p><strong><code>-score_min L,0,-0.6</code></strong>: This sets the
scoring threshold for alignments, impacting the sensitivity and
specificity of the alignment. The default scoring threshold used by
Bismark is L,0,-0.2.</p></li>
<li><p><strong><code>--non_directional</code></strong>: This flag
indicates the sequencing data is non-directional, which is important for
correctly interpreting the methylation status in bisulfite
sequencing.</p></li>
<li><p><strong><code>-1 ${reads_dir}{}_R1_001_val_1.fq.gz -2 ${reads_dir}{}_R2_001_val_2.fq.gz</code></strong>:
Specifies the files for paired-end reads. R1 and R2 files are paired
based on their base filename.</p></li>
<li><p><strong><code>-o Mcap_tg</code></strong>: Sets the output
directory for the alignment results to
<strong><code>Mcap_tg</code></strong>.</p></li>
</ul></li>
</ol>
<div id="optimizing-alignment-sensitivity" class="section level3">
<h3>Optimizing alignment sensitivity</h3>
<p>The code above uses <code>score_min L,0,-0.6</code> as the scoring
threshold for alignment sensitivity and specificity. This is different
than the Bismark default, <code>score_min L,0,-0.2</code>. Reducing the
sensitivity of the alignment can increase the percentage of reads
aligned to the reference genome, but of course the quality of the
alignment may decrease. A good practice is to align a subset of your
data to the reference genome using different <code>score_min</code>
parameters, then comparing mapping efficiency from the mapping reports.
Generally, you want to choose a parameter that maximizes percent
alignment and specificity. Example code</p>
<pre><code>find ${reads_dir}*_R1_001_val_1.fq.gz \
| xargs basename -s _R1_001_val_1.fq.gz | xargs -I{} ${bismark_dir}/bismark \
--path_to_bowtie ${bowtie2_dir} \
-genome ${genome_folder} \
-p 4 \
- u 10000 \
-score_min L,0,-0.6 \
--non_directional \
-1 ${reads_dir}{}_R1_001_val_1.fq.gz \
-2 ${reads_dir}{}_R2_001_val_2.fq.gz \
-o Mcap_tg</code></pre>
<p>This is the same code as above, except <code>-u 10000</code> is used
to run the alignment on a 10,000 bp subset of the data in each input
file.</p>
</div>
</div>
<div id="deduplication" class="section level2">
<h2>Deduplication</h2>
<p>Files are then deduplicated if they are whole genome bisulfite
treated samples. This command will deduplicate the Bismark alignment BAM
file and remove all reads that align to the the very same position and
in the same orientation except for one. This step is recommended for
whole-genome bisulfite samples, but <strong>should not</strong> be used
for reduced representation libraries such as RRBS, amplicon, or target
enrichment libraries. The <a
href="https://rawgit.com/FelixKrueger/Bismark/master/Docs/Bismark_User_Guide.html#ix-notes-about-different-library-types-and-commercial-kits"><code>bismark</code>
manual</a> provides additional information.</p>
<pre><code>find *.bam | \
xargs basename -s .bam | \
xargs -I{} ${bismark_dir}/deduplicate_bismark \
--bam \
--paired \
{}.bam</code></pre>
</div>
</div>
<div id="methylation-quantification" class="section level1">
<h1>Methylation Quantification</h1>
<div id="what-to-expect" class="section level2">
<h2>What to expect</h2>
<p>DNA methylation at CpG locus can vary to 0-100%. How methylated CpGs
are across a genome can vary WIDELY depending on your focal taxa.
Generally, invertebrate genomes are predominantly unmethylated, with
smaller regions of higher methylation in a mosaic pattern. In some
marine invertebrates we see ~15% methylation of CpGs (defined as &gt;
75% methylated) across the genome with most CpGs unmethylated. In these
taxa methylation usually occurs in gene bodies.</p>
<p>The figure below shows the methylation ratio of CpGs across the
Pacific oyster (<em>Crassostrea gigas</em>) genome, as well as the
location of methylation across the genome:</p>
<div class="figure" style="text-align: center">
<img src="DRAFT_DNA_methylation/meth_ratio.png" alt="Frequency distribution of methylation ratios for CpG dinucleotides in oyster gill tissue. [@10.7717/peerj.215](#refs)" width="70%" />
<p class="caption">
Frequency distribution of methylation ratios for CpG dinucleotides in
oyster gill tissue. <a href="#refs"><span class="citation">Gavery and
Roberts (2013)</span></a>
</p>
</div>
<div class="figure" style="text-align: center">
<img src="DRAFT_DNA_methylation/igv.png" alt="Screenshot from IGV. Bedgraph files are displayed. See below for code to create these file (File Conversion)" width="100%" />
<p class="caption">
Screenshot from IGV. Bedgraph files are displayed. See below for code to
create these file (File Conversion)
</p>
</div>
<hr />
<p>Vertebrate genomes, on the other hand, have higher overall levels of
methylation, with methylation occuring in CpG islands. An example
analysis of methylation at individual CG dinucleotides in an elephant
shark, mouse, and zebra fish is shown, below. Results reavel comparable
levels of global methyaltion patterns with the majority of sites being
highly methylated (≥ 80%), and fewer than one tenth being unmethylated
(doi: 10.12688/f1000research.11281.1).</p>
<div class="figure" style="text-align: center">
<img src="DRAFT_DNA_methylation/vert_methyl_distr.png" alt="Genome screenshots of methylation quantified in 2kb running windows over the first 3Mb of chromosome 1 zebrafish and mouse, and of the largest scaffold (NW_006890054.1) in elephant shark (doi: 10.12688/f1000research.11281.1)" width="70%" />
<p class="caption">
Genome screenshots of methylation quantified in 2kb running windows over
the first 3Mb of chromosome 1 zebrafish and mouse, and of the largest
scaffold (NW_006890054.1) in elephant shark (doi:
10.12688/f1000research.11281.1)
</p>
</div>
</div>
<div id="methylation-extraction" class="section level2">
<h2>Methylation extraction</h2>
<p>Methylation levels are then extracted using the
<code>bismark_methylation_extractor</code>. For example:</p>
<pre><code>${bismark_dir}/bismark_methylation_extractor \
--bedGraph \
--counts \
--comprehensive \
--merge_non_CpG \
--multicore 28 \
--buffer_size 75% \
*deduplicated.bam</code></pre>
<p>This particular command is used to extract and process methylation
data from BAM files (generated after sequencing data alignment). Let’s
break down the options:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Options</strong>:</p>
<ul>
<li><p><strong><code>--bedGraph</code></strong>: Generates a BedGraph
file output, which is a format that allows visualization of the
methylation levels across the genome in genome browsers.</p></li>
<li><p><strong><code>--counts</code></strong>: Outputs the count of
methylated and unmethylated calls for each cytosine in the
genome.</p></li>
<li><p><strong><code>--comprehensive</code></strong>: This option makes
the methylation extractor process all cytosines in the genome, not just
those in a CpG context. It combines all three contexts (CpG, CHG, and
CHH) into one output file.</p></li>
<li><p><strong><code>--merge_non_CpG</code></strong>: Merges the
methylation information for CHG and CHH contexts (non-CpG) into a single
file.</p></li>
<li><p><strong><code>--multicore 28</code></strong>: Specifies that the
program should use 28 cores to process the data in parallel, which can
significantly speed up the analysis. This can be adjusted depending on
your computing power.</p></li>
<li><p><strong><code>--buffer_size 75%</code></strong>: This option sets
the buffer size for reading in BAM files to 75% of the available system
memory. It helps optimize the performance of the tool by adjusting how
much memory is used for processing. This can be adjusted depending on
your computing power.</p></li>
<li><p><strong><code>*deduplicated.bam</code></strong>: This is the
input file pattern for the command. It indicates that the methylation
extractor should process all BAM files in the current directory that
have names ending with “deduplicated.bam”. These files are typically the
output of a deduplication process applied to aligned bisulfite
sequencing data.</p></li>
</ul></li>
</ol>
<p>This will create a file with the suffix
<code>deduplicated.bismark.cov.gz</code> (example shown below in
uncompressed format):</p>
<pre><code>NC_035784.1 141 141 37.5    3   5
NC_035784.1 142 142 100 2   0
NC_035784.1 155 155 70  7   3
NC_035784.1 156 156 100 2   0
NC_035784.1 291 291 0   0   2
NC_035784.1 292 292 0   0   3
NC_035784.1 313 313 0   0   1
NC_035784.1 314 314 66.6666666666667    2   1
NC_035784.1 470 470 66.6666666666667    4   2
NC_035784.1 611 611 0   0   4</code></pre>
<p>column organization of the file</p>
<pre><code>&lt;chromosome&gt; &lt;start position&gt; &lt;end position&gt; &lt;methylation percentage&gt; &lt;count methylated&gt; &lt;count unmethylated&gt;</code></pre>
<p>The resultant bedGraph file contains information from
<em>unmerged</em> strands, meaning reads aligning to forward and reverse
strands in the same position are not combined. It is important to merge
the reads from the forward and reserve strands for more accurate
methylation quantification. A genome-wide cytosine report and merged
bedGraph file are generated (including merged bedGraph file) from the
<code>deduplicated.bismark.cov.gz</code> generated above in this example
code:</p>
<pre><code>find *deduplicated.bismark.cov.gz \
| xargs basename -s _trimmed_bismark_bt2.deduplicated.bismark.cov.gz \
| xargs -I{} ${bismark_dir}/coverage2cytosine \
--genome_folder ${genome_folder} \
-o {} \
--merge_CpG \
--zero_based \
{}_trimmed_bismark_bt2.deduplicated.bismark.cov.gz</code></pre>
<p>The script converts coverage files into a format that details
cytosine methylation levels across the genome. Here’s a breakdown of
each step:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Finding Files</strong>:</p>
<ul>
<li><strong><code>find *deduplicated.bismark.cov.gz</code></strong>:
This command is used to search for all files in the current directory
that end with <strong><code>deduplicated.bismark.cov.gz</code></strong>.
These files are coverage files produced by Bismark, typically after
aligning bisulfite-treated DNA sequences. They contain information about
methylation at different sites in the genome.</li>
</ul></li>
<li><p><strong>Extracting Base File Names</strong>:</p>
<ul>
<li><strong><code>| xargs basename -s _trimmed_bismark_bt2.deduplicated.bismark.cov.gz</code></strong>:
This pipeline uses <strong><code>xargs</code></strong> to pass each
found file to <strong><code>basename</code></strong>, which strips the
suffix
<strong><code>_trimmed_bismark_bt2.deduplicated.bismark.cov.gz</code></strong>
from each file name. This step extracts the base names of the files,
which will be used to generate output filenames in the following
steps.</li>
</ul></li>
<li><p><strong>Converting Coverage to Cytosine Methylation
Levels</strong>:</p>
<ul>
<li><p><strong><code>| xargs -I{} ${bismark_dir}/coverage2cytosine</code></strong>:
This pipeline uses <strong><code>xargs</code></strong> again to pass
each base filename to the
<strong><code>coverage2cytosine</code></strong> tool in the Bismark
suite. The placeholder <strong><code>{}</code></strong> is replaced by
the base filename.</p></li>
<li><p><strong><code>--genome_folder ${genome_folder}</code></strong>:
This option specifies the location of the genome folder. The
<strong><code>${genome_folder}</code></strong> variable holds the path
to this folder, which contains reference genome data necessary for the
analysis.</p></li>
<li><p><strong><code>-o {}</code></strong>: The
<strong><code>-o</code></strong> option specifies the output file name,
using the base filename <strong><code>{}</code></strong> obtained from
the previous steps.</p></li>
<li><p><strong><code>--merge_CpG</code></strong>: This flag instructs
the program to merge the methylation information of cytosines in CpG
context, providing a single methylation value for each CpG
site.</p></li>
<li><p><strong><code>--zero_based</code></strong>: This option indicates
that the output file should use zero-based coordinates, which is a
common format in bioinformatics where the first base of a sequence is
numbered 0 instead of 1.</p></li>
<li><p><strong><code>{}_trimmed_bismark_bt2.deduplicated.bismark.cov.gz</code></strong>:
Finally, this specifies the input file for each iteration, where
<strong><code>{}</code></strong> is replaced by the base filename
obtained earlier. These are the coverage files previously identified by
the <strong><code>find</code></strong> command.</p></li>
</ul></li>
</ol>
<p>Output file format <code>.CpG_report.merged_CpG_evidence.cov</code>
(merged bedGraph file)</p>
<pre><code>NC_035785.1 217 219 100.000000  17  0
NC_035785.1 523 525 87.500000   7   1
NC_035785.1 556 558 50.000000   5   5
NC_035785.1 727 729 100.000000  16  0
NC_035785.1 1330    1332    0.000000    0   2
NC_035785.1 1403    1405    0.000000    0   2
NC_035785.1 1494    1496    66.666667   2   1
NC_035785.1 1747    1749    100.000000  8   0
NC_035785.1 2024    2026    100.000000  24  0
NC_035785.1 2054    2056    93.333333   14  1</code></pre>
<p>column organization of the file</p>
<pre><code>&lt;chromosome&gt; &lt;start position&gt; &lt;end position&gt; &lt;methylation percentage&gt; &lt;count methylated&gt; &lt;count unmethylated&gt;</code></pre>
</div>
</div>
<div id="summary-report" class="section level1">
<h1>Summary Report</h1>
<p>Many steps of the Bismark workflow produce reports (.txt files) with
useful information (ex. mapping efficiency, percent cytosine
methylation, etc.). These files can be collated into processing reports
for each individual sample, which can then be further collated into an
<a
href="https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/032120-fds/bismark_summary_report.html">HTML
summary report</a>:</p>
<pre><code>#HTML Processing Report

${bismark_dir}/bismark2report

#Summary Report

${bismark_dir}/bismark2summary</code></pre>
<p>These commands should be run in the same directory where all output
files are stored.</p>
<p><img src="DRAFT_DNA_methylation/bismark-summary-report.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="file-conversions" class="section level1">
<h1>File Conversions</h1>
<p>From here it could be useful to do some file reorganizaion to obtain
bedGraph or tab files for downstream analysis (eg <code>bedtools</code>,
GLMs). Specifically, it may be important to filter the CpG methylation
data to a specific coverage threshold (ex. 5x or 10x). The more read
coverage a CpG has, the higher the confidence in methylation
quantification.</p>
<p>Creating bedGraphs post-merge at a specific coverage threshold (ex.
10x or 5x):</p>
<pre><code>for f in *merged_CpG_evidence.cov
do
  STEM=$(basename &quot;${f}&quot; .CpG_report.merged_CpG_evidence.cov)
  cat &quot;${f}&quot; | awk -F $&#39;\t&#39; &#39;BEGIN {OFS = FS} {if ($5+$6 &gt;= 5) {print $1, $2, $3, $4}}&#39; \
  &gt; &quot;${STEM}&quot;_5x.bedgraph
done</code></pre>
<p>Creating tab files with raw count data for GLMs at 10x and 5x
coverage:</p>
<pre><code>for f in *merged_CpG_evidence.cov
do
  STEM=$(basename &quot;${f}&quot; .CpG_report.merged_CpG_evidence.cov)
  cat &quot;${f}&quot; | awk -F $&#39;\t&#39; &#39;BEGIN {OFS = FS} {if ($5+$6 &gt;= 5) {print $1, $2, $3, $4, $5, $6}}&#39; \
  &gt; &quot;${STEM}&quot;_5x.tab
done</code></pre>
<p>It is sometimes useful to sort BAM files for downstream analysis (eg
methylkit, IGV)</p>
<pre><code># Sort files for methylkit and IGV
find *.bam | \
xargs basename -s .bam | \
xargs -I{} ${samtools} \
sort --threads 28 {}.bam \
-o {}.sorted.bam

# Index sorted files for IGV
find *.sorted.bam | \
xargs basename -s .sorted.bam | \
xargs -I{} ${samtools} \
index -@ 28 {}.sorted.bam</code></pre>
</div>
<div id="next-steps" class="section level1">
<h1>Next Steps</h1>
<p>So you’ve aligned your trimmed reads to your reference genome and
have methylation data for CpGs. Now what?</p>
<div id="removing-c-to-t-c-t-snps" class="section level2">
<h2>Removing C to T (C-&gt;T) SNPs</h2>
<p>Removing cytosine (C) to thymine (T) single nucleotide polymorphisms
(SNPs) in bisulfite sequencing analysis could be something to consider
for a few reasons:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Bisulfite Conversion Process</strong>: In bisulfite
sequencing, unmethylated cytosines are converted to uracil, which is
then read as thymine in sequencing. This conversion is key to
identifying methylated cytosines. However, if a genomic DNA sequence
naturally has a C to T (C-&gt;T) SNP, it can be falsely interpreted as a
converted cytosine, leading to incorrect conclusions about
methylation.</p></li>
<li><p><strong>Reduction of False Positives</strong>: By removing
C-&gt;T SNPs, one can reduce false positives in methylation analysis.
Without removal, these SNPs might be misinterpreted as unmethylated
cytosines that have undergone conversion, skewing the data towards an
apparent increase in methylation.</p></li>
<li><p><strong>Accurate Methylation Mapping</strong>: For precise
mapping of methylation patterns, it’s crucial to distinguish between
true methylation signals and genetic variation (SNPs). Removing C-&gt;T
SNPs allows for a more accurate representation of the methylation status
of the genome.</p></li>
<li><p><strong>Analysis Consistency</strong>: In comparative studies,
where multiple samples or conditions are analyzed, removing C-&gt;T SNPs
ensures consistency across samples. This is especially important in
cases where SNP profiles may vary between samples, which could otherwise
lead to inconsistent methylation data.</p></li>
</ol>
<p>By eliminating these potential sources of error, researchers can have
more confidence in their interpretation of the bisulfite sequencing
data, leading to more reliable and valid conclusions about DNA
methylation patterns.</p>
<p>One way to do this is using BS-Snper (<span class="citation">Gao et
al. (2015)</span>). <a href="https://github.com/hellbelly/BS-Snper"
class="uri">https://github.com/hellbelly/BS-Snper</a>. This program
finds SNPs in Bisulfite treated alignment files. The bash code below
represents steps to identify SNPS and create a list of all C-&gt;T
SNPS.</p>
<pre><code>samtools merge \
Pg_merged.bam \
*.sorted.bam

perl /gscratch/srlab/programs/BS-Snper-master/BS-Snper.pl \
Pg_merged.bam \
--fa Panopea-generosa-v1.0.fa \
--output snp.candidate.out \
--methcg meth.cg \
--methchg meth.chg \
--methchh meth.chh \
--minhetfreq 0.1 \
--minhomfreq 0.85 \
--minquali 15 \
--mincover 10 \
--maxcover 1000 \
--minread2 2 \
--errorate 0.02 \
--mapvalue 20 \
&gt;SNP.vcf 2&gt;SNP.log


grep $&#39;C\tG&#39; output/SNP.vcf &gt; output/CT-SNP.vcf
wc -l output/CT-SNP.vcf</code></pre>
<p>Let’s break down the command and its arguments:</p>
<ol style="list-style-type: decimal">
<li><p><strong><code>perl /gscratch/srlab/programs/BS-Snper-master/BS-Snper.pl</code></strong>:
This part of the command calls the Perl interpreter to execute the
<strong><code>BS-Snper.pl</code></strong> script. The script is located
in the
<strong><code>/gscratch/srlab/programs/BS-Snper-master/</code></strong>
directory.</p></li>
<li><p><strong><code>Pg_merged.bam</code></strong>: This is the input
file for the script, likely a BAM file (Binary Alignment/Map format)
which contains alignment information of bisulfite-treated sequencing
reads.</p></li>
<li><p><strong><code>--fa Panopea-generosa-v1.0.fa</code></strong>: This
option specifies the reference genome file in FASTA format against which
the BAM file will be compared.</p></li>
<li><p><strong><code>--output snp.candidate.out</code></strong>: This
argument sets the name of the output file where the SNP candidates will
be written.</p></li>
<li><p><strong><code>--methcg meth.cg</code>,
<code>--methchg meth.chg</code>,
<code>--methchh meth.chh</code></strong>: These options specify the
output files for methylation analysis for CG, CHG, and CHH contexts,
respectively (where H can be A, C, or T).</p></li>
<li><p><strong><code>--minhetfreq 0.1</code></strong>: Sets the minimum
frequency for heterozygous bases to be considered in SNP calling (here,
10%).</p></li>
<li><p><strong><code>--minhomfreq 0.85</code></strong>: Sets the minimum
frequency for homozygous bases to be considered in SNP calling (here,
85%).</p></li>
<li><p><strong><code>--minquali 15</code></strong>: Sets the minimum
base quality score.</p></li>
<li><p><strong><code>--mincover 10</code>,
<code>--maxcover 1000</code></strong>: These set the minimum and maximum
coverage (number of reads) for a position to be considered in the
analysis.</p></li>
<li><p><strong><code>--minread2 2</code></strong>: The minimum number of
reads required to support a SNP candidate. It could be useful to use the
same values for the minimum read threshold for SNP identification and
for minimum coverage in the methylation data.</p></li>
<li><p><strong><code>--errorate 0.02</code></strong>: Sets the error
rate of sequencing (here, 2%).</p></li>
<li><p><strong><code>--mapvalue 20</code></strong>: Specifies the
mapping quality value, a threshold for considering reads in the
analysis.</p></li>
<li><p><strong><code>&gt;SNP.vcf</code></strong>: Redirects the standard
output to a file named <strong><code>SNP.vcf</code></strong>. This file
will contain the SNP calls in VCF (Variant Call Format) format.</p></li>
<li><p><strong><code>2&gt;SNP.log</code></strong>: Redirects the
standard error output to a file named
<strong><code>SNP.log</code></strong>. This file will contain error
messages and possibly additional log information.</p></li>
</ol>
<p>The resulting could be used to mask the genome and the entire process
described above could be repeated (ie Genome Preparation, Alignment,
Quantification). An alternative and common practice is to remove
cytosines from downstream analysis by removing loci in intermediate
files (ie tab-delimited data files). Below is an example using
<code>antijoin</code> to remove loci that correspond to CT SNPs from 5x
coverage methyation quantification files.</p>
<pre class="r"><code># Read in CT SNP file
ct &lt;- read.csv(&quot;../output/CT-SNP.vcf&quot;, header = FALSE, sep = &quot;\t&quot;) %&gt;%
  mutate(loci = paste0(V1, &quot;_&quot;, V2))


# 1. List all files with _5x.tab suffix
files &lt;- list.files(path = &quot;../data/&quot;, pattern = &quot;_5x.tab$&quot;, full.names = TRUE)

# 2. Iterate over each file
for(file in files) {
  
  # Extract base filename without the directory for naming purposes
  base_name &lt;- basename(file)
  
  # Read the file
  data &lt;- read.csv(file, header = FALSE, sep = &quot;\t&quot;)
  
  # Modify the data
  modified_data &lt;- data %&gt;%
    mutate(loci = paste0(V1, &quot;_&quot;, V2)) %&gt;%
    anti_join(ct, by = &quot;loci&quot;) %&gt;%
    select(-loci)
    
  
  # Write the modified data to an output file
  output_file &lt;- paste0(&quot;../output/f&quot;, base_name)
  write.table(modified_data, file = output_file, sep = &quot;\t&quot;, row.names = FALSE, quote = FALSE, col.names = FALSE)
}</code></pre>
<p>Let’s break down what each part of the script does:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Reading in a CT SNP file</strong>:</p>
<ul>
<li><p><strong><code>ct &lt;- read.csv("../output/CT-SNP.vcf", header = FALSE, sep = "\t")</code></strong>:
This line reads a file named <strong><code>CT-SNP.vcf</code></strong>
from the specified directory. The file is expected to be in a
tab-separated format (<strong><code>sep = "\t"</code></strong>), and it
doesn’t have a header row
(<strong><code>header = FALSE</code></strong>).</p></li>
<li><p><strong><code>mutate(loci = paste0(V1, "_", V2))</code></strong>:
The <strong><code>mutate</code></strong> function from the
<strong><code>dplyr</code></strong> package is used to add a new column
(<strong><code>loci</code></strong>) to the
<strong><code>ct</code></strong> data frame. This column is created by
concatenating the values in the first (<strong><code>V1</code></strong>)
and second columns (<strong><code>V2</code></strong>) of each row,
separated by an underscore.</p></li>
</ul></li>
<li><p><strong>Listing files with a specific suffix</strong>:</p>
<ul>
<li><strong><code>files &lt;- list.files(path = "../data/", pattern = "_5x.tab$", full.names = TRUE)</code></strong>:
This line generates a list of file paths in the
<strong><code>../data/</code></strong> directory. It only includes files
that end with <strong><code>_5x.tab</code></strong> (as indicated by the
<strong><code>pattern</code></strong> argument).</li>
</ul></li>
<li><p><strong>Processing each file</strong>:</p>
<ul>
<li><p>The <strong><code>for(file in files)</code></strong> loop
iterates over each file found in the previous step.</p>
<ul>
<li><p><strong><code>base_name &lt;- basename(file)</code></strong>:
Extracts the file name without the directory path.</p></li>
<li><p><strong><code>data &lt;- read.csv(file, header = FALSE, sep = "\t")</code></strong>:
Reads the current file in the loop. Similar to the first
<strong><code>read.csv</code></strong> call, it expects a tab-separated
file with no header row.</p></li>
<li><p>The <strong><code>mutate</code></strong> and
<strong><code>anti_join</code></strong> functions are used to modify
<strong><code>data</code></strong>:</p>
<ul>
<li><p><strong><code>mutate(loci = paste0(V1, "_", V2))</code></strong>:
Adds a <strong><code>loci</code></strong> column similar to the
<strong><code>ct</code></strong> data frame.</p></li>
<li><p><strong><code>anti_join(ct, by = "loci")</code></strong>: Removes
rows from <strong><code>data</code></strong> that have matching
<strong><code>loci</code></strong> values in the
<strong><code>ct</code></strong> data frame.</p></li>
<li><p><strong><code>select(-loci)</code></strong>: Removes the
<strong><code>loci</code></strong> column from the data.</p></li>
</ul></li>
<li><p>The modified data is then written to an output file:</p>
<ul>
<li><p><strong><code>output_file &lt;- paste0("../output/f", base_name)</code></strong>:
Constructs the output file path.</p></li>
<li><p><strong><code>write.table(modified_data, file = output_file, sep = "\t", row.names = FALSE, quote = FALSE, col.names = FALSE)</code></strong>:
Writes the <strong><code>modified_data</code></strong> to the specified
file. The output is tab-separated, and it doesn’t include row names,
quotes, or column names.</p></li>
</ul></li>
</ul></li>
</ul></li>
</ol>
</div>
<div id="downstream-application" class="section level2">
<h2>Downstream application</h2>
<ul>
<li><p>Integrating with population genetic data. example: <a
href="#refs"><span class="citation">Silliman et al.
(2023)</span></a></p></li>
<li><p>Gene-level methylation analysis: example <a href="#refs"><span
class="citation">Putnam et al. (2022)</span></a>. It is worth
considering the utility of converting loci-level resolution data to
gene-level resolution data.</p></li>
<li><p>Differential loci methylation and/or region analyses. example: <a
href="#refs"><span class="citation">Venkataraman, White, and Roberts
(2022)</span></a>. methylKit is a popular package to conduct this
analysis differential methylation anlaysis</p></li>
<li><p>Feature overlap. example: <a href="#refs"><span
class="citation">Venkataraman et al. (2020)</span></a> Bedtools is
excellent for quantitative assessment of feature overlap and IGV is good
for visual assessment.</p></li>
</ul>
</div>
</div>
<div id="refs" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-10.1093/g3journal/jkab148" class="csl-entry">
Dimond, J. L., N. Nguyen, and S. J. Roberts. 2021. <span>“Dna
Methylation Profiling of a Cnidarian-Algal Symbiosis Using Nanopore
Sequencing.”</span> <em>G3 Genes|Genomes|Genetics</em> 11. <a
href="https://doi.org/10.1093/g3journal/jkab148">https://doi.org/10.1093/g3journal/jkab148</a>.
</div>
<div id="ref-doi:10.1146/annurev-marine-010318-095114"
class="csl-entry">
Eirin-Lopez, Jose M., and Hollie M. Putnam. 2019. <span>“Marine
Environmental Epigenetics.”</span> <em>Annual Review of Marine
Science</em> 11 (1): 335–68. <a
href="https://doi.org/10.1146/annurev-marine-010318-095114">https://doi.org/10.1146/annurev-marine-010318-095114</a>.
</div>
<div id="ref-10.1093/bioinformatics/btv507" class="csl-entry">
Gao, Shengjie, Dan Zou, Likai Mao, Huayu Liu, Pengfei Song, Youguo Chen,
Shancen Zhao, et al. 2015. <span>“<span class="nocase">BS-SNPer: SNP
calling in bisulfite-seq data</span>.”</span> <em>Bioinformatics</em> 31
(24): 4006–8. <a
href="https://doi.org/10.1093/bioinformatics/btv507">https://doi.org/10.1093/bioinformatics/btv507</a>.
</div>
<div id="ref-10.7717/peerj.215" class="csl-entry">
Gavery, Mackenzie R., and Steven B. Roberts. 2013. <span>“Predominant
Intragenic Methylation Is Associated with Gene Expression
Characteristics in a Bivalve Mollusc.”</span> <em>PeerJ</em> 1
(November): e215. <a
href="https://doi.org/10.7717/peerj.215">https://doi.org/10.7717/peerj.215</a>.
</div>
<div id="ref-10.1016/j.csbj.2022.08.051" class="csl-entry">
Gong, W., X. Pan, D. Xu, G. Ji, Y. Wang, Y. Tian, J. Cai, J. Li, Z.
Zhang, and X. Yuan. 2022. <span>“Benchmarking Dna Methylation Analysis
of 14 Alignment Algorithms for Whole Genome Bisulfite Sequencing in
Mammals.”</span> <em>Computational and Structural Biotechnology
Journal</em> 20: 4704–16. <a
href="https://doi.org/10.1016/j.csbj.2022.08.051">https://doi.org/10.1016/j.csbj.2022.08.051</a>.
</div>
<div id="ref-10.1093/bioinformatics/btr167" class="csl-entry">
Krueger, Felix, and Simon R. Andrews. 2011. <span>“<span
class="nocase">Bismark: a flexible aligner and methylation caller for
Bisulfite-Seq applications</span>.”</span> <em>Bioinformatics</em> 27
(11): 1571–72. <a
href="https://doi.org/10.1093/bioinformatics/btr167">https://doi.org/10.1093/bioinformatics/btr167</a>.
</div>
<div id="ref-10.1038/nmeth.1923" class="csl-entry">
Langmead, B., and S. L. Salzberg. 2012. <span>“Fast Gapped-Read
Alignment with Bowtie 2.”</span> <em>Nature Methods</em> 9: 357–59. <a
href="https://doi.org/10.1038/nmeth.1923">https://doi.org/10.1038/nmeth.1923</a>.
</div>
<div id="ref-10.1093/bib/bbab021" class="csl-entry">
Nunn, A., C. Otto, P. F. Stadler, and D. Langenberger. 2021.
<span>“Comprehensive Benchmarking of Software for Mapping Whole Genome
Bisulfite Data: From Read Alignment to Dna Methylation Analysis.”</span>
<em>Briefings in Bioinformatics</em> 22. <a
href="https://doi.org/10.1093/bib/bbab021">https://doi.org/10.1093/bib/bbab021</a>.
</div>
<div id="ref-pedersen2014fast" class="csl-entry">
Pedersen, Brent S., Kenneth Eyring, Subhajyoti De, Ivana V. Yang, and
David A. Schwartz. 2014. <span>“Fast and Accurate Alignment of Long
Bisulfite-Seq Reads.”</span> <a
href="https://arxiv.org/abs/1401.1129">https://arxiv.org/abs/1401.1129</a>.
</div>
<div id="ref-10.1101/2022.06.24.497506" class="csl-entry">
Putnam, H. M., S. A. Trigg, S. J. White, L. H. Spencer, B. Vadopalas, A.
Natarajan, J. Hetzel, et al. 2022. <span>“Dynamic Dna Methylation
Contributes to Carryover Effects and Beneficial Acclimatization in
Geoduck Clams.”</span> <a
href="https://doi.org/10.1101/2022.06.24.497506">https://doi.org/10.1101/2022.06.24.497506</a>.
</div>
<div id="ref-10.1093/gbe/evad013" class="csl-entry">
Silliman, K., L. H. Spencer, S. J. White, and S. B. Roberts. 2023.
<span>“Epigenetic and Genetic Population Structure Is Coupled in a
Marine Invertebrate.”</span> <em>Genome Biology and Evolution</em> 15.
<a
href="https://doi.org/10.1093/gbe/evad013">https://doi.org/10.1093/gbe/evad013</a>.
</div>
<div id="ref-10.3389/fmars.2020.00225" class="csl-entry">
Venkataraman, Y. R., A. M. Downey-Wall, J. B. Ries, I. Westfield, S. J.
White, S. B. Roberts, and K. E. Lotterhos. 2020. <span>“General Dna
Methylation Patterns and Environmentally-Induced Differential
Methylation in the Eastern Oyster (Crassostrea Virginica).”</span>
<em>Frontiers in Marine Science</em> 7. <a
href="https://doi.org/10.3389/fmars.2020.00225">https://doi.org/10.3389/fmars.2020.00225</a>.
</div>
<div id="ref-10.1186/s12864-022-08781-5" class="csl-entry">
Venkataraman, Y. R., S. J. White, and S. B. Roberts. 2022.
<span>“Differential Dna Methylation in Pacific Oyster Reproductive
Tissue in Response to Ocean Acidification.”</span> <em>BMC Genomics</em>
23. <a
href="https://doi.org/10.1186/s12864-022-08781-5">https://doi.org/10.1186/s12864-022-08781-5</a>.
</div>
<div id="ref-https://doi.org/10.1111/1755-0998.13542" class="csl-entry">
Wanamaker, Shelly A., Yaamini R. Venkataraman, Mackenzie R. Gavery,
Steven B. Roberts, Debashish Bhattacharya, Alan Downey-Wall, Jose M.
Eirin-Lopez, et al. 2022. <span>“Invertebrate Methylomes Provide Insight
into Mechanisms of Environmental Tolerance and Reveal Methodological
Biases.”</span> <em>Molecular Ecology Resources</em> 22 (4): 1247–61. <a
href="https://doi.org/10.1111/1755-0998.13542">https://doi.org/10.1111/1755-0998.13542</a>.
</div>
<div id="ref-10.1186/1471-2105-10-232" class="csl-entry">
Xi, Y., and W. Li. 2009. <span>“Bsmap: Whole Genome Bisulfite Sequence
Mapping Program.”</span> <em>BMC Bioinformatics</em> 10. <a
href="https://doi.org/10.1186/1471-2105-10-232">https://doi.org/10.1186/1471-2105-10-232</a>.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
