```{r,message = FALSE,echo = FALSE}
library(knitr)
knitr::opts_chunk$set(echo = TRUE)
library(knitcitations)
library(kableExtra)
opts_chunk$set(fig.width = 10,
               fig.height = 5,
               cache = FALSE)
cite_options(citation_format = "pandoc", max.names = 3, style = "html", 
             hyperlink = "to.doc")
```

---
title: 'DNA Methylation Assessment'
subtitle: '*Yaamini Venkataraman*, *Sam White*, and *Steven Roberts*'
bibliography: DRAFT_DNA_methylation/DNA-Methylation.bib
---

Initial publication year: TBD <br> [How to cite](https://marineomics.github.io/#How_to_Cite)

# Introduction

Thanks to advances in sample preparation and sequencing methods, there is an influx of research examining epigenomics in non-model systems. Broadly, epigenetics can be defined as changes to gene expression that do not arise from changes in the DNA sequence. DNA methylation, or addition of a methyl (CH<sub>3</sub>) group to a cytosine base adjacent to a guanine (CpG) is one of the more commonly studies epigenetic mechanisms, partly because of its environmental sensitivity and potential role in phenotypic plasticity. For a good review of epigenetic mechanisms including DNA methylation in marine organisms see @doi:10.1146/annurev-marine-010318-095114.

The purpose of this tutorial is to provide an example of aligning bisulfite-treated and enzymatically converted DNA sequence data and discuss needs for potential downstream applications. The principle behind creating bisulfite-treated DNA libraries is that when cytosines are not methylated, then they will be converted to uracil, which ultimately result in thymine nucleotides in the sequence data. @https://doi.org/10.1111/1755-0998.13542 offers a comparison of three methods for quantifying DNA methylation at single base-pair resolution using whole genome bisulfite sequencing (WGBS), reduced representation bisulfite sequencing (RRBS), and methyl-CpG binding domain bisulfite sequencing (MBDBS). When aligning bisulfite-converted data to a reference genome the percent methylation at a given CpG loci is determined by examining the ratio of thymines to cytosines. For example if there is a given cytosine adjacent to a guanine (ie CpG locus) with 30% of the reads containing thymine, one would consider this cytosine loci (CpG) to be 70% methylated. Since bisulfite treatment can damage DNA, newer library preparation and sequencing methods (ex. [NEBNextÂ® Enzymatic Methyl-seq Kit; EM-seq](https://www.neb.com/en-us/products/e7120-nebnext-enzymatic-methyl-seq-kit#Protocols,%20Manuals%20&%20Usage) are starting to gain popularity. As EM-Seq produces loci-level data, alignment of EM-Seq data is similar to WGBS.

For a good review of epigenetic mechanisms including DNA methylation in marine organisms see @doi:10.1146/annurev-marine-010318-095114. @https://doi.org/10.1111/1755-0998.13542 offers a comparison of three methods for quantifying DNA methylation at single base-pair resolution using whole genome bisulfite sequencing (WGBS), reduced representation bisulfite sequencing (RRBS), and methyl-CpG binding domain bisulfite sequencing (MBDBS). The tutorial below is based on WGBS, though the general workflow would be consistent.

The tutorial will cover the following:\
- Sequence quality assessment\
- Read alignment\
- Methylation quantification\
- File conversions\
- Potential downstream applications. Although this tutorial will not provide code for downstream applications such as differential methylation analysis, it will link to other open-access resources and examples.

The tutorial below is based on WGBS from a species of marine intertebrate (*Montipora capitata* coral) examined in @https://doi.org/10.1111/1755-0998.13542, and all raw data can be accessed under NCBI Bioproject PRJNA691891 if you would like to follow the tutorial with this dataset. The general workflow would be consistent for other library preparation and sequencing methods that produce loci-level data. The tutorial indicates places where special analytical consideration should be given to other data types.

# Sequence Quality

Following trimming one would expect to see FastQC per base sequencing content similar to the plot shown below. This particular data set was trimmed with the following parameters:

```         
/gscratch/srlab/programs/TrimGalore-0.4.5/trim_galore \
--output_dir /gscratch/scrubbed/strigg/analyses/20200311/WGBS_MBD \
--paired \
--fastqc_args \
--outdir /gscratch/scrubbed/strigg/analyses/20200311/WGBS_MBD/FASTQC \
--threads 28 \
--illumina \
--clip_R1 10 \
--clip_R2 10 \
--three_prime_clip_R1 10 \
--three_prime_clip_R2 10 \
--path_to_cutadapt /gscratch/srlab/programs/miniconda3/bin/cutadapt \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth17_R1_001.fastq.gz \
```

```{r meth-bp, echo = FALSE, out.width = "70%", fig.align = "center"}
knitr::include_graphics("DRAFT_DNA_methylation/meth-bp-chart.png")
```

For WGBS and MBDBS data, it is general practice to hard-trim 10 bp from the beginning and end of each read. [RRBS data often needs different trimming specifications](https://rawgit.com/FelixKrueger/Bismark/master/Docs/Bismark_User_Guide.html#ix-notes-about-different-library-types-and-commercial-kits).

# Read Alignment

## Considerations

The software used as part of this tutorial is **Bismark Bisulfite Mapper** @10.1093/bioinformatics/btr167. As with any software it is best to be famililar with the [User Manual](https://rawgit.com/FelixKrueger/Bismark/master/Docs/Bismark_User_Guide.html). The content below will provided with the assumption that the reader has read the manual and is meant to serve as guidance based on experience working with marine invertebrates.

The first step in the process in preparing the genome. Example code:

```         
# Directories and programs
bismark_dir="/programs/Bismark-0.21.0"
bowtie2_dir="/programs/bowtie2-2.3.4.1-linux-x86_64/"
genome_folder="/where/the/fastafile/lives/"

${bismark_dir}/bismark_genome_preparation \
--verbose \
--parallel 28 \
--path_to_aligner ${bowtie2_dir} \
${genome_folder}
```

1.  **Bismark Genome Preparation Command**:

    -   **`${bismark_dir}/bismark_genome_preparation`**: This is the command to run the genome preparation part of Bismark. The **`${bismark_dir}`** variable is expanded to the path where Bismark is installed, so the script knows where to find the **`bismark_genome_preparation`** program.

    -   **`--verbose`**: This flag makes the program output more detailed information about what it is doing, which is helpful for debugging or understanding the process.

    -   **`--parallel 28`**: This option tells Bismark to use 28 threads in parallel to speed up the process.

    -   **`--path_to_aligner ${bowtie2_dir}`**: This specifies the path to the aligner (Bowtie2) that Bismark will use. The **`${bowtie2_dir}`** variable is expanded to the path where Bowtie2 is installed.

    -   **`${genome_folder}`**: Finally, this specifies the location of the genome files. The script uses the **`${genome_folder}`** variable, which holds the path to these files.

Bismark will create two individual folders within this directory, one for a C-\>T converted genome and the other one for the G-\>A converted genome. After creating C-\>T and G-\>A versions of the genome they will be indexed in parallel usingthe indexer. Bismark will create the following subdirectories as outputs.

```         
./Bisulfite_Genome
./Bisulfite_Genome/GA_conversion
./Bisulfite_Genome/CT_conversion
```

For aligning the trimmed reads to the genome the following code structure is used

```         
find ${reads_dir}*_R1_001_val_1.fq.gz \
| xargs basename -s _R1_001_val_1.fq.gz | xargs -I{} ${bismark_dir}/bismark \
--path_to_bowtie ${bowtie2_dir} \
-genome ${genome_folder} \
-p 4 \
-score_min L,0,-0.6 \
--non_directional \
-1 ${reads_dir}{}_R1_001_val_1.fq.gz \
-2 ${reads_dir}{}_R2_001_val_2.fq.gz \
-o Mcap_tg
```

This will create BAM files (sequence alignment files)

Specifically this is what the code chunk does:

1.  **Finding Files and Preparing Filenames**:

    -   **`find ${reads_dir}*_R1_001_val_1.fq.gz`**: This command uses the **`find`** utility to search for files in the directory specified by the variable **`reads_dir`**. The files it looks for have a naming pattern ending with **`_R1_001_val_1.fq.gz`**, which typically denotes the first file of a pair in paired-end sequencing data.

    -   **`| xargs basename -s _R1_001_val_1.fq.gz`**: The **`xargs`** command takes the list of files found and passes them to **`basename`**, which strips the suffix **`_R1_001_val_1.fq.gz`** from each filename. This step extracts the base name of each file, which is used to pair the R1 and R2 files correctly in the following steps.

2.  **Running Bismark for Sequence Alignment**:

    -   **`| xargs -I{} ${bismark_dir}/bismark`**: The **`xargs`** command here is used to pass the base filenames to Bismark for alignment. **`{}`** is a placeholder for each base filename.

    -   **`--path_to_bowtie ${bowtie2_dir}`**: This option specifies the path to the Bowtie2 aligner, which Bismark uses for the alignment process.

    -   **`-genome ${genome_folder}`**: Specifies the directory where the genome files, prepared in advance, are located.

    -   **`-p 4`**: Sets the number of processing threads to 4, allowing Bismark to perform the alignment in parallel, which can speed up the process.

    -   **`-score_min L,0,-0.6`**: This sets the scoring threshold for alignments, impacting the sensitivity and specificity of the alignment.

    -   **`--non_directional`**: This flag indicates the sequencing data is non-directional, which is important for correctly interpreting the methylation status in bisulfite sequencing.

    -   **`-1 ${reads_dir}{}_R1_001_val_1.fq.gz -2 ${reads_dir}{}_R2_001_val_2.fq.gz`**: Specifies the files for paired-end reads. R1 and R2 files are paired based on their base filename.

    -   **`-o Mcap_tg`**: Sets the output directory for the alignment results to **`Mcap_tg`**.

Files are then deduplicated if they are whole genome bisulfite treated samples. This command will deduplicate the Bismark alignment BAM file and remove all reads that align to the the very same position and in the same orientation except for one. This step is recommended for whole-genome bisulfite samples, but **should not** be used for reduced representation libraries such as RRBS, amplicon, or target enrichment libraries. The [`bismark` manual](https://rawgit.com/FelixKrueger/Bismark/master/Docs/Bismark_User_Guide.html#ix-notes-about-different-library-types-and-commercial-kits) provides additional information.

```         
find *.bam | \
xargs basename -s .bam | \
xargs -I{} ${bismark_dir}/deduplicate_bismark \
--bam \
--paired \
{}.bam
```

# Methylation Quantification

## What to expect

DNA methylation at CpG locus can vary to 0-100%. Like wise methylation at the gene level can vary and is commonly established as mean methylation level of CpG across gene, assuming some majority degree of gene coverage. Similarly with considering the methylation level of a given loci often time some threshold might be set as seen in code below. From an taxa perspective it can vary WIDELY on what to expect. The figure below shows the mehthylation ratio of CpGs across the oyster genome. In some marine invertebrates we see \~15% methylation of CpGs (defined as \> 75% methylated) across the genome with most CpGs unmethylated. In these taxa methylation usually occurs in gene bodies.

```{r meth-ratio, echo = FALSE, out.width = "70%", fig.align = "center"}
knitr::include_graphics("DRAFT_DNA_methylation/meth_ratio.png")
```

*Frequency distribution of methylation ratios for CpG dinucleotides in oyster gill tissue.* @10.7717/peerj.215

Location of methylation across the genome varies across taxa. Here is what you might expect to see in a marine invertebrate.

```{r meth-igv, echo = FALSE, out.width = "70%", fig.align = "center"}
knitr::include_graphics("DRAFT_DNA_methylation/igv.png")
```

*Screenshot from IGV. Bedgraph files are displayed. See below for code to create these file (File Conversion)*

Methylation levels are then extracted using the `bismark_methylation_extractor`. For example:

```         
${bismark_dir}/bismark_methylation_extractor \
--bedGraph \
--counts \
--comprehensive \
--merge_non_CpG \
--multicore 28 \
--buffer_size 75% \
*deduplicated.bam
```

This particular command is used to extract and process methylation data from BAM files (generated after sequencing data alignment). Let's break down the options:

1.  **Options**:

    -   **`--bedGraph`**: Generates a BedGraph file output, which is a format that allows visualization of the methylation levels across the genome in genome browsers.

    -   **`--counts`**: Outputs the count of methylated and unmethylated calls for each cytosine in the genome.

    -   **`--comprehensive`**: This option makes the methylation extractor process all cytosines in the genome, not just those in a CpG context. It combines all three contexts (CpG, CHG, and CHH) into one output file.

    -   **`--merge_non_CpG`**: Merges the methylation information for CHG and CHH contexts (non-CpG) into a single file.

    -   **`--multicore 28`**: Specifies that the program should use 28 cores to process the data in parallel, which can significantly speed up the analysis.

    -   **`--buffer_size 75%`**: This option sets the buffer size for reading in BAM files to 75% of the available system memory. It helps optimize the performance of the tool by adjusting how much memory is used for processing.

    -   **`*deduplicated.bam`**: This is the input file pattern for the command. It indicates that the methylation extractor should process all BAM files in the current directory that have names ending with "deduplicated.bam". These files are typically the output of a deduplication process applied to aligned bisulfite sequencing data.

This will create a file with the suffix `deduplicated.bismark.cov.gz` (example shown below in uncompressed format). The resultant bedGraph file contains information from *unmerged* strands, meaning reads aligning to forward and reverse strands in the same position are not combined. Later in this workflow, a bedGraph with *merged* information will be created.

```         
NC_035784.1 141 141 37.5    3   5
NC_035784.1 142 142 100 2   0
NC_035784.1 155 155 70  7   3
NC_035784.1 156 156 100 2   0
NC_035784.1 291 291 0   0   2
NC_035784.1 292 292 0   0   3
NC_035784.1 313 313 0   0   1
NC_035784.1 314 314 66.6666666666667    2   1
NC_035784.1 470 470 66.6666666666667    4   2
NC_035784.1 611 611 0   0   4
```

column organization of the file

```         
<chromosome> <start position> <end position> <methylation percentage> <count methylated> <count unmethylated>
```

A genome-wide cytosine report is generated (including merged bedGraph file), with this code as an example:

```         
find *deduplicated.bismark.cov.gz \
| xargs basename -s _trimmed_bismark_bt2.deduplicated.bismark.cov.gz \
| xargs -I{} ${bismark_dir}/coverage2cytosine \
--genome_folder ${genome_folder} \
-o {} \
--merge_CpG \
--zero_based \
{}_trimmed_bismark_bt2.deduplicated.bismark.cov.gz
```

The script converts coverage files into a format that details cytosine methylation levels across the genome. Here's a breakdown of each step:

1.  **Finding Files**:

    -   **`find *deduplicated.bismark.cov.gz`**: This command is used to search for all files in the current directory that end with **`deduplicated.bismark.cov.gz`**. These files are coverage files produced by Bismark, typically after aligning bisulfite-treated DNA sequences. They contain information about methylation at different sites in the genome.

2.  **Extracting Base File Names**:

    -   **`| xargs basename -s _trimmed_bismark_bt2.deduplicated.bismark.cov.gz`**: This pipeline uses **`xargs`** to pass each found file to **`basename`**, which strips the suffix **`_trimmed_bismark_bt2.deduplicated.bismark.cov.gz`** from each file name. This step extracts the base names of the files, which will be used to generate output filenames in the following steps.

3.  **Converting Coverage to Cytosine Methylation Levels**:

    -   **`| xargs -I{} ${bismark_dir}/coverage2cytosine`**: This pipeline uses **`xargs`** again to pass each base filename to the **`coverage2cytosine`** tool in the Bismark suite. The placeholder **`{}`** is replaced by the base filename.

    -   **`--genome_folder ${genome_folder}`**: This option specifies the location of the genome folder. The **`${genome_folder}`** variable holds the path to this folder, which contains reference genome data necessary for the analysis.

    -   **`-o {}`**: The **`-o`** option specifies the output file name, using the base filename **`{}`** obtained from the previous steps.

    -   **`--merge_CpG`**: This flag instructs the program to merge the methylation information of cytosines in CpG context, providing a single methylation value for each CpG site.

    -   **`--zero_based`**: This option indicates that the output file should use zero-based coordinates, which is a common format in bioinformatics where the first base of a sequence is numbered 0 instead of 1.

    -   **`{}_trimmed_bismark_bt2.deduplicated.bismark.cov.gz`**: Finally, this specifies the input file for each iteration, where **`{}`** is replaced by the base filename obtained earlier. These are the coverage files previously identified by the **`find`** command.

Output file format `.CpG_report.merged_CpG_evidence.cov` (merged bedGraph file)

```         
NC_035785.1 217 219 100.000000  17  0
NC_035785.1 523 525 87.500000   7   1
NC_035785.1 556 558 50.000000   5   5
NC_035785.1 727 729 100.000000  16  0
NC_035785.1 1330    1332    0.000000    0   2
NC_035785.1 1403    1405    0.000000    0   2
NC_035785.1 1494    1496    66.666667   2   1
NC_035785.1 1747    1749    100.000000  8   0
NC_035785.1 2024    2026    100.000000  24  0
NC_035785.1 2054    2056    93.333333   14  1
```

# File Conversions

From here it could be useful to do some simple file reorganizaion to obtain bedGraph or tab files for downstream analysis (eg bedtools, GLMs).

Creating bedGraphs post-merge at a specific coverage threshold (ex. 10x or 5x):

```         
for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 10) {print $1, $2, $3, $4}}' \
  > "${STEM}"_10x.bedgraph
done

for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 5) {print $1, $2, $3, $4}}' \
  > "${STEM}"_5x.bedgraph
done
```

Creating tab files with raw count data for GLMs at 10x and 5x coverage:

```         
for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 10) {print $1, $2, $3, $4, $5, $6}}' \
  > "${STEM}"_10x.tab
done

for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 5) {print $1, $2, $3, $4, $5, $6}}' \
  > "${STEM}"_5x.tab
done
```

It is sometimes useful to sort BAM files for downstream analysis (eg methylkit, IGV)

```         
# Sort files for methylkit and IGV
find *.bam | \
xargs basename -s .bam | \
xargs -I{} ${samtools} \
sort --threads 28 {}.bam \
-o {}.sorted.bam

# Index sorted files for IGV
find *.sorted.bam | \
xargs basename -s .sorted.bam | \
xargs -I{} ${samtools} \
index -@ 28 {}.sorted.bam
```

# Downstream applications

-   Gene-level methylation analysis: people do this, but worth considering the utility of converting loci-level resolution data to gene-level resolution data
-   methylKit
-   IGV
-   Analysis with other datasets (ex. pop gen, RNA-Seq)

# References
