<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sara JS Wuitchik" />


<title>FASTQ -&gt; VCF workflow for WGS data</title>

<script src="site_libs/header-attrs-2.10/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="tutorial.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MarineOmics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="panels.html">Panel Seminars</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Population Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="choosing-population-genetics.html">Choosing a Population Genomics Approach</a>
    </li>
    <li>
      <a href="RADseq.html">RADseq</a>
    </li>
    <li>
      <a href="WGS_intro.html">Whole Genome Resequencing</a>
    </li>
    <li>
      <a href="poolseq.html">Poolseq</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Functional Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RNA_coming.html">RNAseq (coming soon!)</a>
    </li>
  </ul>
</li>
<li>
  <a href="https://github.com/MarineOmics/marineomics.github.io/discussions">Discussion Forum</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">FASTQ -&gt; VCF workflow for WGS data</h1>
<h4 class="author">Sara JS Wuitchik</h4>

</div>


<div id="fastq2bam" class="section level1">
<h1>fastq2bam</h1>
<p>This is a more fleshed out explanation of the pipeline developed and maintained by <a href="https://github.com/harvardinformatics">Harvard Informatics</a> in the <a href="https://github.com/harvardinformatics/shortRead_mapping_variantCalling">shortRead_mapping_variantCalling repo</a></p>
<div id="trimming-adapters" class="section level3">
<h3>Trimming adapters</h3>
<p>Before aligning your sequences to a reference genome, you will need to trim off the adapters. We use FastP in our pipeline as below, but there are other programs that could be suitable (e.g., Trimmomatic, TrimGalore/Cutadapt, BBDuk).</p>
<pre class="bash"><code>fastp --in1 {r1} --in2 {r2} --out1 {r1} --out2 {r2} --detect_adapter_for_pe 2&gt; {summary.out}</code></pre>
<p>Where <br> <code>--in</code> and <code>--out</code> are the flags for the input and output reads <br> <code>--detect_adapter_for_pe</code> enables auto-detection of adapters for paired end data (because auto-detection is only enabled for single end data by default).</p>
<p>FastP (and other trimming software) can be used for quality control before aligning your sequences to the reference genome, but there are arguments on both sides for whether or not trimming your reads at this stage is useful (add references in here).</p>
</div>
<div id="mapping" class="section level3">
<h3>Mapping</h3>
<p>Our pipeline uses the Burrows-Wheeler transform (Li and Durbin 2010) to align reads to the reference genome, but there are many software that can be used for aligning short reads to a reference (e.g., Bowtie2, STAR). To make the alignment process more efficient, you need to index the reference genome.</p>
<pre class="bash"><code>bwa index {reference.fa}</code></pre>
<p>Where <code>reference.fa</code> is your reference genome. Once your reference has been indexed, you can do a local alignment of your FASTQs, and pipe directly into <code>samtools</code> to sort the alignment by coordinate and converted to the BAM (Binary ALignment/Map) format.<br></p>
<pre class="bash"><code>bwa mem -M -R {string} {ref} {r1} {r2} | samtools sort -o {output.bam}</code></pre>
<p>Where <br> <code>-M</code> marks shorter split hits as secondary (required for Picard compatibility downstream) <br> <code>-R</code> defines a read group header e.g., <span class="citation">@RG</span>\tID:{sample}\tSM:{sample}\tPL:ILLUMINA <br> <code>{ref}</code> is the indexed reference genome <br> <code>{r1}</code> and <code>{r2}</code> are the first and second reads you are aligning <br> <code>-o</code> is the output file name</p>
<p>Once you have a sorted and indexed alignment, you can locate and tag duplicate records, which are reads that have originated from a single fragment of DNA. This can happen during sample preparation (e.g., during PCR) or from sequencing artifacts (e.g., optical duplicates).If you have multiple runs for the same sample, you would want to merge and index the BAMs before the deduplication tagging, which you can do with <code>samtools merge</code>.</p>
<pre class="bash"><code>picard MarkDuplicates I={input.bam} O={output.bam} METRICS_FILE={output} REMOVE_DUPLICATES=false TAGGING_POLICY=All</code></pre>
<p>Where <br> <code>I</code> is the input BAM that has been coordinate sorted (the output of <code>samtools sort</code>) <br> <code>O</code> is the output BAM file <br> <code>METRICS_FILE</code> writes the deduplication metrics to the output file name (this is a required argument) <br> <code>REMOVE_DUPLICATES</code> writes the apropriate flags to the record rather than removing the reads without telling you what they are (<code>false</code> is the default setting) <br> <code>TAGGING_POLICY</code> will set the duplicates to be recorded in the DT attribute when the parameter is set to <code>All</code></p>
<p>Once you have tagged the duplicated records in your BAM, you will want to generate a .bai index file</p>
<pre class="bash"><code>picard BuildBamIndex I={input.bam}</code></pre>
<p>Where <br> <code>I</code> is the input dedup’d BAM (the output of <code>picard MarkDuplicates</code>)</p>
</div>
<div id="quality-control-and-filtering" class="section level3">
<h3>Quality control and filtering</h3>
<p>It’s important to take a look at your alignments to check for quality, potentially filter out sites or samples that don’t meet adequate thresholds, etc. The pipeline outputs various statistics automatically along the way, but we have collected some of the QC commands here. <br></p>
<p>We use <code>samtools</code> to compute the depth at each position or region, with the tabulated output containing information for the chromosome, start position, end position, number of reads aligned in that region after filtering, number of covered bases with a depth greater than or equal to 1 in that region, the proportion of coverage bases, the mean depth of coverage, mean baseQ, and mean mapQ.</p>
<pre class="bash"><code>samtools coverage --output {output} {input.bam}</code></pre>
<p>Where <br> <code>--output</code> is the file name you want to write the output to <br> <code>input.bam</code> is the input BAM file <br><br></p>
<p>To produce a summary of the alignment metrics, including the quality of read alignments and the quality filters, we use <code>Picard</code></p>
<pre class="bash"><code>picard ValidateSamFile I={input.bam} R={reference.fa} O={output}</code></pre>
<p>Where <br> <code>I</code> is the input BAM file <br> <code>R</code> is the reference genome <br> <code>O</code> is the file name you want to write the output to <br><br></p>
<p>To ensure the alignments meet file format specifications:</p>
<pre class="bash"><code>picard ValidateSamFile I={input.bam} R={reference.fa} O={output} IGNORE=VALID_TAG_NM || true</code></pre>
</div>
</div>
<div id="bam2vcf" class="section level1">
<h1>bam2vcf</h1>
<div id="variant-calling" class="section level3">
<h3>Variant Calling</h3>
<p>Two of the major variant callers you can use (at the time of writing) are Genome Analysis Toolkit (GATK) and freebayes. You can call variants with either of these software in our pipeline and the downstream steps are the same, so here we present parameters for variant calling with both GATK and freebayes. Regardless of the software used, variant calling is computationally intensive but can be parallelized for more efficient resource use by splitting the reference genome into intervals. The interval creation is automated in our pipeline but parameter guidelines will differ depending on your reference genome. <br><br></p>
<p>To call germline single nucleotide polymorphisms (SNPs) and insertion/deletions (indels) via local re-assembly of haplotypes using GATK:</p>
<pre class="bash"><code>gatk HaplotypeCaller -R {reference.fa} -I {input.bam} -O {output.gvcf} -L {interval.list} --emit-ref-confidence GVCF --min-pruning 2 --min-dangling-branch-length 4</code></pre>
<p>Where <br> <code>-R</code> is the reference genome <br> <code>-I</code> is the input BAM <br> <code>-O</code> is the output gVCF <br> <code>-L</code> is the interval file <br> <code>--emit-ref-confidence</code> yields the reference confidence scores as gVCF <br> <code>--min-pruning</code> sets the minimum support to not prune paths in the graph <br> <code>--min-dangling-branch-length</code> sets the minimum lenght of a dangling branch to attempt recovery <br><br></p>
<p>A note about coverage: when calling variants where the coverage is less than 10x, we suggest setting <code>--min-pruning</code> and <code>--min-dangling-branch-length</code> each to 1. The parameters in the code above are more appropriate for data where the coverage exceeds 10x. <br></p>
<p>There are some constraints when using clusters and job scheduling managers, including command lengths. When you have many samples and are trying to run GATK on all of them, the command may get too long, causing SLURM (or another job scheduler) to throw an error. To get around this, you can create DB map files (which is done automatically in our pipeline). Then we can import many gVCFs into a GenomicsDB for a given scaffold:</p>
<pre class="bash"><code>export TILEDB_DISABLE_FILE_LOCKING=1
gatk GenomicsDBImport --genomicsdb-workspace-path {output.db} -L --sample-name-map {input.db}</code></pre>
<p>Where <br> the <code>export</code> command can remedy sluggish performance <br> <code>--genomicsdb-workspace-path</code> is the working directory to point to for GenomicsDB <br> <code>-L</code> is the interval file <br> <code>--sample-name-map</code> is the path to a file that contains a sample to file map in a tab delimited format <br><br></p>
<p>Now you can use the genomic databases from <code>gatk GenomicsDBImport</code> to create VCF files (one per interval file)</p>
<pre class="bash"><code>gatk GenotypeGVCFs -R {reference.fa} -V {input.DB} -O {output.vcf}</code></pre>
<p>Where <br> <code>-R</code> is the reference genome <br> <code>-V</code> is the DB map file <br> <code>-O</code> is the output VCF <br><br></p>
<p>If you want to use freebayes the variant caller, rather than starting with <code>gatk HaplotypeCaller</code>, you would instead use:</p>
<pre class="bash"><code>freebayes -f {reference.fa} -r {interval} -g {maxDP} {inputs.bam} &gt; {output.vcf}</code></pre>
<p>Where <br> <code>-f</code> is the reference genome (creates a .fai index if one doesn’t already exist) <br> <code>-r</code> limits the analysis to a specific interval (region) <br> <code>-g</code> skips processing of alignments overlapping positions with coverage greater than the value; we use the maxDP for this parameter <br> <code>{inputs.bam}</code> are the input BAMs; you can input all BAMs at once but you need to set the output as one VCF per interval <br> <code>{output.vcf}</code> the output VCF for each interval e.g., {i}.vcf per interval <br><br></p>
<p>Regardless of the variant caller you use, before combining all the VCFs for the intervals into one final VCF, it is computationally more efficient to filter each of the VCFs, then gather them.</p>
<pre class="bash"><code>gatk VariantFiltration -R {reference.fa} -V {input.vcf} --output {output.vcf} \
--filter-name &quot;RPRS_filter&quot; \
--filter-expression &quot;(vc.isSNP() &amp;&amp; (vc.hasAttribute(&#39;ReadPosRankSum&#39;) &amp;&amp; ReadPosRankSum &lt; -8.0)) || ((vc.isIndel() || vc.isMixed()) &amp;&amp; (vc.hasAttribute(&#39;ReadPosRankSum&#39;) &amp;&amp; ReadPosRankSum &lt; -20.0)) || (vc.hasAttribute(&#39;QD&#39;) &amp;&amp; QD &lt; 2.0)&quot; \
--filter-name &quot;FS_SOR_filter&quot; \
--filter-expression &quot;(vc.isSNP() &amp;&amp; ((vc.hasAttribute(&#39;FS&#39;) &amp;&amp; FS &gt; 60.0) || (vc.hasAttribute(&#39;SOR&#39;) &amp;&amp;  SOR &gt; 3.0))) || ((vc.isIndel() || vc.isMixed()) &amp;&amp; ((vc.hasAttribute(&#39;FS&#39;) &amp;&amp; FS &gt; 200.0) || (vc.hasAttribute(&#39;SOR&#39;) &amp;&amp;  SOR &gt; 10.0)))&quot; \
--filter-name &quot;MQ_filter&quot; \
--filter-expression &quot;vc.isSNP() &amp;&amp; ((vc.hasAttribute(&#39;MQ&#39;) &amp;&amp; MQ &lt; 40.0) || (vc.hasAttribute(&#39;MQRankSum&#39;) &amp;&amp; MQRankSum &lt; -12.5))&quot; \
--filter-name &quot;QUAL_filter&quot; \
--filter-expression &quot;QUAL &lt; 30.0&quot; \
--invalidate-previous-filters

gatk GatherVcfs {input.vcfs} -O {output.vcf}</code></pre>
<p>Where <br> <code>-R</code> is the reference genome <br> <code>-V</code> is the input VCF (per interval) <code>--output</code> is the filtered VCF (per interval) <br> <code>{input.vcfs}</code> is a custom script invoked by the pipeline to gather all the filtered VCFs for each interval into a single input command using the intervals list <br> <code>-O</code> is the final output VCF <br></p>
<p>The filter expressions more or less follow the GATK hard filtering recommendations but are split to treat SNPs and indels with different thresholds within each filter expression, as well as keeping different quality tags separate to make downstream filtering easier.</p>
<p><br></p>
<div id="quality-control-and-statistics" class="section level4">
<h4>Quality control and statistics <br></h4>
<p>NB: this section will likely be better suited to be included in a QC section (and it’s not fully fleshed out itself yet), but here are two of the pipeline outputs that can be used for downstream QC. <br></p>
<pre class="bash"><code>vcftools --gzvcf {input.vcf} --remove-filtered-all --minDP 1 --missing-indv --stdout &gt; {output}</code></pre>
<p>Where <br> <code>--gzvcf</code> is the input VCF, gzipped (use <code>--vcf</code> if working with an uncompressed VCF) <br> <code>--remove-filtered-all</code> removes all the sites with a FILTER flag other than PASS <br> <code>--minDP</code> includes only the genotypes greater than or equal to the parameter value (this requires the DP FORMAT tag to be specified) <br> <code>--missing-indv</code> generates a file reporting the missingness per indivdual <br> <code>--stdout</code> redirects the output to stdout so it can be redirected to a file name of your choosing (in this code, the file name would be where <code>{output}</code> is) <br> <br></p>
<pre class="bash"><code>bedtools intersect -a {input.intervals} -b {input.vcf} -c &gt; {output}</code></pre>
<p>Where <br> <code>-a</code> is the intervals BED produced by the pipeline automatically for use in freebayes <br> <code>-b</code> is the final VCF file output by <code>gatk GatherVcfs</code> <br> <code>-c</code> reports the number of hits in <code>-b</code> for each entry in <code>-a</code> <br> <code>{output}</code> is the file name of the output, which is a text file that reports the number of SNPs per interval</p>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
