<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Introduction</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="tutorial.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MarineOmics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="ADMIN_02_contributions.html">Contributions</a>
</li>
<li>
  <a href="ADMIN_03_panels.html">Panel Seminars</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Population Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="POP_01_choosing_population_genetics.html">Choosing a Population Genomics Approach</a>
    </li>
    <li>
      <a href="POP_04_WGS_intro.html">Whole Genome Resequencing</a>
    </li>
    <li>
      <a href="POP_02_RADseq.html">Reduced Representation Sequencing</a>
    </li>
    <li>
      <a href="POP_03_poolseq.html">Poolseq</a>
    </li>
    <li>
      <a href="POP_05_RDAtraitPredictionTutorial.html">Redundancy Analysis (RDA) Trait Prediction</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Functional Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="FUN_01_DGE_comparison_v2.html">Mutifactorial RNAseq</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Genome-Phenome
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">coming soon!</li>
  </ul>
</li>
<li>
  <a href="https://github.com/MarineOmics/marineomics.github.io/discussions">Discussion Forum</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'G-53GH9PV49T', 'auto');
  ga('send', 'pageview');

</script>

<div id="header">



<h1 class="title toc-ignore">Introduction</h1>

</div>


<div id="guidelines-for-omics-analyses" class="section level1">
<h1>Guidelines for ’omics analyses</h1>
<p>On this website you will find a growing and evolving collection of
guidelines, tutorials, recommended readings, and method evaluations for
various ’omics analyses that are commonly used in studies of nonmodel
systems (particularly marine species). Instead of providing rigid “best
practices”, we aim to encourage data exploration and rigorous quality
control through “best principles”, paying particular attention to topics
for which open access guidelines are scarce. The guidelines are written
by members of the <a href="#marineomics">MarineOmics working group</a>
or the community, then shared for peer review. Comments on current
website content or suggestions for new content can be made through the
<a
href="https://github.com/MarineOmics/marineomics.github.io/discussions">Discussion
Forum</a>.</p>
<p>Based on the expertise of the MarineOmics working group, the first
topics include:</p>
<div id="topics" class="section level3 tabset tabset-fade tabset-pills">
<h3 class="tabset tabset-fade tabset-pills">Topics</h3>
<div id="population-genomics" class="section level4">
<h4>Population Genomics</h4>
<ul>
<li><a href="POP_01_choosing_population_genetics.html">Choosing a
population genomics sequencing approach</a><br />
</li>
<li><a href="POP_04_WGS_intro.html">Whole Genome Resequencing (WGS)</a>
<ul>
<li><a
href="https://github.com/therkildsen-lab/genomic-data-analysis/blob/master/lcwgs_data_analysis.md">Low
coverage WGS</a></li>
</ul></li>
<li><a href="POP_02_RADseq.html">Reduced Representation
Sequencing</a></li>
<li><a href="POP_03_poolseq.html">Poolseq</a></li>
<li><a href="POP_05_RDAtraitPredictionTutorial.html">Redundancy Analysis
(RDA) Trait Prediction</a></li>
</ul>
</div>
<div id="functional-genomics" class="section level4">
<h4>Functional Genomics</h4>
<ul>
<li><a href="FUN_01_DGE_comparison_v2.html">Multifactorial RNAseq
analyses</a><br />
</li>
<li>Align and count RNAseq reads (coming soon!)
<ul>
<li>De novo, reference-based, and Tag-seq methods</li>
</ul></li>
</ul>
</div>
<div id="genome-phenome" class="section level4">
<h4>Genome-Phenome</h4>
<ul>
<li>coming soon!</li>
</ul>
</div>
</div>
<div id="section" class="section level3 unnumbered">
<h3 class="unnumbered"></h3>
</div>
<div
id="the-guidelines-portion-of-the-website-should-be-considered-a-preprint.-if-you-are-interested-in-contributing-to-the-peer-review-process-check-out-the-discussion-forum."
class="section level3 unlisted unnumbered">
<h3 class="unlisted unnumbered">The guidelines portion of the website
should be considered a “preprint”. If you are interested in contributing
to the peer review process, check out the <a
href="discussion.html">Discussion Forum</a>.</h3>
</div>
</div>
<div id="panel-seminars" class="section level1">
<h1>Panel Seminars</h1>
<p>To facilitate discussions about practices and pitfalls in applying
’omics to non-model systems, we hosted <a href="ADMIN_03_panels.html">Q
&amp; A panel seminars</a> with experts on topics ranging from RADseq,
population genomics with WGS, and transcriptomics. Check them out on <a
href="https://www.youtube.com/channel/UCosG6JfIj7k0DizfsD5ADfQ">YouTube</a>!</p>
</div>
<div id="marineomics" class="section level1">
<h1>What is the MarineOmics working group?</h1>
<p>The original MarineOmics working group is part of the <a
href="https://rcn-ecs.github.io/">RCN for Evolution in Changing
Seas</a>, and was formed to promote <strong>reproducible</strong> and
<strong>robust</strong> genomic research in nonmodel species, but
especially marine and aquatic studies. Methods in next generation
sequencing and bioinformatics advance at a pace that quickly render any
static set of best practices obsolete. MarineOmics aims to support
robust genomic research in ecology and evolution by disseminating
guidelines through this dynamic, open-sourced website addressing
introductory and advanced topics of population- and functional genomics.
<em>We are always looking for folks to join and write content for the
website, if interested please <a href="#contact">contact us</a> and see
the <a href="#contributions">below section</a>!</em></p>
<p><strong>Rigorous</strong>: following best principles to identify and
mitigate sources of error in data generation and analysis</p>
<p><strong>Reproducible</strong>: when others can reproduce the results
with accessible data, code, and documentation</p>
</div>
<div id="reproducibility-in-marine-genomics" class="section level1">
<h1>Reproducibility in Marine Genomics</h1>
<p>MarineOmics is also interested in evaluating and improving
reproducibility in studies that use genomics to study adaptation in
marine systems. We are conducting a meta-analysis of reproducibility
across genomic studies of environmental adaptation in marine populations
with the aim of identifying obstacles and opportunities for reproducing
and synthesizing population genomic research. In Spring 2021, three
undergraduate research interns contributed to this literature survey,
with funding from the <a
href="https://rcn-ecs.github.io/">RCN-ECS</a>.</p>
</div>
<div id="contributions" class="section level1">
<h1>Making contributions to the MarineOmics.github.io website</h1>
<p>We are excited to incorporate new material from contributors of
various backgrounds, expertise, and experience. To date these tutorials,
guidelines, and discussions have been authored and led by graduate
students and postdocs with contributions from scientists at all levels,
including undergraduates, technicians, and faculty members. A diverse
range of perspectives and backgrounds will make these tutorials more
inclusive, informative, and useful for everyone in the marine-omics
community. If you are interested in developing an information page or
tutorial for the website, we ask that you review our <a
href="ADMIN_01_submissions_instructions.html">page with instructions for
submitting content</a> and contact our Website Manager Jason Johns
(jasonjohns[at]ucsb.edu).</p>
</div>
<div id="internship" class="section level1">
<h1>Internship</h1>
<p>In Spring 2021, MarineOmics hosted three virtual undergraduate
research interns with funding from the <a
href="https://rcn-ecs.github.io/">RCN-ECS</a>: Jecy Klinkam, Kaitlin
Macaranas, and Chris Mantegna. These interns were paired with graduate
student mentors within the MarineOmics group, as well as participated in
training activities while contributing to the Reproducibility literature
review.</p>
</div>
<div id="contact" class="section level1">
<h1>Contact</h1>
<p>For questions about the website and its materials, post to the <a
href="https://github.com/MarineOmics/marineomics.github.io/discussions">Discussion
Forum</a>! For inquiries about the MarineOmics Working Group activities,
please contact us at <a href="mailto:rcnecs.marineomics@gmail.com"
class="email">rcnecs.marineomics@gmail.com</a>.</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
