```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
---
title: "Introduction"
---

# Guidelines for 'omics analyses  

On this website you will find a growing and evolving collection of guidelines, tutorials, recommended readings, and method evaluations for various 'omics analyses that are commonly used in studies of nonmodel systems (particularly marine species). Instead of providing rigid "best practices", we aim to encourage data exploration and rigorous quality control through "best principles", paying particular attention to topics for which open access guidelines are scarce. The guidelines are written by members of the [MarineOmics working group](#marineomics) or the community, then shared for peer review. Comments on current website content or suggestions for new content can be made through the [Discussion Forum](https://github.com/MarineOmics/marineomics.github.io/discussions).

Based on the expertise of the MarineOmics working group, the first topics include:  

### Topics {.tabset .tabset-fade .tabset-pills}
#### Population Genomics  

* [Choosing a population genomics sequencing approach](POP_01_choosing_population_genetics.html)  
* [Whole Genome Resequencing (WGS)](POP_04_WGS_intro.html)  
  * [Low coverage WGS](https://github.com/therkildsen-lab/genomic-data-analysis/blob/master/lcwgs_data_analysis.md)
* [Reduced Representation Sequencing](POP_02_RADseq.html) 
* [Poolseq](POP_03_poolseq.html)
* [Redundancy Analysis (RDA) Trait Prediction](POP_05_RDAtraitPredictionTutorial.html)

#### Functional Genomics  

* [Multifactorial RNAseq analyses](FUN_01_DGE_comparison_v2.html)  
* Align and count RNAseq reads (coming soon!)
    * De novo, reference-based, and Tag-seq methods

#### Genome-Phenome

* coming soon!

### {-}

### The guidelines portion of the website should be considered a "preprint". If you are interested in contributing to the peer review process, check out the [Discussion Forum](discussion.html). {.unlisted .unnumbered}

# Panel Seminars 
To facilitate discussions about practices and pitfalls in applying 'omics to non-model systems, we hosted [Q & A panel seminars](ADMIN_03_panels.html) with experts on topics ranging from RADseq, population genomics with WGS, and transcriptomics. Check them out on [YouTube](https://www.youtube.com/channel/UCosG6JfIj7k0DizfsD5ADfQ)!

# What is the MarineOmics working group? {#marineomics} 

The original MarineOmics working group is part of the [RCN for Evolution in Changing Seas](https://rcn-ecs.github.io/), and was formed to promote **reproducible** and **robust** genomic research in nonmodel species, but especially marine and aquatic studies. Methods in next generation sequencing and bioinformatics advance at a pace that quickly render any static set of best practices obsolete. MarineOmics aims to support robust genomic research in ecology and evolution by disseminating guidelines through this dynamic, open-sourced website addressing introductory and advanced topics of population- and functional genomics. *We are always looking for folks to join and write content for the website, if interested please [contact us](#contact) and see the [below section](#contributions)!*

**Rigorous**:  following best principles to identify and mitigate sources of error in data generation and analysis  

**Reproducible**: when others can reproduce the results with accessible data, code, and documentation

# Reproducibility in Marine Genomics  
MarineOmics is also interested in evaluating and improving reproducibility in studies that use genomics to study adaptation in marine systems. We are conducting a meta-analysis of reproducibility across genomic studies of environmental adaptation in marine populations with the aim of identifying obstacles and opportunities for reproducing and synthesizing population genomic research. In Spring 2021, three undergraduate research interns contributed to this literature survey, with funding from the [RCN-ECS](https://rcn-ecs.github.io/). 

# Making contributions to the MarineOmics.github.io website {#contributions}
We are excited to incorporate new material from contributors of various backgrounds, expertise, and experience. To date these tutorials, guidelines,  and discussions have been authored and led by graduate students and postdocs with contributions from scientists at all levels, including undergraduates, technicians, and faculty members. A diverse range of perspectives and backgrounds will make these tutorials more inclusive, informative, and useful for everyone in the marine-omics community. If you are interested in developing an information page or tutorial for the website,  we ask that you review our [page with instructions for submitting content](ADMIN_01_submissions_instructions.html) and contact our Website Manager Jason Johns (jasonjohns[at]ucsb.edu).

# Internship 
In Spring 2021, MarineOmics hosted three virtual undergraduate research interns with funding from the [RCN-ECS](https://rcn-ecs.github.io/): Jecy Klinkam, Kaitlin Macaranas, and Chris Mantegna. These interns were paired with graduate student mentors within the MarineOmics group, as well as participated in training activities while contributing to the Reproducibility literature review.


# Contact {#contact}
For questions about the website and its materials, post to the [Discussion Forum](https://github.com/MarineOmics/marineomics.github.io/discussions)!
For inquiries about the MarineOmics Working Group activities, please contact us at rcnecs.marineomics@gmail.com.

