```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
---
title: "Welcome"
---

On this website you will find a growing and evolving collection of guidelines, tutorials, recommended readings, and method evaluations for various 'omics analyses that are commonly used in studies of nonmodel systems (particularly marine species). Instead of providing rigid "best practices", we aim to encourage data exploration and rigorous quality control through ["best principles"](#best_principles), paying particular attention to topics for which open access guidelines are scarce. The guidelines are written by members of the [MarineOmics working group](#marineomics) and/or the community, then shared for community input. Comments on current website content or suggestions for new content can be made through the [Discussion Forum](https://github.com/MarineOmics/marineomics.github.io/discussions).

# Best Principles <a id="best_principles"></a>

The backbone underlying the content on the website comes from a general set of “best principles” that should be applied in genomics studies, irrespective of the specific sequencing method used. Instead of providing rigid “best practices", the website promotes “best principles”, defined as a guiding set of values and goals that can be tailored to one’s hypotheses and guided by one's data, as opposed to a specific set of instructions. Motivated by rigor and reproducibility, "best principles" are designed to encourage data exploration and critical thinking during analysis and evaluation instead of ticking off boxes on a protocol or step-list. Please visit our [best principles page](ADMIN_04_best_principles.html) for a detailed outline.

# Population Genomics  

* [Choosing a population genomics sequencing approach](POP_01_choosing_population_genetics.html)  
* [Whole Genome Resequencing (WGS)](POP_04_WGS_intro.html)  
  * [Low coverage WGS](https://github.com/therkildsen-lab/genomic-data-analysis/blob/master/lcwgs_data_analysis.md)
* [Reduced Representation Sequencing](RADseq.html) 
* [Poolseq](POP_03_poolseq.html)
* [Redundancy Analysis (RDA) Trait Prediction](POP_05_RDAtraitPredictionTutorial.html)
* [PCA](POP_08_PCA.html)

# Functional Genomics  

* [Multifactorial RNAseq analyses](DGE_comparison_v2.html)  
* Align and count RNAseq reads (coming soon!)
    * De novo, reference-based, and Tag-seq methods

# Genome-Phenome
* coming soon!


# Panel Seminars 

To facilitate discussions about practices and pitfalls in applying 'omics to non-model systems, we hosted Q & A panel seminars with experts on topics ranging from RADseq, population genomics with WGS, and transcriptomics in May-June 2021. 3-5 expert panelists discussed each topic, with questions coming from the working group and the audience. Videos are available on YouTube!


| Date        | Topic       | Panelists     | Video     |
| :---        | :---        | :---          | :---      |
| May 7         | GBS/RADseq genotyping pipelines     | [Melanie LaCava](https://melanielacava.weebly.com/), [Alicia Mastretta-Yanes](http://mastrettayanes-lab.org/), [Isaac Overcast](https://isaacovercast.github.io/), [Jon Puritz](https://www.marineevoeco.com/)   | [View video](https://www.youtube.com/watch?v=C74GBESeIq4)     |
| May 21       |  WGS genotyping (low and high coverage)     | [Jeffrey Good](http://www.thegoodlab.org/), [Mikhail "Misha" Matz](https://matzlab.weebly.com/), [Noah Reid](http://nreid.github.io/), [Tim Sackton](https://scholar.harvard.edu/tsackton), [Nina Overgaard Therkildsen](https://www.therkildsenlab.com/) | [View Video](https://www.youtube.com/watch?v=ObJ3OT3dwSw)    |
| June 4       |  Population genomic analyses and filtering data     | [Zach Gompert](https://gompertlab.com/), [Morgan Kelly](http://www.morgankelly.biology.lsu.edu/), [Katie Lotterhos](https://sites.google.com/site/katielotterhos/home), [Jeff Ross-Ibarra](https://rilab.ucdavis.edu/) |  [View Video](https://youtu.be/dZyuqVEGPMo)   |
| June 18       |  RNAseq (experimental design through differential expression)     | [Ana Conesa](http://conesalab.org/), [Sarah Davies](http://sites.bu.edu/davieslab/), [Martin Hölzer](https://hoelzer-lab.github.io/), [Joanna Kelley](https://labs.wsu.edu/genomes/), [Harold Pimentel](https://pimentellab.com/)     |  [View Video](https://youtu.be/-gQ59m2cck0)   |




# How to Contribute
We are excited to incorporate new material from contributors of various backgrounds, expertise, and experience. To date these tutorials, guidelines,  and discussions have been authored and led by graduate students and postdocs with contributions from scientists at all levels, including undergraduates and faculty members. If you are interested in developing an information page or tutorial for the website, we ask that you review our [page with instructions for developing content](ADMIN_01_submissions_instructions.html) and contact our Website Manager Jason Johns (jasonjohns[at]ucsb.edu).


# Background
The original MarineOmics working group is part of the [RCN for Evolution in Changing Seas](https://rcn-ecs.github.io/), and was formed to promote **reproducible** and **robust** genomic research in nonmodel species, but especially marine and aquatic studies. Methods in next generation sequencing and bioinformatics advance at a pace that quickly render any static set of best practices obsolete. MarineOmics aims to support robust genomic research in ecology and evolution by disseminating guidelines through this dynamic, open-sourced website addressing introductory and advanced topics of population- and functional genomics.

**Rigorous**:  following best principles to identify and mitigate sources of error in data generation and analysis  

**Reproducible**: when others can reproduce the results with accessible data, code, and documentation

MarineOmics is also interested in evaluating and improving reproducibility in studies that use genomics to study adaptation in marine systems. We are conducting a meta-analysis of reproducibility across genomic studies of environmental adaptation in marine populations with the aim of identifying obstacles and opportunities for reproducing and synthesizing population genomic research. In Spring 2021, three undergraduate research interns contributed to this literature survey, with funding from the [RCN-ECS](https://rcn-ecs.github.io/). 

In Spring 2021, MarineOmics hosted three virtual undergraduate research interns with funding from the [RCN-ECS](https://rcn-ecs.github.io/): Jecy Klinkam, Kaitlin Macaranas, and Chris Mantegna. These interns were paired with graduate student mentors within the MarineOmics group, as well as participated in training activities while contributing to the Reproducibility literature review.

# How to Cite{#how_to_cite}

If referencing marineomics.io, please use the following citation:

Bogan, S. N., Johns, J., Griffiths, J. S., Davenport, D., Smith, S. J., Schaal, S. M., Downey-Wall, A., Lou, R. N., Lotterhos, K., Guidry, M. E., Rivera, H. E., McGirr, J. A., Puritz, J. B., Roberts, S. B., and Silliman, K. (2023). A dynamic web resource for robust and reproducible genomics in nonmodel species: marineomics.io. *Methods in Ecology and Evolution*, 00, 1–8. https://doi.org/10.1111/2041-210X.14219
<br>

If citing a specific page, please use the above citation and a reference to the individual page as follows:
<br>

LastName1, Initials1, LastName2, Initials2, etc. (Publication year). Webpage title. URL

For the page "Choosing a Population Genomics Approach" the citation would be written as

Silliman, K. and Lou, R. N. (2022). Choosing a Population Genomics Approach. MarineOmics. https://marineomics.github.io/POP_01_choosing_population_genetics.html

# Contact {#contact}
For questions about the website and its materials, post to the [Discussion Forum](https://github.com/MarineOmics/marineomics.github.io/discussions)!
For inquiries about the MarineOmics Working Group activities, please contact us at rcnecs.marineomics@gmail.com.