<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Whole Genome Resequencing for Population Genomics</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="tutorial.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MarineOmics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="ADMIN_02_contributions.html">Contributions</a>
</li>
<li>
  <a href="ADMIN_03_panels.html">Panel Seminars</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Population Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="POP_01_choosing_population_genetics.html">Choosing a Population Genomics Approach</a>
    </li>
    <li>
      <a href="POP_04_WGS_intro.html">Whole Genome Resequencing</a>
    </li>
    <li>
      <a href="POP_02_RADseq.html">Reduced Representation Sequencing</a>
    </li>
    <li>
      <a href="POP_03_poolseq.html">Poolseq</a>
    </li>
    <li>
      <a href="POP_05_RDAtraitPredictionTutorial.html">Redundancy Analysis (RDA) Trait Prediction</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Functional Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="FUN_01_DGE_comparison_v2.html">Mutifactorial RNAseq</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Genome-Phenome
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">coming soon!</li>
  </ul>
</li>
<li>
  <a href="https://github.com/MarineOmics/marineomics.github.io/discussions">Discussion Forum</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'G-53GH9PV49T', 'auto');
  ga('send', 'pageview');

</script>

<div id="header">



<h1 class="title toc-ignore">Whole Genome Resequencing for Population
Genomics</h1>
<h3 class="subtitle"><em>Katherine Silliman, Nicolas Lou</em></h3>

</div>


<p><br></p>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>For the purpose of this guide, we use “whole genome (re)sequencing
(WGS)) to refer to methods where a reference genome already exists
(whether for the focal species or a related species), and uniquely
barcoded samples are sequenced and then mapped to most or all of the
reference genome. This method can provide high density genetic variants
(e.g., SNPs) and structural information for population genomic analyses,
while also facilitating functional insights if the reference genome is
functionally annotated. This is separate from whole-genome <em>de
novo</em> sequencing, which aims to produce a reference genome by
sequencing and assembling a complete genome for a species for the first
time.</p>
<p>An excellent and thorough review of the methodology and consideration
for WGS approaches, especially as they apply to non-model organisms, is
<a href="https://doi.org/10.1111/mec.14264"><span
class="citation">(Fuentes-Pardo and Ruzzante 2017)</span></a>. But
briefly, current WGS approaches can be broadly categorized into two
types: high-to-moderate-coverage WGS, and low-coverage WGS. Their major
distinction is that with high-to-moderate-coverage WGS, each individual
is sequenced at a depth with which genotype can be confidently called at
most sites, whereas with low-coverage WGS, each individual is sequenced
at a depth too low to call genotype with, and downstream analyses should
take such genotype uncertainties into acount.</p>
<p>However, the line between high-to-moderate-coverage WGS and
low-coverage WGS is not always as clear-cut as presented above. For
example, with moderate-coverage WGS (e.g. 5-20x), many sites within an
individual can still have low coverage due to random sampling, resulting
in unreliable genotype calls and/or missing data that can become
problematic in downstream analysis. Therefore, it could be preferrable
to avoid hard-calling genotypes with moderate-coverage WGS in certain
applications. On the other hand, in populations with high levels of
linkage disequilibrium (LD), it could be possible to leverage LD to
carry out genotype imputation, making genotype calling a lot more
accurate with low-coverage WGS data. Imputation is more likely to be
successful when a high-quality reference panel exists in the system (<a
href="https://www.science.org/doi/10.1126/science.aba4674"><span
class="citation">Fuller et al. (2020)</span></a>, <a
href="https://www.nature.com/articles/s41588-020-00756-0"><span
class="citation">Rubinacci et al. (2021)</span></a>), but methods for
imputation without such reference panels have also been developed (in
which case a very large sample size would be required, <a
href="https://www.nature.com/articles/ng.3594"><span
class="citation">Davies et al. (2016)</span></a>).</p>
<p>This said, here are the main differences between these two approaches
in practice: with same sample size, high-to-moderate-coverage WGS tends
to provide higher resolution data that are more versatile and less
susceptible to technical artefacts (especially those caused by
sequencing errors) when compared with low-coverage WGS, but it could be
a lot more costly. Low-coverage WGS, in contrast, can be used to achieve
higher sample size with a fixed budget, which can then contribute to
higher-resolution population-level inferences, but it does require a
different computational toolbox that takes genotype uncertainties into
account, and is thus constrained by limitations of the current toolbox
(see Section 6 in <a
href="https://onlinelibrary.wiley.com/doi/abs/10.1111/mec.16077"><span
class="citation">(Lou et al. 2021)</span></a> for a more detailed
discussion on this).</p>
<p><br></p>
</div>
<div id="tutorials" class="section level1">
<h1>Tutorials</h1>
<p>Here we provide a couple different tutorials by our working group
members, as well as links to other detailed tutorials on the web. The
goal of these MarineOmics tutorials are to provide extensive details on
“why” certain parameters are chosen, and some guidance on how to
evaluate different parameter options to fit your data.</p>
<p><br></p>
<div id="high-to-moderate-coverage-wgs" class="section level2">
<h2>High to moderate coverage WGS</h2>
<ul>
<li><p><a
href="POP_06_fastq2vcf_june2022_update/POP_06_fastq2vcf_june2022_update.html">Fastq-to-VCF
SnakeMake pipeline</a>: in-depth explanation of an automated short-read
mapping and variant calling pipeline maintained by <a
href="https://github.com/harvardinformatics/shortRead_mapping_variantCalling">Harvard
Informatics</a>. Useful if you plan to adopt this pre-packaged automated
and parallelizable pipeline, and would like to understand its different
components, but not necessarily change it substantially.</p></li>
<li><p><a
href="POP_07_WGS_Full_Walkthrough/POP_07_WGS_Full_Walkthrough.html">Fastq-to-VCF
workflow</a>: detailed walkthrough of processing 15x sequencing depth
WGS data for cod, from raw reads to a VCF. Useful if you would like to
run each component of the pipeline yourself and potentially tweak some
of them for your own purpose. In other words, you can more easily add,
skip, or change parts of this pipeline, but will lose the convenience
offered by an automated pipeline.</p></li>
</ul>
<p><br></p>
</div>
<div id="low-coverage-wgs" class="section level2">
<h2>Low coverage WGS</h2>
<p>The quality control and read alignment part of the pipeline for
high-to-moderate-coverage WGS also applies for low-coverage WGS.
Therefore, the two tutorials for high-to-moderate-coverage WGS are also
useful for low-coverage WGS until the point where variants and genotypes
are called. In addition to these, here are some resources specifically
designed for low-coverage WGS.</p>
<ul>
<li><p><a
href="https://github.com/nt246/lcwgs-guide-tutorial">Low-coverage WGS
tutorial</a>: a tutorial for the processing and analysis of low-coverage
WGS data (i.e. from raw fastq files to population genomic inference),
with example datasets and hands-on exercises. It is associated with the
paper <a
href="https://onlinelibrary.wiley.com/doi/abs/10.1111/mec.16077"><span
class="citation">(Lou et al. 2021)</span></a>.</p></li>
<li><p><a
href="https://github.com/therkildsen-lab/batch-effect/blob/main/tutorial/tutorial.md">Detection
and mitigation of batch effects</a>: a tutorial for the detection and
mitigation of batch effects with low-coverage WGS data, with example
datasets and hands-on exercises. It is associated with the paper <a
href="https://onlinelibrary.wiley.com/doi/abs/10.1111/1755-0998.13559"><span
class="citation">Lou and Therkildsen (2021)</span></a>.</p></li>
<li><p><a
href="https://github.com/therkildsen-lab/genomic-data-analysis/blob/master/lcwgs_data_analysis.md">Low-coverage
WGS data analysis pipeline</a>: a collection of scripts for the
efficient and reproducible analysis of low-coverage WGS data (i.e. from
bam to population genomic inference).</p></li>
<li><p><a
href="https://github.com/therkildsen-lab/data-processing/blob/master/lcwgs_data_processing.md">Low-coverage
WGS data processing pipeline</a>: a collection of script for the
efficient and reproducible processing of low-coverage WGS data
(i.e. from raw fastq to bam). This pipeline should also be compatible
with high-to-moderate-coverage WGS data.</p></li>
</ul>
<p><br></p>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Davies2016-rq" class="csl-entry">
Davies, Robert W, Jonathan Flint, Simon Myers, and Richard Mott. 2016.
<span>“Rapid Genotype Imputation from Sequence Without Reference
Panels.”</span> <em>Nat. Genet.</em> 48 (8): 965–69. <a
href="https://doi.org/10.1038/ng.3594">https://doi.org/10.1038/ng.3594</a>.
</div>
<div id="ref-Fuentes-Pardo2017-nu" class="csl-entry">
Fuentes-Pardo, Angela P, and Daniel E Ruzzante. 2017.
<span>“Whole-Genome Sequencing Approaches for Conservation Biology:
Advantages, Limitations and Practical Recommendations.”</span> <em>Mol.
Ecol.</em> 26 (20): 5369–5406. <a
href="https://doi.org/10.1111/mec.14264">https://doi.org/10.1111/mec.14264</a>.
</div>
<div id="ref-Fuller2020" class="csl-entry">
Fuller, Zachary L., Veronique J. L. Mocellin, Luke A. Morris, Neal
Cantin, Jihanne Shepherd, Luke Sarre, Julie Peng, et al. 2020.
<span>“Population Genetics of the Coral &lt;i&gt;acropora
Millepora&lt;/i&gt;: Toward Genomic Prediction of Bleaching.”</span>
<em>Science</em> 369 (6501): eaba4674. <a
href="https://doi.org/10.1126/science.aba4674">https://doi.org/10.1126/science.aba4674</a>.
</div>
<div id="ref-Lou2021-me" class="csl-entry">
Lou, Runyang Nicolas, Arne Jacobs, Aryn P Wilder, and Nina Overgaard
Therkildsen. 2021. <span>“A Beginner’s Guide to Low-Coverage Whole
Genome Sequencing for Population Genomics.”</span> <em>Mol. Ecol.</em>,
July. <a
href="https://doi.org/10.1111/mec.16077">https://doi.org/10.1111/mec.16077</a>.
</div>
<div id="ref-Lou2021-ew" class="csl-entry">
Lou, Runyang Nicolas, and Nina Overgaard Therkildsen. 2021. <span>“Batch
Effects in Population Genomic Studies with Low-Coverage Whole Genome
Sequencing Data: Causes, Detection, and Mitigation.”</span> <em>Authorea
Preprints</em>, August. <a
href="https://doi.org/10.22541/au.162791857.78788821/v2">https://doi.org/10.22541/au.162791857.78788821/v2</a>.
</div>
<div id="ref-Rubinacci2021-dm" class="csl-entry">
Rubinacci, Simone, Diogo M Ribeiro, Robin J Hofmeister, and Olivier
Delaneau. 2021. <span>“Efficient Phasing and Imputation of Low-Coverage
Sequencing Data Using Large Reference Panels.”</span> <em>Nat.
Genet.</em> 53 (1): 120–26. <a
href="https://doi.org/10.1038/s41588-020-00756-0">https://doi.org/10.1038/s41588-020-00756-0</a>.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
