<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Required Components</title>

<script src="site_libs/header-attrs-2.10/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="tutorial.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MarineOmics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="panels.html">Panel Seminars</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Population Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="choosing-population-genetics.html">Choosing a Population Genomics Approach</a>
    </li>
    <li>
      <a href="RADseq.html">RADseq</a>
    </li>
    <li>
      <a href="WGS_intro.html">Whole Genome Resqequencing</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Functional Genomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RNA_coming.html">RNAseq (coming soon!)</a>
    </li>
  </ul>
</li>
<li>
  <a href="discussion.html">Discussion Forum</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Required Components</h1>

</div>


<p>This website is an introduction to some of the principles of reproducible science in R. While we are using R and other tools for reproducible research, our focus here is not to teach you the tools, but how you can use them to make your research more reproducible. Before we get started on the bulk of the content, we will go over how to install the software needed for this tutorial and briefly introduce working with R.</p>
<div id="installing-r-and-rstudio" class="section level2">
<h2>Installing R and RStudio</h2>
<p>First, install R and RStudio:</p>
<ol style="list-style-type: decimal">
<li><p>Download and install the <a href="http://www.r-project.org">R</a> statistical computing and graphing environment. This works cross-platform on Windows, OS X and Linux operating systems.</p></li>
<li><p>Download and install the free, open source edition of the <a href="https://www.rstudio.com/products/rstudio/download/">RStudio Desktop</a> integrated development environment (IDE), which we recommend. This is basically a point-and-click interface for R that includes a text editor, file browser, and some other conveniences.</p></li>
</ol>
</div>
<div id="installing-the-required-packages" class="section level2">
<h2>Installing the required packages</h2>
<p>The following packages are used in this primer:</p>
<ol style="list-style-type: decimal">
<li><a href="http://cran.r-project.org/web/packages/rmarkdown/index.html">rmarkdown</a> (creating reports)</li>
<li><a href="http://cran.r-project.org/web/packages/agricolae/index.html">agricolae</a> (agricultural disease data analysis)</li>
<li><a href="http://cran.r-project.org/web/packages/ggplot2/index.html">ggplot2</a> (graphs)</li>
<li><a href="http://cran.r-project.org/web/packages/poppr/index.html">poppr</a> (genetic data analysis)</li>
<li><a href="http://cran.r-project.org/web/packages/dplyr/index.html">dplyr</a> (data manipulation)</li>
<li><a href="http://cran.r-project.org/web/packages/tidyr/index.html">tidyr</a> (data manipulation)</li>
</ol>
<p>To install these packages, open RStudio and copy and paste the following code into the console:</p>
<pre class="r"><code>my_packages &lt;- c(&quot;rmarkdown&quot;,&quot;poppr&quot;, &quot;agricolae&quot;, &quot;dplyr&quot;, &quot;tidyr&quot;, &quot;ggplot2&quot;)
install.packages(my_packages, repos = &quot;http://cran.rstudio.com&quot;, dependencies = TRUE)</code></pre>
<p>Congratulations! You should now be all set for using R. To ensure that everything is set up correctly, you can go through steps 1 and 2 in the <a href="Markdown.html">markdown chapter</a>.</p>
</div>
<div id="optional-software" class="section level2">
<h2>Optional software</h2>
<div id="pdfs-with-latex" class="section level3">
<h3>PDFs with LaTeX</h3>
<p>If you want to create PDF documents, you will need a <a href="https://latex-project.org/"><span class="math inline">\(\LaTeX\)</span></a> installation. For OSX and Ubuntu users, this can be a large download and you will want to ensure you have a good connection:</p>
<ul>
<li><a href="https://tug.org/mactex/">LaTeX for OSX</a></li>
<li><a href="http://miktex.org/">LaTeX for Windows</a></li>
<li><a href="https://help.ubuntu.com/community/LaTeX">LaTeX for Ubuntu</a></li>
</ul>
<p>To those that know of and fear LaTeX: don’t worry, you don’t need to write any LaTeX code to produce PDFs from Markdown.</p>
</div>
<div id="git" class="section level3">
<h3>Git</h3>
<p>Git is a version control program that we will cover, but since installation requirements can vary between operating systems, we are not requiring it for the workshop. However, if you would like to install it, <a href="http://happygitwithr.com/install-git.html">Here’s a website that covers installation for the major operating systems</a>.</p>
</div>
</div>
<div id="a-quick-introduction-to-r-using-rstudio" class="section level2">
<h2>A quick introduction to R using RStudio</h2>
<p>Let’s review some of the basic features and functions of R. To start R, open the RStudio application from your programs folder or start menu. This will initialize your R session. To exit R, simply close the RStudio application.</p>
<blockquote>
<p>Note that R is a case-sensitive language!</p>
</blockquote>
<p>Let’s get comfortable with R by submitting the following command on the console (where R prompts you with a <code>&gt;</code> in the lower left RStudio window pane) that will tell you something about your version of R and the packages you have, which is useful for reporting reproducible research.</p>
<pre class="r"><code>sessionInfo() # This command will tell you information about your current R session</code></pre>
<pre><code>## R version 3.6.3 (2020-02-29)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.3 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8   
##  [6] LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] tools     stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] kableExtra_1.3.4     SNPRelate_1.20.1     forcats_0.5.1        stringr_1.4.0        dplyr_1.0.7          purrr_0.3.4          readr_2.0.1         
##  [8] tibble_3.1.3         tidyverse_1.3.1      SeqArray_1.26.2      gdsfmt_1.22.0        ggplot2_3.3.5        adegenet_2.1.4       ade4_1.7-17         
## [15] tidyr_1.1.3          knitcitations_1.0.12 xtable_1.8-4         knitr_1.33          
## 
## loaded via a namespace (and not attached):
##   [1] readxl_1.3.1           backports_1.2.1        systemfonts_1.0.3      plyr_1.8.6             igraph_1.2.6           sp_1.4-5              
##   [7] splines_3.6.3          usethis_2.0.1          GenomeInfoDb_1.22.1    digest_0.6.27          htmltools_0.5.1.1      gdata_2.18.0          
##  [13] fansi_0.5.0            magrittr_2.0.1         memoise_2.0.0          cluster_2.1.2          tzdb_0.1.2             remotes_2.4.0         
##  [19] Biostrings_2.54.0      modelr_0.1.8           gmodels_2.18.1         svglite_2.0.0          prettyunits_1.1.1      colorspace_2.0-2      
##  [25] rvest_1.0.1            haven_2.4.3            xfun_0.25              callr_3.7.0            crayon_1.4.1           RCurl_1.98-1.3        
##  [31] jsonlite_1.7.2         ape_5.5                glue_1.4.2             gtable_0.3.0           zlibbioc_1.32.0        XVector_0.26.0        
##  [37] webshot_0.5.2          seqinr_4.2-8           pkgbuild_1.2.0         BiocGenerics_0.32.0    scales_1.1.1           DBI_1.1.1             
##  [43] bibtex_0.4.2.3         Rcpp_1.0.7             viridisLite_0.4.0      spData_0.3.10          units_0.7-2            spdep_1.1-8           
##  [49] proxy_0.4-26           stats4_3.6.3           httr_1.4.2             ellipsis_0.3.2         farver_2.1.0           pkgconfig_2.0.3       
##  [55] sass_0.4.0             dbplyr_2.1.1           deldir_0.2-10          utf8_1.2.2             labeling_0.4.2         tidyselect_1.1.1      
##  [61] rlang_0.4.11           reshape2_1.4.4         later_1.2.0            munsell_0.5.0          cellranger_1.1.0       cachem_1.0.5          
##  [67] cli_3.0.1              generics_0.1.0         devtools_2.4.2         broom_0.7.9            evaluate_0.14          fastmap_1.1.0         
##  [73] yaml_2.2.1             RefManageR_1.3.0       processx_3.5.2         fs_1.5.0               nlme_3.1-152           mime_0.11             
##  [79] xml2_1.3.2             rstudioapi_0.13        compiler_3.6.3         curl_4.3.2             e1071_1.7-8            testthat_3.0.4        
##  [85] reprex_2.0.1           bslib_0.2.5.1          stringi_1.7.3          highr_0.9              ps_1.6.0               desc_1.3.0            
##  [91] lattice_0.20-44        Matrix_1.3-4           classInt_0.4-3         vegan_2.5-7            permute_0.9-5          vctrs_0.3.8           
##  [97] pillar_1.6.2           LearnBayes_2.15.1      lifecycle_1.0.0        BiocManager_1.30.16    jquerylib_0.1.4        bitops_1.0-7          
## [103] raster_3.4-13          httpuv_1.6.1           GenomicRanges_1.38.0   R6_2.5.0               promises_1.2.0.1       KernSmooth_2.23-20    
## [109] IRanges_2.20.2         sessioninfo_1.1.1      codetools_0.2-18       boot_1.3-28            MASS_7.3-54            gtools_3.9.2          
## [115] assertthat_0.2.1       pkgload_1.2.1          rprojroot_2.0.2        withr_2.4.2            S4Vectors_0.24.4       GenomeInfoDbData_1.2.2
## [121] mgcv_1.8-36            expm_0.999-6           parallel_3.6.3         hms_1.1.0              grid_3.6.3             coda_0.19-4           
## [127] class_7.3-19           rmarkdown_2.10         sf_1.0-2               shiny_1.6.0            lubridate_1.7.10</code></pre>
<blockquote>
<p>Note that any text after a ‘#’ symbol is a comment and does not affect the code execution; you can just type <code>getwd()</code> after the “&gt;”.</p>
</blockquote>
</div>
<div id="packages-and-getting-help" class="section level2">
<h2>Packages and getting help</h2>
<p>One way that R shines above other languages is that R packages in CRAN are all documented and easy to install. Help files are written in HTML and give the user a brief overview of:</p>
<ul>
<li>the purpose of a function</li>
<li>the parameters it takes</li>
<li>the output it yields</li>
<li>examples demonstrating its usage</li>
</ul>
<p>To get help on any R function, type a question mark before the empty function. Here’s an example of how to get help about the <code>gather()</code> function from the <em>tidyr</em> package:</p>
<pre class="r"><code>library(&#39;tidyr&#39;) # The package with the gather() function.
?gather          # open the R documentation of the function gather()</code></pre>
<p>If you want to run the examples, you can either copy and paste the commands to your R console, or you can run them all with:</p>
<pre class="r"><code>example(&quot;gather&quot;, package = &quot;tidyr&quot;)</code></pre>
<p>Other ways of getting help:</p>
<pre class="r"><code>help(package = &quot;tidyr&quot;)  # Get help for a package.
help(&quot;gather&quot;)           # Get help for the gather function
?gather                  # same as above
??multilocus             # Search for help that has the keyword &#39;multilocus&#39; in all packages</code></pre>
<p>Some packages include vignettes that can have different formats such as being introductions, tutorials, or reference cards in PDF format. You can look at a list of vignettes in all packages by typing:</p>
<pre class="r"><code>browseVignettes()                     # see vignettes from all packages
browseVignettes(package = &#39;poppr&#39;)    # see vignettes from a specific package.</code></pre>
<p>and to look at a specific vignette you can type:</p>
<pre class="r"><code>vignette(&#39;mlg&#39;)</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
